// @wix/cli/build/chunk-TKDDSH55.js
import { createRequire as _createRequire } from 'node:module';
const require = _createRequire(import.meta.url);
import {
  z
} from "./chunk-ZXYGJZOO.js";
import {
  init_esm_shims
} from "./chunk-4EFJZ3GQ.js";

// ../cli-core-definitions/src/index.ts
init_esm_shims();
import { join as join2 } from "node:path";

// ../velo-github-layout-definitions/build/index.js
init_esm_shims();
import { join } from "node:path";
function getGitIgnoredFolder(projectFolder = "") {
  return join(projectFolder, ".wix");
}
function getTypesDir(projectFolder) {
  return join(getGitIgnoredFolder(projectFolder), "types");
}
function getSourceFolder(projectFolder) {
  return join(projectFolder, "src");
}
function getPagesFolder(projectFolder = "") {
  return join(getSourceFolder(projectFolder), "pages");
}

// ../cli-core-definitions/src/index.ts
var SENTRY_DSN = "https://d15bb47ad34c428f92d0ac434a5734eb@o37417.ingest.sentry.io/4504116669317120";
var WIX_CONFIG_FILENAME = "wix.config.json";
var WIX_LOCK_FILENAME = "wix.lock";
var VERSION_CACHE_FILENAME = "version.cache.json";
var USER_CONFIG_FILENAME = "user.config.json";
var userConfigSchema = z.object({
  telemetry: z.boolean().default(true),
  tunneling: z.boolean().default(false)
});
function getWixConfigFilePath(projectFolder) {
  return join2(projectFolder, WIX_CONFIG_FILENAME);
}
function getDebugLogFilePath(projectFolder) {
  return join2(getGitIgnoredFolder(projectFolder), "debug.log");
}
function getUserFeedbackCacheFilePath(projectFolder) {
  return join2(getGitIgnoredFolder(projectFolder), "user-feedback.cache.json");
}
function getWixLockFilePath(projectFolder) {
  return join2(projectFolder, WIX_LOCK_FILENAME);
}

export {
  getTypesDir,
  getSourceFolder,
  getPagesFolder,
  SENTRY_DSN,
  WIX_CONFIG_FILENAME,
  WIX_LOCK_FILENAME,
  VERSION_CACHE_FILENAME,
  USER_CONFIG_FILENAME,
  userConfigSchema,
  getWixConfigFilePath,
  getDebugLogFilePath,
  getUserFeedbackCacheFilePath,
  getWixLockFilePath
};
