// @wix/cli/build/chunk-W57SYDA7.js
import { createRequire as _createRequire } from 'node:module';
const require = _createRequire(import.meta.url);
import {
  useBackendAsAServiceClient
} from "./chunk-MC5QQOVO.js";
import {
  ENV_FILE_NAME,
  getEnvFilePath,
  useProjectModel
} from "./chunk-HP474T73.js";
import {
  SiteAuthProvider
} from "./chunk-67NUBEMP.js";
import {
  Alert,
  Spinner,
  useAsync
} from "./chunk-RXATQXHV.js";
import {
  require_react
} from "./chunk-NRAQAV6T.js";
import {
  updateEnvFile
} from "./chunk-Z3Y5NFYG.js";
import {
  require_lib
} from "./chunk-IVMK6TVY.js";
import {
  __toESM,
  init_esm_shims
} from "./chunk-4EFJZ3GQ.js";

// ../cli-astro-commands/src/components/EnvPullCommand/index.ts
init_esm_shims();

// ../cli-astro-commands/src/components/EnvPullCommand/EnvPullCommand.tsx
init_esm_shims();
var import_react2 = __toESM(require_react(), 1);

// ../cli-astro-commands/src/components/EnvPullCommand/EnvPull.tsx
init_esm_shims();
var import_react = __toESM(require_react(), 1);
var import_variant = __toESM(require_lib(), 1);
var ENV_NAME = "prod";
var EnvPull = () => {
  const { model } = useProjectModel();
  const backendAsAServiceClient = useBackendAsAServiceClient();
  const { status } = useAsync(async () => {
    const result = await backendAsAServiceClient.getAppEnvironmentByName(
      model.config.appId,
      ENV_NAME
    );
    await updateEnvFile(getEnvFilePath(model.projectFolder), result.variables);
  }, []);
  return (0, import_variant.match)(status, {
    Loading: () => /* @__PURE__ */ import_react.default.createElement(Spinner, { text: `Pulling environment "${ENV_NAME}" variables...` }),
    Success: () => /* @__PURE__ */ import_react.default.createElement(Alert, { type: "success" }, 'Environment variables written to "', ENV_FILE_NAME, '"'),
    Error: () => null,
    NotRequested: () => null
  });
};

// ../cli-astro-commands/src/components/EnvPullCommand/EnvPullCommand.tsx
var EnvPullCommand = () => {
  const {
    model: {
      config: { siteId }
    }
  } = useProjectModel();
  return /* @__PURE__ */ import_react2.default.createElement(SiteAuthProvider, { siteId }, /* @__PURE__ */ import_react2.default.createElement(EnvPull, null));
};

export {
  EnvPullCommand
};
