// @wix/cli/build/chunk-CSMKGG4V.js
import { createRequire as _createRequire } from 'node:module';
const require = _createRequire(import.meta.url);
import {
  init_esm_shims
} from "./chunk-4EFJZ3GQ.js";

// ../../node_modules/@wix/ambassador-document-management-editor-deployments-v1-deployment/build/es/types.impl.js
init_esm_shims();
var DeploymentType;
(function(DeploymentType2) {
  DeploymentType2["UNKNOWN"] = "UNKNOWN";
  DeploymentType2["EDITOR_REVISION"] = "EDITOR_REVISION";
})(DeploymentType || (DeploymentType = {}));
var PipelineStatus;
(function(PipelineStatus2) {
  PipelineStatus2["UNDEFINED"] = "UNDEFINED";
  PipelineStatus2["IN_PROGRESS"] = "IN_PROGRESS";
  PipelineStatus2["SUCCESS"] = "SUCCESS";
  PipelineStatus2["ERROR"] = "ERROR";
  PipelineStatus2["INVOKE_ERROR"] = "INVOKE_ERROR";
  PipelineStatus2["TIMEOUT"] = "TIMEOUT";
  PipelineStatus2["SKIPPED"] = "SKIPPED";
})(PipelineStatus || (PipelineStatus = {}));
var TaskStatus;
(function(TaskStatus2) {
  TaskStatus2["UNDEFINED"] = "UNDEFINED";
  TaskStatus2["SUCCESS"] = "SUCCESS";
  TaskStatus2["ERROR"] = "ERROR";
  TaskStatus2["RUNNING"] = "RUNNING";
  TaskStatus2["ABORTED"] = "ABORTED";
})(TaskStatus || (TaskStatus = {}));
var DeploymentPipelinesStatus;
(function(DeploymentPipelinesStatus2) {
  DeploymentPipelinesStatus2["UNDEFINED"] = "UNDEFINED";
  DeploymentPipelinesStatus2["NONE"] = "NONE";
  DeploymentPipelinesStatus2["IN_PROGRESS"] = "IN_PROGRESS";
  DeploymentPipelinesStatus2["SUCCESS"] = "SUCCESS";
  DeploymentPipelinesStatus2["ERROR"] = "ERROR";
})(DeploymentPipelinesStatus || (DeploymentPipelinesStatus = {}));
var RCLabel;
(function(RCLabel2) {
  RCLabel2["UNKNOWN"] = "UNKNOWN";
  RCLabel2["RELEASE_MANAGER"] = "RELEASE_MANAGER";
  RCLabel2["BLOCKS"] = "BLOCKS";
  RCLabel2["WIX_CLI"] = "WIX_CLI";
  RCLabel2["MOBILE_APP_BUILDER"] = "MOBILE_APP_BUILDER";
  RCLabel2["SEO"] = "SEO";
  RCLabel2["FEEDBACK"] = "FEEDBACK";
  RCLabel2["ACCESSIBILITY"] = "ACCESSIBILITY";
})(RCLabel || (RCLabel = {}));
var PublishType;
(function(PublishType2) {
  PublishType2["NONE"] = "NONE";
  PublishType2["SITE"] = "SITE";
  PublishType2["RC"] = "RC";
})(PublishType || (PublishType = {}));
var SortOrder;
(function(SortOrder2) {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
})(SortOrder || (SortOrder = {}));
var WebhookIdentityType;
(function(WebhookIdentityType2) {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
})(WebhookIdentityType || (WebhookIdentityType = {}));

export {
  TaskStatus,
  DeploymentPipelinesStatus
};
