// @wix/cli/build/chunk-IRABMMYU.js
import { createRequire as _createRequire } from 'node:module';
const require = _createRequire(import.meta.url);
import {
  useHttpClient
} from "./chunk-4R5RNIW3.js";
import {
  ComponentType
} from "./chunk-HP474T73.js";
import {
  isHttpApplicationError,
  isHttpError,
  pRetry,
  resolveUrl,
  serializer,
  toURLSearchParams
} from "./chunk-67NUBEMP.js";
import {
  require_react
} from "./chunk-NRAQAV6T.js";
import {
  getTestOverrides
} from "./chunk-WYHHEOWO.js";
import {
  z
} from "./chunk-ZXYGJZOO.js";
import {
  CliError,
  CliErrorCode
} from "./chunk-IVMK6TVY.js";
import {
  __toESM,
  init_esm_shims
} from "./chunk-4EFJZ3GQ.js";

// ../dev-center-client/src/useDevCenterClient.ts
init_esm_shims();
var import_react = __toESM(require_react(), 1);

// ../dev-center-client/src/dev-center-client.ts
init_esm_shims();

// ../../node_modules/@wix/ambassador-devcenter-apps-v1-app-old/build/es/http.impl.js
init_esm_shims();
var __Array = {
  checkboxGroupOptions: "_checkboxGroup",
  validation: "_arrayType"
};
var __Boolean = { checkboxOptions: "_checkbox" };
var __Function = { description: "_description" };
var __Number = {
  minimum: "google.protobuf.FloatValue",
  maximum: "google.protobuf.FloatValue",
  multipleOf: "google.protobuf.FloatValue"
};
var __Object = { validation: "_objectType" };
var __String = {
  textInputOptions: "_textInput",
  radioGroupOptions: "_radioGroup",
  dropdownOptions: "_dropdown",
  dateTimeOptions: "_dateTimeInput",
  phoneInputOptions: "_phoneInput",
  dateInputOptions: "_dateInput",
  timeInputOptions: "_timeInput",
  datePickerOptions: "_datePicker"
};
var _actionProviderSPIConfig = { actionConfig: "_actionSPIConfig" };
var _actionSPIConfig = { icon: "_wixCommonImage" };
var _adminConfigurableTextInput = {
  requiredSettings: "google.protobuf.FieldMask",
  adminConfigurableSettings: "google.protobuf.FieldMask"
};
var _andCondition = { conditions: "_conditionNode" };
var _appData = { components: "_component", pricingModels: "_pricingModel" };
var _appEmbedData = { image: "_media" };
var _arrayItems = { data: "_dataItems", dataItem: "_dataItem" };
var _arrayType = { items: "_arrayTypeArrayItems" };
var _arrayTypeArrayItems = {
  numberOptions: "_numberType",
  objectOptions: "_objectType"
};
var _audioData = { audio: "_media", coverImage: "_media" };
var _authenticatorConfig = { expectedInputs: "_expectedInputs" };
var _background = { image: "_media" };
var _billing = {
  monthlyPrice: "DOUBLE",
  yearlyPrice: "DOUBLE",
  oneTimePrice: "DOUBLE",
  meteredBilling: "_meteredBilling"
};
var _checkbox = { label: "_richContent" };
var _checkboxGroup = {
  description: "_richContent",
  options: "_checkboxGroupOption"
};
var _checkboxGroupOption = { media: "_mediaItem" };
var _commonImage = { urlExpirationDate: "google.protobuf.Timestamp" };
var _component = { compData: "_componentData" };
var _componentData = {
  widgetComponentData: "_widgetComponentData",
  widgetOutOfIframeData: "_widgetOutOfIframeComponentData",
  studioWidgetComponentData: "_studioWidgetComponentData",
  shippingLabelCarrier: "_shippingLabelCarrierSpiConfig",
  fictionalShippingProvider: "_shippingProviderConfig",
  automationsActionProvider: "_actionProviderSPIConfig",
  identityAuthenticator: "_authenticatorConfig",
  devCenterTestingComponent: "_devCenterTestingComponentData",
  multilingualTranslationSchema: "_multilingualTranslationSchema",
  formTemplate: "_formTemplate",
  functionsShopPriceProvider: "_functionsShopPriceSpiConfig",
  function: "__Function",
  papiProvider: "_pluginConfig",
  multilingualTranslationSchemaGroup: "_schemaGroup",
  editorReactComponent: "_editorReactComponent"
};
var _componentsMultilineAddress = { description: "_richContent" };
var _conditionNode = { and: "_andCondition", or: "_orCondition" };
var _cssNumber = {
  minimum: "google.protobuf.FloatValue",
  maximum: "google.protobuf.FloatValue",
  multipleOf: "google.protobuf.FloatValue"
};
var _dataItem = { number: "__Number", arrayItems: "_arrayItems" };
var _dataItems = { items: "Map#_dataItem" };
var _dateInput = { description: "_richContent" };
var _datePicker = { description: "_richContent" };
var _dateTimeInput = { description: "_richContent" };
var _description = { representativeImage: "_wixCommonImage" };
var _design = { poll: "_pollDesign" };
var _devCenterTestingComponentData = {
  translatableMessageInOneOf: "_messageContainingTranslatables",
  translatableMessage: "_messageContainingTranslatables",
  mapOfTranslatableMessages: "Map#_messageContainingTranslatables",
  translatableRichContent: "_richContent",
  nonTranslatableRichContent: "_richContent",
  richContentMap: "Map#_richContent"
};
var _displayField = { richContentOptions: "_richContentOptions" };
var _donationInput = { description: "_richContent" };
var _dropdown = { description: "_richContent" };
var _editorElement = {
  style: "Map#_styleItem",
  data: "Map#_dataItem",
  elements: "Map#_elementItem",
  presets: "Map#_presetItem"
};
var _editorPresence = { presetsEditorPresence: "_presetEditorPresence" };
var _editorReactComponent = { editorElement: "_editorElement" };
var _elementItem = {
  inlineElement: "_inlineElement",
  refElement: "_refElement"
};
var _elementStyleDefaults = { elements: "Map#_elementStyleDefaults" };
var _expectedInputs = {
  expectedStartInputs: "_predefinedExpectedInput",
  expectedVerifyInputs: "_predefinedExpectedInput"
};
var _field = { inputOptions: "_inputField", displayOptions: "_displayField" };
var _fileUpload = { description: "_richContent" };
var _focalPoint = { x: "DOUBLE", y: "DOUBLE" };
var _formTemplate = {
  createdDate: "google.protobuf.Timestamp",
  updatedDate: "google.protobuf.Timestamp",
  fields: "_field",
  deletedFields: "_field",
  rules: "_v2Rule",
  submitSettings: "_submitSettings"
};
var _functionsShopPriceSpiConfig = {
  shopNumber: "google.protobuf.DoubleValue"
};
var _galleryData = { items: "_item", options: "_galleryOptions" };
var _galleryOptions = { item: "_itemStyle" };
var _getAppByVersionRequest = {};
var _getAppResponse = {
  dateCreated: "google.protobuf.Timestamp",
  data: "_appData"
};
var _image = { urlExpirationDate: "google.protobuf.Timestamp" };
var _imageData = { image: "_media" };
var _inlineElement = {
  style: "Map#_styleItem",
  data: "Map#_dataItem",
  elements: "Map#_elementItem",
  presets: "Map#_presetItem"
};
var _inputField = {
  stringOptions: "__String",
  numberOptions: "_inputField_Number",
  booleanOptions: "__Boolean",
  arrayOptions: "__Array",
  objectOptions: "__Object",
  wixFileOptions: "_wixFile",
  paymentOptions: "_payment",
  multilineAddressOptions: "_multilineAddress"
};
var _inputField_Number = {
  numberInputOptions: "_numberInput",
  ratingInputOptions: "_ratingInput",
  validation: "_numberType"
};
var _item = { image: "_itemImage", video: "_video" };
var _itemImage = { media: "_media" };
var _itemStyle = { ratio: "google.protobuf.DoubleValue" };
var _mapData = { mapSettings: "_mapSettings" };
var _mapSettings = {
  lat: "google.protobuf.DoubleValue",
  lng: "google.protobuf.DoubleValue"
};
var _media = { duration: "google.protobuf.DoubleValue" };
var _mediaItem = { image: "_commonImage" };
var _messageContainingTranslatables = {
  innerTranslatableRichContent: "_richContent"
};
var _meteredBilling = {
  monthlyBaseFee: "DOUBLE",
  additionalCharges: "google.protobuf.DoubleValue"
};
var _multilineAddress = {
  multilineAddressOptions: "_componentsMultilineAddress"
};
var _multilingualTranslationSchema = { referenceData: "_schema" };
var _node = {
  galleryData: "_galleryData",
  imageData: "_imageData",
  mapData: "_mapData",
  pollData: "_pollData",
  appEmbedData: "_appEmbedData",
  videoData: "_videoData",
  audioData: "_audioData",
  nodes: "_node"
};
var _numberInput = {
  default: "google.protobuf.DoubleValue",
  description: "_richContent"
};
var _numberType = {
  maximum: "google.protobuf.DoubleValue",
  minimum: "google.protobuf.DoubleValue",
  multipleOf: "google.protobuf.DoubleValue",
  enum: "google.protobuf.DoubleValue"
};
var _objectType = { properties: "Map#_propertiesType" };
var _orCondition = { conditions: "_conditionNode" };
var _packageDimension = {
  width: "DOUBLE",
  length: "DOUBLE",
  height: "google.protobuf.DoubleValue"
};
var _packageType = { dimension: "_packageDimension", image: "_image" };
var _payment = {
  checkboxGroupOptions: "_productCheckboxGroup",
  donationInputOptions: "_donationInput"
};
var _phoneInput = { description: "_richContent" };
var _plan = { billing: "_billing" };
var _pluginConfig = { startingEnforcementDate: "google.protobuf.Timestamp" };
var _poll = { image: "_media", options: "_pollOption" };
var _pollData = { poll: "_poll", design: "_design" };
var _pollDesign = { background: "_background" };
var _pollOption = { image: "_media" };
var _predefinedExpectedInput = {
  configuration: "_predefinedExpectedInputConfiguration"
};
var _predefinedExpectedInputConfiguration = {
  textInput: "_adminConfigurableTextInput"
};
var _presetEditorPresence = { wixMediaThumbnail: "_wixCommonImage" };
var _presetItem = { elementsStyleDefaults: "Map#_elementStyleDefaults" };
var _pricingModel = { plans: "_plan" };
var _productCheckboxGroup = {
  description: "_richContent",
  options: "_productCheckboxGroupOption"
};
var _productCheckboxGroupOption = { media: "_mediaItem" };
var _propertiesType = {
  numberOptions: "_numberType",
  arrayOptions: "_arrayType"
};
var _radioGroup = { description: "_richContent" };
var _ratingInput = { description: "_richContent" };
var _refElement = {
  resolvedElement: "_inlineElement",
  elementsDefaults: "Map#_refInnerElementDefaults"
};
var _refInnerElementDefaults = { elements: "Map#_refInnerElementDefaults" };
var _richContent = { nodes: "_node", metadata: "_v1Metadata" };
var _richContentOptions = { richContent: "_richContent" };
var _schema = {
  createdDate: "google.protobuf.Timestamp",
  updatedDate: "google.protobuf.Timestamp",
  fields: "Map#_v1SchemaField"
};
var _schemaGroup = {
  createdDate: "google.protobuf.Timestamp",
  updatedDate: "google.protobuf.Timestamp"
};
var _shippingLabelCarrierSpiConfig = { packageTypes: "_packageType" };
var _shippingProviderConfig = { shippingPrice: "DOUBLE" };
var _signature = { description: "_richContent" };
var _studioWidgetComponentData = {
  widgetDisplay: "_widgetDisplay",
  editorPresence: "_editorPresence"
};
var _styleItem = { number: "_cssNumber" };
var _submitSettings = { thankYouMessageOptions: "_thankYouMessageOptions" };
var _textInput = { description: "_richContent" };
var _thankYouMessageOptions = { richContent: "_richContent" };
var _timeInput = { description: "_richContent" };
var _v1Metadata = {
  createdTimestamp: "google.protobuf.Timestamp",
  updatedTimestamp: "google.protobuf.Timestamp"
};
var _v1SchemaField = { index: "google.protobuf.DoubleValue" };
var _v2Rule = { expression: "_conditionNode" };
var _video = { media: "_media", thumbnail: "_media" };
var _videoData = { video: "_media", thumbnail: "_media" };
var _widgetComponentData = { widgetDisplay: "_widgetDisplay" };
var _widgetDisplay = { price: "google.protobuf.FloatValue" };
var _widgetOutOfIframeComponentData = { widgetData: "_widgetComponentData" };
var _wixCommonImage = {
  urlExpirationDate: "google.protobuf.Timestamp",
  focalPoint: "_focalPoint"
};
var _wixFile = {
  fileUploadOptions: "_fileUpload",
  signatureOptions: "_signature"
};
function resolveComWixpressAppserviceApiAppsUrl(opts) {
  var domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/app-service-webapp",
        destPath: ""
      }
    ],
    "dev._base_domain_": [
      {
        srcPath: "/_api/app-service",
        destPath: "/api"
      },
      {
        srcPath: "/_api/app-service-webapp",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/app-service",
        destPath: "/api"
      },
      {
        srcPath: "/_api/app-service-webapp",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/app-service",
        destPath: "/api"
      },
      {
        srcPath: "/_api/app-service-webapp",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/app-service",
        destPath: "/api"
      },
      {
        srcPath: "/_api/app-service-webapp",
        destPath: ""
      }
    ],
    "bo._base_domain_": [
      {
        srcPath: "/_api/app-service",
        destPath: "/api"
      },
      {
        srcPath: "/_api/app-service-webapp",
        destPath: ""
      },
      {
        srcPath: "/app-service-webapp",
        destPath: "/api"
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/_api/app-service",
        destPath: "/api"
      },
      {
        srcPath: "/_api/app-service-webapp",
        destPath: ""
      },
      {
        srcPath: "/app-service-webapp",
        destPath: "/api"
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/_api/app-service",
        destPath: "/api"
      },
      {
        srcPath: "/_api/app-service-webapp",
        destPath: ""
      },
      {
        srcPath: "/app-service-webapp",
        destPath: "/api"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/app-service",
        destPath: "/api"
      },
      {
        srcPath: "/_api/app-service-webapp",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/app-service",
        destPath: "/api"
      },
      {
        srcPath: "/_api/app-service-webapp",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
function getAppByVersion(payload) {
  var _a = serializer(_getAppByVersionRequest, {}), toReq = _a.toJSON, fromReq = _a.fromJSON;
  var fromRes = serializer(_getAppResponse, {
    __Array,
    __Boolean,
    __Function,
    __Number,
    __Object,
    __String,
    _actionProviderSPIConfig,
    _actionSPIConfig,
    _adminConfigurableTextInput,
    _andCondition,
    _appData,
    _appEmbedData,
    _arrayItems,
    _arrayType,
    _arrayTypeArrayItems,
    _audioData,
    _authenticatorConfig,
    _background,
    _billing,
    _checkbox,
    _checkboxGroup,
    _checkboxGroupOption,
    _commonImage,
    _component,
    _componentData,
    _componentsMultilineAddress,
    _conditionNode,
    _cssNumber,
    _dataItem,
    _dataItems,
    _dateInput,
    _datePicker,
    _dateTimeInput,
    _description,
    _design,
    _devCenterTestingComponentData,
    _displayField,
    _donationInput,
    _dropdown,
    _editorElement,
    _editorPresence,
    _editorReactComponent,
    _elementItem,
    _elementStyleDefaults,
    _expectedInputs,
    _field,
    _fileUpload,
    _focalPoint,
    _formTemplate,
    _functionsShopPriceSpiConfig,
    _galleryData,
    _galleryOptions,
    _image,
    _imageData,
    _inlineElement,
    _inputField,
    _inputField_Number,
    _item,
    _itemImage,
    _itemStyle,
    _mapData,
    _mapSettings,
    _media,
    _mediaItem,
    _messageContainingTranslatables,
    _meteredBilling,
    _multilineAddress,
    _multilingualTranslationSchema,
    _node,
    _numberInput,
    _numberType,
    _objectType,
    _orCondition,
    _packageDimension,
    _packageType,
    _payment,
    _phoneInput,
    _plan,
    _pluginConfig,
    _poll,
    _pollData,
    _pollDesign,
    _pollOption,
    _predefinedExpectedInput,
    _predefinedExpectedInputConfiguration,
    _presetEditorPresence,
    _presetItem,
    _pricingModel,
    _productCheckboxGroup,
    _productCheckboxGroupOption,
    _propertiesType,
    _radioGroup,
    _ratingInput,
    _refElement,
    _refInnerElementDefaults,
    _richContent,
    _richContentOptions,
    _schema,
    _schemaGroup,
    _shippingLabelCarrierSpiConfig,
    _shippingProviderConfig,
    _signature,
    _studioWidgetComponentData,
    _styleItem,
    _submitSettings,
    _textInput,
    _thankYouMessageOptions,
    _timeInput,
    _v1Metadata,
    _v1SchemaField,
    _v2Rule,
    _video,
    _videoData,
    _widgetComponentData,
    _widgetDisplay,
    _widgetOutOfIframeComponentData,
    _wixCommonImage,
    _wixFile
  }).fromJSON;
  function __getAppByVersion(_a2) {
    var host = _a2.host;
    var serializedData = toReq(payload);
    var metadata = {
      entityFqdn: "wix.devcenter.apps.v1.app_old",
      method: "GET",
      methodFqn: "com.wixpress.appservice.api.Apps.GetAppByVersion",
      url: resolveComWixpressAppserviceApiAppsUrl({
        protoPath: "/v1/apps/{appId}/version/{version}",
        data: serializedData,
        host
      }),
      params: toURLSearchParams(serializedData),
      transformResponse: fromRes,
      fallback: [
        {
          method: "GET",
          url: resolveComWixpressAppserviceApiAppsUrl({
            protoPath: "/v1/apps/{appId}/version/{version}",
            data: serializedData,
            host
          }),
          params: toURLSearchParams(serializedData)
        }
      ]
    };
    return metadata;
  }
  __getAppByVersion.fromReq = fromReq;
  __getAppByVersion.__isAmbassador = true;
  return __getAppByVersion;
}

// ../../node_modules/@wix/ambassador-devcenter-apps-v1-app/build/es/http.impl.js
init_esm_shims();
var _queryAppRequest = {};
var _queryAppResponse = { apps: "_v1App" };
var _v1App = {
  createdDate: "google.protobuf.Timestamp",
  updatedDate: "google.protobuf.Timestamp"
};
var _v1CreateAppRequest = { app: "_v1App" };
var _v1CreateAppResponse = { app: "_v1App" };
var _v1GetAppRequest = {};
var _v1GetAppResponse = { app: "_v1App" };
function resolveWixDevcenterAppsV1AppsServiceUrl(opts) {
  var domainToMappings = {
    "editor._base_domain_": [
      {
        srcPath: "/apps-service",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/apps-service",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/apps-service",
        destPath: ""
      }
    ],
    "bo._base_domain_": [
      {
        srcPath: "/apps-service",
        destPath: ""
      },
      {
        srcPath: "/apps-service",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/apps-service",
        destPath: ""
      },
      {
        srcPath: "/apps-service",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/apps-service",
        destPath: ""
      },
      {
        srcPath: "/apps-service",
        destPath: ""
      }
    ],
    "dev._base_domain_": [
      {
        srcPath: "/apps-service",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/apps-service",
        destPath: "/api"
      },
      {
        srcPath: "/apps-service",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/apps-service",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
function createApp(payload) {
  var _a = serializer(_v1CreateAppRequest, {
    _v1App
  }), toReq = _a.toJSON, fromReq = _a.fromJSON;
  var fromRes = serializer(_v1CreateAppResponse, { _v1App }).fromJSON;
  function __createApp(_a2) {
    var host = _a2.host;
    var serializedData = toReq(payload);
    var metadata = {
      entityFqdn: "wix.devcenter.apps.v1.app",
      method: "POST",
      methodFqn: "wix.devcenter.apps.v1.AppsService.CreateApp",
      url: resolveWixDevcenterAppsV1AppsServiceUrl({
        protoPath: "/v1/apps",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: fromRes
    };
    return metadata;
  }
  __createApp.fromReq = fromReq;
  __createApp.__isAmbassador = true;
  return __createApp;
}
function getApp(payload) {
  var _a = serializer(_v1GetAppRequest, {}), toReq = _a.toJSON, fromReq = _a.fromJSON;
  var fromRes = serializer(_v1GetAppResponse, { _v1App }).fromJSON;
  function __getApp(_a2) {
    var host = _a2.host;
    var serializedData = toReq(payload);
    var metadata = {
      entityFqdn: "wix.devcenter.apps.v1.app",
      method: "GET",
      methodFqn: "wix.devcenter.apps.v1.AppsService.GetApp",
      url: resolveWixDevcenterAppsV1AppsServiceUrl({
        protoPath: "/v1/apps/{appId}",
        data: serializedData,
        host
      }),
      params: toURLSearchParams(serializedData),
      transformResponse: fromRes
    };
    return metadata;
  }
  __getApp.fromReq = fromReq;
  __getApp.__isAmbassador = true;
  return __getApp;
}
function queryApp(payload) {
  var _a = serializer(_queryAppRequest, {}), toReq = _a.toJSON, fromReq = _a.fromJSON;
  var fromRes = serializer(_queryAppResponse, { _v1App }).fromJSON;
  function __queryApp(_a2) {
    var host = _a2.host;
    var serializedData = toReq(payload);
    var metadata = {
      entityFqdn: "wix.devcenter.apps.v1.app",
      method: "POST",
      methodFqn: "wix.devcenter.apps.v1.AppsService.QueryApp",
      url: resolveWixDevcenterAppsV1AppsServiceUrl({
        protoPath: "/v1/apps/query",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: fromRes
    };
    return metadata;
  }
  __queryApp.fromReq = fromReq;
  __queryApp.__isAmbassador = true;
  return __queryApp;
}

// ../../node_modules/@wix/ambassador-headless-v1-o-auth-app/build/es/http.impl.js
init_esm_shims();
var _createOAuthAppRequest = { oAuthApp: "_oAuthApp" };
var _createOAuthAppResponse = { oAuthApp: "_oAuthApp" };
var _oAuthApp = { createdDate: "google.protobuf.Timestamp" };
var _updateOAuthAppRequest = {
  mask: "google.protobuf.FieldMask",
  oAuthApp: "_oAuthApp"
};
var _updateOAuthAppResponse = { oAuthApp: "_oAuthApp" };
function resolveWixHeadlessV1OAuthAppServiceUrl(opts) {
  var domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/oauth-app-service",
        destPath: ""
      },
      {
        srcPath: "/oauth-app",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/oauth-app",
        destPath: ""
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/oauth-app-service",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
function createOAuthApp(payload) {
  var _a = serializer(_createOAuthAppRequest, { _oAuthApp }), toReq = _a.toJSON, fromReq = _a.fromJSON;
  var fromRes = serializer(_createOAuthAppResponse, {
    _oAuthApp
  }).fromJSON;
  function __createOAuthApp(_a2) {
    var host = _a2.host;
    var serializedData = toReq(payload);
    var metadata = {
      entityFqdn: "wix.headless.v1.o_auth_app",
      method: "POST",
      methodFqn: "wix.headless.v1.OAuthAppService.CreateOAuthApp",
      url: resolveWixHeadlessV1OAuthAppServiceUrl({
        protoPath: "/v1/oauth-apps",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: fromRes
    };
    return metadata;
  }
  __createOAuthApp.fromReq = fromReq;
  __createOAuthApp.__isAmbassador = true;
  return __createOAuthApp;
}
function updateOAuthApp(payload) {
  var _a = serializer(_updateOAuthAppRequest, { _oAuthApp }), toReq = _a.toJSON, fromReq = _a.fromJSON;
  var fromRes = serializer(_updateOAuthAppResponse, {
    _oAuthApp
  }).fromJSON;
  function __updateOAuthApp(_a2) {
    var host = _a2.host;
    var serializedData = toReq(payload);
    var metadata = {
      entityFqdn: "wix.headless.v1.o_auth_app",
      method: "PATCH",
      methodFqn: "wix.headless.v1.OAuthAppService.UpdateOAuthApp",
      url: resolveWixHeadlessV1OAuthAppServiceUrl({
        protoPath: "/v1/oauth-apps/{oAuthApp.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: fromRes
    };
    return metadata;
  }
  __updateOAuthApp.fromReq = fromReq;
  __updateOAuthApp.__isAmbassador = true;
  return __updateOAuthApp;
}

// ../../node_modules/@wix/ambassador-devcenter-apps-v1-app-version/build/es/http.impl.js
init_esm_shims();
var _appVersion = {
  createdDate: "google.protobuf.Timestamp",
  updatedDate: "google.protobuf.Timestamp"
};
var _queryAppVersionRequest = {};
var _queryAppVersionResponse = { appVersions: "_appVersion" };
function resolveWixDevcenterAppsVersionsV1AppVersionsServiceUrl(opts) {
  var domainToMappings = {
    "bo._base_domain_": [
      {
        srcPath: "/app-versions",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/app-versions",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/app-versions",
        destPath: ""
      }
    ],
    "dev._base_domain_": [
      {
        srcPath: "/app-versions",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/app-versions",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/app-versions",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/app-versions",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/app-versions",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
function queryAppVersion(payload) {
  var _a = serializer(_queryAppVersionRequest, {}), toReq = _a.toJSON, fromReq = _a.fromJSON;
  var fromRes = serializer(_queryAppVersionResponse, {
    _appVersion
  }).fromJSON;
  function __queryAppVersion(_a2) {
    var host = _a2.host;
    var serializedData = toReq(payload);
    var metadata = {
      entityFqdn: "wix.devcenter.apps.v1.app_version",
      method: "POST",
      methodFqn: "wix.devcenter.apps.versions.v1.AppVersionsService.QueryAppVersion",
      url: resolveWixDevcenterAppsVersionsV1AppVersionsServiceUrl({
        protoPath: "/v1/app-versions/query",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: fromRes
    };
    return metadata;
  }
  __queryAppVersion.fromReq = fromReq;
  __queryAppVersion.__isAmbassador = true;
  return __queryAppVersion;
}

// ../../node_modules/@wix/ambassador-devcenter-apps-v1-app-version/build/es/types.impl.js
init_esm_shims();
var State;
(function(State3) {
  State3["UNKNOWN_STATE"] = "UNKNOWN_STATE";
  State3["UNLOCKED_DRAFT"] = "UNLOCKED_DRAFT";
  State3["LOCKED_DRAFT"] = "LOCKED_DRAFT";
  State3["PRODUCTION"] = "PRODUCTION";
})(State || (State = {}));
var Status;
(function(Status4) {
  Status4["UNKNOWN"] = "UNKNOWN";
  Status4["DEPRECATED"] = "DEPRECATED";
  Status4["ARCHIVED"] = "ARCHIVED";
})(Status || (Status = {}));
var CreatedByType;
(function(CreatedByType3) {
  CreatedByType3["UNKNOWN_CREATED_BY"] = "UNKNOWN_CREATED_BY";
  CreatedByType3["USER"] = "USER";
  CreatedByType3["SERVICE"] = "SERVICE";
  CreatedByType3["ACCOUNT"] = "ACCOUNT";
})(CreatedByType || (CreatedByType = {}));
var PromoteStrategy;
(function(PromoteStrategy2) {
  PromoteStrategy2["UNKNOWN_STRATEGY"] = "UNKNOWN_STRATEGY";
  PromoteStrategy2["ENTIRE_APP"] = "ENTIRE_APP";
})(PromoteStrategy || (PromoteStrategy = {}));
var VersionStatus;
(function(VersionStatus2) {
  VersionStatus2["NONE_STATUS"] = "NONE_STATUS";
  VersionStatus2["DRAFT"] = "DRAFT";
  VersionStatus2["SUBMITTED"] = "SUBMITTED";
  VersionStatus2["IN_REVIEW"] = "IN_REVIEW";
  VersionStatus2["APPROVED"] = "APPROVED";
  VersionStatus2["PUBLISHED"] = "PUBLISHED";
  VersionStatus2["DECLINED"] = "DECLINED";
  VersionStatus2["ARCHIVED"] = "ARCHIVED";
  VersionStatus2["HIDDEN"] = "HIDDEN";
  VersionStatus2["RELEASED"] = "RELEASED";
})(VersionStatus || (VersionStatus = {}));
var SortOrder;
(function(SortOrder2) {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
})(SortOrder || (SortOrder = {}));
var PlanStatus;
(function(PlanStatus2) {
  PlanStatus2["UNKNOWN"] = "UNKNOWN";
  PlanStatus2["VISIBLE"] = "VISIBLE";
  PlanStatus2["HIDDEN"] = "HIDDEN";
})(PlanStatus || (PlanStatus = {}));
var PlanDescription;
(function(PlanDescription2) {
  PlanDescription2["UNKNOWN_DESCRIPTION"] = "UNKNOWN_DESCRIPTION";
  PlanDescription2["BENEFITS"] = "BENEFITS";
  PlanDescription2["CUSTOM_TEXT"] = "CUSTOM_TEXT";
})(PlanDescription || (PlanDescription = {}));
var BusinessModel;
(function(BusinessModel2) {
  BusinessModel2["None"] = "None";
  BusinessModel2["FREEMIUM"] = "FREEMIUM";
  BusinessModel2["PREMIUM"] = "PREMIUM";
  BusinessModel2["FREE"] = "FREE";
  BusinessModel2["WIX_PREMIUM"] = "WIX_PREMIUM";
  BusinessModel2["DYNAMIC"] = "DYNAMIC";
})(BusinessModel || (BusinessModel = {}));
var Action;
(function(Action2) {
  Action2["CREATED"] = "CREATED";
  Action2["UPDATED"] = "UPDATED";
  Action2["RESTORED"] = "RESTORED";
  Action2["DELETED"] = "DELETED";
})(Action || (Action = {}));
var WebhookIdentityType;
(function(WebhookIdentityType2) {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
})(WebhookIdentityType || (WebhookIdentityType = {}));

// ../../node_modules/@wix/ambassador-devcenter-myapps-v1-my-app/build/es/http.impl.js
init_esm_shims();
var _myApp = {
  createdDate: "google.protobuf.Timestamp",
  updatedDate: "google.protobuf.Timestamp",
  reviews: "_reviews"
};
var _queryMyAppsRequest = {};
var _queryMyAppsResponse = { myApps: "_myApp" };
var _reviews = { averageRating: "google.protobuf.DoubleValue" };
function resolveWixDevcenterMyappsV1MyAppsServiceUrl(opts) {
  var domainToMappings = {
    "dev._base_domain_": [
      {
        srcPath: "/my-apps",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/my-apps",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/my-apps",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/my-apps",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/my-apps",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
function queryMyApps(payload) {
  var _a = serializer(_queryMyAppsRequest, {}), toReq = _a.toJSON, fromReq = _a.fromJSON;
  var fromRes = serializer(_queryMyAppsResponse, {
    _myApp,
    _reviews
  }).fromJSON;
  function __queryMyApps(_a2) {
    var host = _a2.host;
    var serializedData = toReq(payload);
    var metadata = {
      entityFqdn: "wix.devcenter.myapps.v1.my_app",
      method: "POST",
      methodFqn: "wix.devcenter.myapps.v1.MyAppsService.QueryMyApps",
      url: resolveWixDevcenterMyappsV1MyAppsServiceUrl({
        protoPath: "/v1/my-apps/query",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: fromRes
    };
    return metadata;
  }
  __queryMyApps.fromReq = fromReq;
  __queryMyApps.__isAmbassador = true;
  return __queryMyApps;
}

// ../../node_modules/@wix/ambassador-devcenter-dsm-v1-development-site/build/es/http.impl.js
init_esm_shims();
var _createDevelopmentSiteRequest = {};
var _createDevelopmentSiteResponse = {};
var _getDevelopmentSitesLimitRequest = {};
var _getDevelopmentSitesLimitResponse = {};
var _getDevelopmentSitesRequest = {};
var _getDevelopmentSitesResponse = {};
function resolveComWixpressDevcenterDsmDevelopmentSitesManagerUrl(opts) {
  var domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/development-sites-manager",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/development-sites-manager",
        destPath: ""
      },
      {
        srcPath: "/_api/development-sites-manager",
        destPath: ""
      }
    ],
    "bo._base_domain_": [
      {
        srcPath: "/development-sites-manager",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/development-sites-manager",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/development-sites-manager",
        destPath: ""
      }
    ],
    "www.wixapps.net": [
      {
        srcPath: "/development-sites-manager/v1/development-sites",
        destPath: ""
      }
    ],
    "dev._base_domain_": [
      {
        srcPath: "/development-sites-manager",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/development-sites-manager/v1/development-sites",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/development-sites-manager",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/development-sites-manager",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/development-sites-manager",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/development-sites-manager",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
function createDevelopmentSite(payload) {
  var _a = serializer(_createDevelopmentSiteRequest, {}), toReq = _a.toJSON, fromReq = _a.fromJSON;
  var fromRes = serializer(_createDevelopmentSiteResponse, {}).fromJSON;
  function __createDevelopmentSite(_a2) {
    var host = _a2.host;
    var serializedData = toReq(payload);
    var metadata = {
      entityFqdn: "wix.devcenter.dsm.v1.development_site",
      method: "POST",
      methodFqn: "com.wixpress.devcenter.dsm.DevelopmentSitesManager.CreateDevelopmentSite",
      url: resolveComWixpressDevcenterDsmDevelopmentSitesManagerUrl({
        protoPath: "/v1/development-sites",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: fromRes
    };
    return metadata;
  }
  __createDevelopmentSite.fromReq = fromReq;
  __createDevelopmentSite.__isAmbassador = true;
  return __createDevelopmentSite;
}
function getDevelopmentSites(payload) {
  var _a = serializer(_getDevelopmentSitesRequest, {}), toReq = _a.toJSON, fromReq = _a.fromJSON;
  var fromRes = serializer(_getDevelopmentSitesResponse, {}).fromJSON;
  function __getDevelopmentSites(_a2) {
    var host = _a2.host;
    var serializedData = toReq(payload);
    var metadata = {
      entityFqdn: "wix.devcenter.dsm.v1.development_site",
      method: "GET",
      methodFqn: "com.wixpress.devcenter.dsm.DevelopmentSitesManager.GetDevelopmentSites",
      url: resolveComWixpressDevcenterDsmDevelopmentSitesManagerUrl({
        protoPath: "/v1/development-sites",
        data: serializedData,
        host
      }),
      params: toURLSearchParams(serializedData),
      transformResponse: fromRes
    };
    return metadata;
  }
  __getDevelopmentSites.fromReq = fromReq;
  __getDevelopmentSites.__isAmbassador = true;
  return __getDevelopmentSites;
}
function getDevelopmentSitesLimit(payload) {
  var _a = serializer(_getDevelopmentSitesLimitRequest, {}), toReq = _a.toJSON, fromReq = _a.fromJSON;
  var fromRes = serializer(_getDevelopmentSitesLimitResponse, {}).fromJSON;
  function __getDevelopmentSitesLimit(_a2) {
    var host = _a2.host;
    var serializedData = toReq(payload);
    var metadata = {
      entityFqdn: "wix.devcenter.dsm.v1.development_site",
      method: "GET",
      methodFqn: "com.wixpress.devcenter.dsm.DevelopmentSitesManager.GetDevelopmentSitesLimit",
      url: resolveComWixpressDevcenterDsmDevelopmentSitesManagerUrl({
        protoPath: "/v1/development-sites-limit",
        data: serializedData,
        host
      }),
      params: toURLSearchParams(serializedData),
      transformResponse: fromRes
    };
    return metadata;
  }
  __getDevelopmentSitesLimit.fromReq = fromReq;
  __getDevelopmentSitesLimit.__isAmbassador = true;
  return __getDevelopmentSitesLimit;
}

// ../../node_modules/@wix/ambassador-devcenter-app-market-listing-v1-market-listing/build/es/http.impl.js
init_esm_shims();
var _marketListing = {
  createdDate: "google.protobuf.Timestamp",
  updatedDate: "google.protobuf.Timestamp"
};
var _queryMarketListingRequest = {};
var _queryMarketListingResponse = { marketListing: "_marketListing" };
var _updateMarketListingRequest = {
  mask: "google.protobuf.FieldMask",
  marketListing: "_marketListing"
};
var _updateMarketListingResponse = { marketListing: "_marketListing" };
function resolveComWixpressDevcenterAppmarketlistingMarketListingServiceUrl(opts) {
  var domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/app-market-listing",
        destPath: ""
      }
    ],
    "bo._base_domain_": [
      {
        srcPath: "/app-market-listing",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/app-market-listing",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/app-market-listing",
        destPath: ""
      }
    ],
    "dev._base_domain_": [
      {
        srcPath: "/app-market-listing",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/app-market-listing",
        destPath: ""
      },
      {
        srcPath: "/_api/app-market-listing",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/app-market-listing",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/app-market-listing",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/app-market-listing",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/app-market-listing",
        destPath: ""
      }
    ],
    "platform.rise.ai": [
      {
        srcPath: "/v1/marketListings",
        destPath: "/v1/marketListings"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
function updateMarketListing(payload) {
  var _a = serializer(_updateMarketListingRequest, { _marketListing }), toReq = _a.toJSON, fromReq = _a.fromJSON;
  var fromRes = serializer(_updateMarketListingResponse, {
    _marketListing
  }).fromJSON;
  function __updateMarketListing(_a2) {
    var host = _a2.host;
    var serializedData = toReq(payload);
    var metadata = {
      entityFqdn: "wix.devcenter.app_market_listing.v1.market_listing",
      method: "PUT",
      methodFqn: "com.wixpress.devcenter.appmarketlisting.MarketListingService.UpdateMarketListing",
      url: resolveComWixpressDevcenterAppmarketlistingMarketListingServiceUrl({
        protoPath: "/v1/marketListings/{marketListing.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: fromRes
    };
    return metadata;
  }
  __updateMarketListing.fromReq = fromReq;
  __updateMarketListing.__isAmbassador = true;
  return __updateMarketListing;
}
function queryMarketListing(payload) {
  var _a = serializer(_queryMarketListingRequest, {}), toReq = _a.toJSON, fromReq = _a.fromJSON;
  var fromRes = serializer(_queryMarketListingResponse, {
    _marketListing
  }).fromJSON;
  function __queryMarketListing(_a2) {
    var host = _a2.host;
    var serializedData = toReq(payload);
    var metadata = {
      entityFqdn: "wix.devcenter.app_market_listing.v1.market_listing",
      method: "POST",
      methodFqn: "com.wixpress.devcenter.appmarketlisting.MarketListingService.QueryMarketListing",
      url: resolveComWixpressDevcenterAppmarketlistingMarketListingServiceUrl({
        protoPath: "/v1/marketListings/query",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: fromRes
    };
    return metadata;
  }
  __queryMarketListing.fromReq = fromReq;
  __queryMarketListing.__isAmbassador = true;
  return __queryMarketListing;
}

// ../../node_modules/@wix/ambassador-devcenter-apps-v1-app-template/build/es/http.impl.js
init_esm_shims();
var _app = {
  createdDate: "google.protobuf.Timestamp",
  updatedDate: "google.protobuf.Timestamp"
};
var _createAppFromTemplateRequest = {};
var _createAppFromTemplateResponse = { app: "_app" };
var _createBlocksAppFromTemplateRequest = {};
var _createBlocksAppFromTemplateResponse = { app: "_app" };
var _listAppTemplatesRequest = {};
var _listAppTemplatesResponse = {};
function resolveWixDevcenterAppsTemplatesV1AppTemplatesServiceUrl(opts) {
  var domainToMappings = {
    "dev._base_domain_": [
      {
        srcPath: "/app-templates-service",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/app-templates-service",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
function createAppFromTemplate(payload) {
  var _a = serializer(_createAppFromTemplateRequest, {}), toReq = _a.toJSON, fromReq = _a.fromJSON;
  var fromRes = serializer(_createAppFromTemplateResponse, {
    _app
  }).fromJSON;
  function __createAppFromTemplate(_a2) {
    var host = _a2.host;
    var serializedData = toReq(payload);
    var metadata = {
      entityFqdn: "wix.devcenter.apps.v1.app_template",
      method: "POST",
      methodFqn: "wix.devcenter.apps.templates.v1.AppTemplatesService.CreateAppFromTemplate",
      url: resolveWixDevcenterAppsTemplatesV1AppTemplatesServiceUrl({
        protoPath: "/v1/create-app-from-template",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: fromRes
    };
    return metadata;
  }
  __createAppFromTemplate.fromReq = fromReq;
  __createAppFromTemplate.__isAmbassador = true;
  return __createAppFromTemplate;
}
function createBlocksAppFromTemplate(payload) {
  var _a = serializer(_createBlocksAppFromTemplateRequest, {}), toReq = _a.toJSON, fromReq = _a.fromJSON;
  var fromRes = serializer(_createBlocksAppFromTemplateResponse, { _app }).fromJSON;
  function __createBlocksAppFromTemplate(_a2) {
    var host = _a2.host;
    var serializedData = toReq(payload);
    var metadata = {
      entityFqdn: "wix.devcenter.apps.v1.app_template",
      method: "POST",
      methodFqn: "wix.devcenter.apps.templates.v1.AppTemplatesService.CreateBlocksAppFromTemplate",
      url: resolveWixDevcenterAppsTemplatesV1AppTemplatesServiceUrl({
        protoPath: "/v1/create-blocks-app-from-template",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: fromRes
    };
    return metadata;
  }
  __createBlocksAppFromTemplate.fromReq = fromReq;
  __createBlocksAppFromTemplate.__isAmbassador = true;
  return __createBlocksAppFromTemplate;
}
function listAppTemplates(payload) {
  var _a = serializer(_listAppTemplatesRequest, {}), toReq = _a.toJSON, fromReq = _a.fromJSON;
  var fromRes = serializer(_listAppTemplatesResponse, {}).fromJSON;
  function __listAppTemplates(_a2) {
    var host = _a2.host;
    var serializedData = toReq(payload);
    var metadata = {
      entityFqdn: "wix.devcenter.apps.v1.app_template",
      method: "GET",
      methodFqn: "wix.devcenter.apps.templates.v1.AppTemplatesService.ListAppTemplates",
      url: resolveWixDevcenterAppsTemplatesV1AppTemplatesServiceUrl({
        protoPath: "/v1/app-templates",
        data: serializedData,
        host
      }),
      params: toURLSearchParams(serializedData),
      transformResponse: fromRes
    };
    return metadata;
  }
  __listAppTemplates.fromReq = fromReq;
  __listAppTemplates.__isAmbassador = true;
  return __listAppTemplates;
}

// ../../node_modules/@wix/ambassador-devcenter-managedapps-v1-managed-app/build/es/http.impl.js
init_esm_shims();
var __Array2 = {
  checkboxGroupOptions: "_checkboxGroup",
  validation: "_arrayType"
};
var __Boolean2 = { checkboxOptions: "_checkbox" };
var __Number2 = {
  numberInputOptions: "_numberInput",
  ratingInputOptions: "_ratingInput",
  validation: "_numberType"
};
var __Object2 = { validation: "_objectType" };
var __String2 = {
  textInputOptions: "_textInput",
  radioGroupOptions: "_radioGroup",
  dropdownOptions: "_dropdown",
  dateTimeOptions: "_dateTimeInput",
  phoneInputOptions: "_phoneInput",
  dateInputOptions: "_dateInput",
  timeInputOptions: "_timeInput",
  datePickerOptions: "_datePicker"
};
var _actionProviderSPIConfig2 = { actionConfig: "_actionSPIConfig" };
var _actionSPIConfig2 = { icon: "_commonImage" };
var _adminConfigurableTextInput2 = {
  requiredSettings: "google.protobuf.FieldMask",
  adminConfigurableSettings: "google.protobuf.FieldMask"
};
var _andCondition2 = { conditions: "_conditionNode" };
var _api_Number = {
  minimum: "google.protobuf.FloatValue",
  maximum: "google.protobuf.FloatValue",
  multipleOf: "google.protobuf.FloatValue"
};
var _apiArrayItems = { data: "_dataItems", dataItem: "_dataItem" };
var _appEmbedData2 = { image: "_media" };
var _arrayItems2 = {
  numberOptions: "_numberType",
  objectOptions: "_objectType"
};
var _arrayType2 = { items: "_arrayItems" };
var _audioData2 = { audio: "_media", coverImage: "_media" };
var _authenticatorConfig2 = { expectedInputs: "_expectedInputs" };
var _background2 = { image: "_media" };
var _checkbox2 = { label: "_richContent" };
var _checkboxGroup2 = { description: "_richContent", options: "_option" };
var _commonImage2 = {
  urlExpirationDate: "google.protobuf.Timestamp",
  focalPoint: "_focalPoint"
};
var _component2 = { compData: "_componentData" };
var _componentData2 = {
  widgetComponentData: "_widgetComponentData",
  widgetOutOfIframeData: "_widgetOutOfIframeComponentData",
  studioWidgetComponentData: "_studioWidgetComponentData",
  shippingLabelCarrier: "_shippingLabelCarrierSpiConfig",
  fictionalShippingProvider: "_shippingProviderConfig",
  automationsActionProvider: "_actionProviderSPIConfig",
  identityAuthenticator: "_authenticatorConfig",
  devCenterTestingComponent: "_devCenterTestingComponentData",
  multilingualTranslationSchema: "_multilingualTranslationSchema",
  formTemplate: "_formTemplate",
  functionsShopPriceProvider: "_functionsShopPriceSpiConfig",
  function: "_function",
  papiProvider: "_pluginConfig",
  multilingualTranslationSchemaGroup: "_schemaGroup",
  editorReactComponent: "_editorReactComponent"
};
var _conditionNode2 = { and: "_andCondition", or: "_orCondition" };
var _cssNumber2 = {
  minimum: "google.protobuf.FloatValue",
  maximum: "google.protobuf.FloatValue",
  multipleOf: "google.protobuf.FloatValue"
};
var _dataItem2 = { number: "_api_Number", arrayItems: "_apiArrayItems" };
var _dataItems2 = { items: "Map#_dataItem" };
var _dateInput2 = { description: "_richContent" };
var _datePicker2 = { description: "_richContent" };
var _dateTimeInput2 = { description: "_richContent" };
var _description2 = { representativeImage: "_commonImage" };
var _design2 = { poll: "_pollDesign" };
var _devCenterTestingComponentData2 = {
  translatableMessageInOneOf: "_messageContainingTranslatables",
  translatableMessage: "_messageContainingTranslatables",
  mapOfTranslatableMessages: "Map#_messageContainingTranslatables",
  translatableRichContent: "_richContent",
  nonTranslatableRichContent: "_richContent",
  richContentMap: "Map#_richContent"
};
var _discount = { saleInfo: "_saleInfo" };
var _displayField2 = { richContentOptions: "_richContentOptions" };
var _donationInput2 = { description: "_richContent" };
var _dropdown2 = { description: "_richContent" };
var _editorElement2 = {
  style: "Map#_styleItem",
  data: "Map#_dataItem",
  elements: "Map#_elementItem",
  presets: "Map#_presetItem"
};
var _editorPresence2 = { presetsEditorPresence: "_presetEditorPresence" };
var _editorReactComponent2 = { editorElement: "_editorElement" };
var _elementItem2 = {
  inlineElement: "_inlineElement",
  refElement: "_refElement"
};
var _elementStyleDefaults2 = { elements: "Map#_elementStyleDefaults" };
var _expectedInputs2 = {
  expectedStartInputs: "_predefinedExpectedInput",
  expectedVerifyInputs: "_predefinedExpectedInput"
};
var _field2 = { inputOptions: "_inputField", displayOptions: "_displayField" };
var _fileUpload2 = { description: "_richContent" };
var _focalPoint2 = { x: "DOUBLE", y: "DOUBLE" };
var _formTemplate2 = {
  createdDate: "google.protobuf.Timestamp",
  updatedDate: "google.protobuf.Timestamp",
  fields: "_field",
  deletedFields: "_field",
  rules: "_v2Rule",
  submitSettings: "_submitSettings"
};
var _freeTrialData = { endDate: "google.protobuf.Timestamp" };
var _function = { description: "_description" };
var _functionsShopPriceSpiConfig2 = {
  shopNumber: "google.protobuf.DoubleValue"
};
var _galleryData2 = { items: "_item", options: "_galleryOptions" };
var _galleryOptions2 = { item: "_itemStyle" };
var _image2 = { urlExpirationDate: "google.protobuf.Timestamp" };
var _imageData2 = { image: "_media" };
var _inlineElement2 = {
  style: "Map#_styleItem",
  data: "Map#_dataItem",
  elements: "Map#_elementItem",
  presets: "Map#_presetItem"
};
var _inputField2 = {
  stringOptions: "__String",
  numberOptions: "__Number",
  booleanOptions: "__Boolean",
  arrayOptions: "__Array",
  objectOptions: "__Object",
  wixFileOptions: "_wixFile",
  paymentOptions: "_payment",
  multilineAddressOptions: "_inputFieldMultilineAddress"
};
var _inputFieldMultilineAddress = {
  multilineAddressOptions: "_multilineAddress"
};
var _item2 = { image: "_itemImage", video: "_video" };
var _itemImage2 = { media: "_media" };
var _itemStyle2 = { ratio: "google.protobuf.DoubleValue" };
var _managedApp = {
  premiumInfo: "_premiumInfo",
  components: "_component",
  pricingDetails: "_pricingDetails"
};
var _managedAppsRequest = {};
var _managedAppsResponse = { managedApps: "_managedApp" };
var _mapData2 = { mapSettings: "_mapSettings" };
var _mapSettings2 = {
  lat: "google.protobuf.DoubleValue",
  lng: "google.protobuf.DoubleValue"
};
var _media2 = { duration: "google.protobuf.DoubleValue" };
var _mediaItem2 = { image: "_upstreamWixCommonImage" };
var _messageContainingTranslatables2 = {
  innerTranslatableRichContent: "_richContent"
};
var _multilineAddress2 = { description: "_richContent" };
var _multilingualTranslationSchema2 = { referenceData: "_schema" };
var _node2 = {
  galleryData: "_galleryData",
  imageData: "_imageData",
  mapData: "_mapData",
  pollData: "_pollData",
  appEmbedData: "_appEmbedData",
  videoData: "_videoData",
  audioData: "_audioData",
  nodes: "_node"
};
var _numberInput2 = {
  default: "google.protobuf.DoubleValue",
  description: "_richContent"
};
var _numberType2 = {
  maximum: "google.protobuf.DoubleValue",
  minimum: "google.protobuf.DoubleValue",
  multipleOf: "google.protobuf.DoubleValue",
  enum: "google.protobuf.DoubleValue"
};
var _objectType2 = { properties: "Map#_propertiesType" };
var _option = { media: "_mediaItem" };
var _orCondition2 = { conditions: "_conditionNode" };
var _packageDimension2 = {
  width: "DOUBLE",
  length: "DOUBLE",
  height: "google.protobuf.DoubleValue"
};
var _packageType2 = { dimension: "_packageDimension", image: "_image" };
var _payment2 = {
  checkboxGroupOptions: "_productCheckboxGroup",
  donationInputOptions: "_donationInput"
};
var _phoneInput2 = { description: "_richContent" };
var _pluginConfig2 = { startingEnforcementDate: "google.protobuf.Timestamp" };
var _poll2 = { image: "_media", options: "_pollOption" };
var _pollData2 = { poll: "_poll", design: "_design" };
var _pollDesign2 = { background: "_background" };
var _pollOption2 = { image: "_media" };
var _predefinedExpectedInput2 = {
  configuration: "_predefinedExpectedInputConfiguration"
};
var _predefinedExpectedInputConfiguration2 = {
  textInput: "_adminConfigurableTextInput"
};
var _premiumInfo = { freeTrialData: "_freeTrialData" };
var _presetEditorPresence2 = { wixMediaThumbnail: "_commonImage" };
var _presetItem2 = { elementsStyleDefaults: "Map#_elementStyleDefaults" };
var _pricingDetails = { discount: "_discount" };
var _productCheckboxGroup2 = {
  description: "_richContent",
  options: "_productCheckboxGroupOption"
};
var _productCheckboxGroupOption2 = { media: "_mediaItem" };
var _propertiesType2 = {
  numberOptions: "_numberType",
  arrayOptions: "_arrayType"
};
var _radioGroup2 = { description: "_richContent" };
var _ratingInput2 = { description: "_richContent" };
var _refElement2 = {
  resolvedElement: "_inlineElement",
  elementsDefaults: "Map#_refInnerElementDefaults"
};
var _refInnerElementDefaults2 = { elements: "Map#_refInnerElementDefaults" };
var _richContent2 = { nodes: "_node", metadata: "_v1Metadata" };
var _richContentOptions2 = { richContent: "_richContent" };
var _saleInfo = {
  startDate: "google.protobuf.Timestamp",
  endDate: "google.protobuf.Timestamp"
};
var _schema2 = {
  createdDate: "google.protobuf.Timestamp",
  updatedDate: "google.protobuf.Timestamp",
  fields: "Map#_v1SchemaField"
};
var _schemaGroup2 = {
  createdDate: "google.protobuf.Timestamp",
  updatedDate: "google.protobuf.Timestamp"
};
var _shippingLabelCarrierSpiConfig2 = { packageTypes: "_packageType" };
var _shippingProviderConfig2 = { shippingPrice: "DOUBLE" };
var _signature2 = { description: "_richContent" };
var _studioWidgetComponentData2 = {
  widgetDisplay: "_widgetDisplay",
  editorPresence: "_editorPresence"
};
var _styleItem2 = { number: "_cssNumber" };
var _submitSettings2 = { thankYouMessageOptions: "_thankYouMessageOptions" };
var _textInput2 = { description: "_richContent" };
var _thankYouMessageOptions2 = { richContent: "_richContent" };
var _timeInput2 = { description: "_richContent" };
var _upstreamWixCommonImage = {
  urlExpirationDate: "google.protobuf.Timestamp"
};
var _v1Metadata2 = {
  createdTimestamp: "google.protobuf.Timestamp",
  updatedTimestamp: "google.protobuf.Timestamp"
};
var _v1SchemaField2 = { index: "google.protobuf.DoubleValue" };
var _v2Rule2 = { expression: "_conditionNode" };
var _video2 = { media: "_media", thumbnail: "_media" };
var _videoData2 = { video: "_media", thumbnail: "_media" };
var _widgetComponentData2 = { widgetDisplay: "_widgetDisplay" };
var _widgetDisplay2 = { price: "google.protobuf.FloatValue" };
var _widgetOutOfIframeComponentData2 = { widgetData: "_widgetComponentData" };
var _wixFile2 = {
  fileUploadOptions: "_fileUpload",
  signatureOptions: "_signature"
};
function resolveComWixpressDevcenterManagedappsManagedAppsUrl(opts) {
  var domainToMappings = {
    "bo._base_domain_": [
      {
        srcPath: "/managed-apps",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/managed-apps",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/managed-apps",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/managed-apps",
        destPath: ""
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/managed-apps",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/managed-apps",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/managed-apps",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/managed-apps",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/managed-apps",
        destPath: ""
      }
    ],
    "wixapis.com": [
      {
        srcPath: "/managed-apps",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
function managedApps(payload) {
  var _a = serializer(_managedAppsRequest, {}), toReq = _a.toJSON, fromReq = _a.fromJSON;
  var fromRes = serializer(_managedAppsResponse, {
    __Array: __Array2,
    __Boolean: __Boolean2,
    __Number: __Number2,
    __Object: __Object2,
    __String: __String2,
    _actionProviderSPIConfig: _actionProviderSPIConfig2,
    _actionSPIConfig: _actionSPIConfig2,
    _adminConfigurableTextInput: _adminConfigurableTextInput2,
    _andCondition: _andCondition2,
    _apiArrayItems,
    _api_Number,
    _appEmbedData: _appEmbedData2,
    _arrayItems: _arrayItems2,
    _arrayType: _arrayType2,
    _audioData: _audioData2,
    _authenticatorConfig: _authenticatorConfig2,
    _background: _background2,
    _checkbox: _checkbox2,
    _checkboxGroup: _checkboxGroup2,
    _commonImage: _commonImage2,
    _component: _component2,
    _componentData: _componentData2,
    _conditionNode: _conditionNode2,
    _cssNumber: _cssNumber2,
    _dataItem: _dataItem2,
    _dataItems: _dataItems2,
    _dateInput: _dateInput2,
    _datePicker: _datePicker2,
    _dateTimeInput: _dateTimeInput2,
    _description: _description2,
    _design: _design2,
    _devCenterTestingComponentData: _devCenterTestingComponentData2,
    _discount,
    _displayField: _displayField2,
    _donationInput: _donationInput2,
    _dropdown: _dropdown2,
    _editorElement: _editorElement2,
    _editorPresence: _editorPresence2,
    _editorReactComponent: _editorReactComponent2,
    _elementItem: _elementItem2,
    _elementStyleDefaults: _elementStyleDefaults2,
    _expectedInputs: _expectedInputs2,
    _field: _field2,
    _fileUpload: _fileUpload2,
    _focalPoint: _focalPoint2,
    _formTemplate: _formTemplate2,
    _freeTrialData,
    _function,
    _functionsShopPriceSpiConfig: _functionsShopPriceSpiConfig2,
    _galleryData: _galleryData2,
    _galleryOptions: _galleryOptions2,
    _image: _image2,
    _imageData: _imageData2,
    _inlineElement: _inlineElement2,
    _inputField: _inputField2,
    _inputFieldMultilineAddress,
    _item: _item2,
    _itemImage: _itemImage2,
    _itemStyle: _itemStyle2,
    _managedApp,
    _mapData: _mapData2,
    _mapSettings: _mapSettings2,
    _media: _media2,
    _mediaItem: _mediaItem2,
    _messageContainingTranslatables: _messageContainingTranslatables2,
    _multilineAddress: _multilineAddress2,
    _multilingualTranslationSchema: _multilingualTranslationSchema2,
    _node: _node2,
    _numberInput: _numberInput2,
    _numberType: _numberType2,
    _objectType: _objectType2,
    _option,
    _orCondition: _orCondition2,
    _packageDimension: _packageDimension2,
    _packageType: _packageType2,
    _payment: _payment2,
    _phoneInput: _phoneInput2,
    _pluginConfig: _pluginConfig2,
    _poll: _poll2,
    _pollData: _pollData2,
    _pollDesign: _pollDesign2,
    _pollOption: _pollOption2,
    _predefinedExpectedInput: _predefinedExpectedInput2,
    _predefinedExpectedInputConfiguration: _predefinedExpectedInputConfiguration2,
    _premiumInfo,
    _presetEditorPresence: _presetEditorPresence2,
    _presetItem: _presetItem2,
    _pricingDetails,
    _productCheckboxGroup: _productCheckboxGroup2,
    _productCheckboxGroupOption: _productCheckboxGroupOption2,
    _propertiesType: _propertiesType2,
    _radioGroup: _radioGroup2,
    _ratingInput: _ratingInput2,
    _refElement: _refElement2,
    _refInnerElementDefaults: _refInnerElementDefaults2,
    _richContent: _richContent2,
    _richContentOptions: _richContentOptions2,
    _saleInfo,
    _schema: _schema2,
    _schemaGroup: _schemaGroup2,
    _shippingLabelCarrierSpiConfig: _shippingLabelCarrierSpiConfig2,
    _shippingProviderConfig: _shippingProviderConfig2,
    _signature: _signature2,
    _studioWidgetComponentData: _studioWidgetComponentData2,
    _styleItem: _styleItem2,
    _submitSettings: _submitSettings2,
    _textInput: _textInput2,
    _thankYouMessageOptions: _thankYouMessageOptions2,
    _timeInput: _timeInput2,
    _upstreamWixCommonImage,
    _v1Metadata: _v1Metadata2,
    _v1SchemaField: _v1SchemaField2,
    _v2Rule: _v2Rule2,
    _video: _video2,
    _videoData: _videoData2,
    _widgetComponentData: _widgetComponentData2,
    _widgetDisplay: _widgetDisplay2,
    _widgetOutOfIframeComponentData: _widgetOutOfIframeComponentData2,
    _wixFile: _wixFile2
  }).fromJSON;
  function __managedApps(_a2) {
    var host = _a2.host;
    var serializedData = toReq(payload);
    var metadata = {
      entityFqdn: "wix.devcenter.managedapps.v1.managed_app",
      method: "GET",
      methodFqn: "com.wixpress.devcenter.managedapps.ManagedApps.ManagedApps",
      url: resolveComWixpressDevcenterManagedappsManagedAppsUrl({
        protoPath: "/v1/managed-apps",
        data: serializedData,
        host
      }),
      params: toURLSearchParams(serializedData),
      transformResponse: fromRes
    };
    return metadata;
  }
  __managedApps.fromReq = fromReq;
  __managedApps.__isAmbassador = true;
  return __managedApps;
}

// ../../node_modules/@wix/ambassador-devcenter-managedapps-v1-managed-app/build/es/types.impl.js
init_esm_shims();
var UninstallActions;
(function(UninstallActions2) {
  UninstallActions2["NONE_ACTION"] = "NONE_ACTION";
  UninstallActions2["NO_ACTION"] = "NO_ACTION";
  UninstallActions2["REMOVE_ALL_PRODUCT"] = "REMOVE_ALL_PRODUCT";
  UninstallActions2["SHIPPING_RATES_REGION"] = "SHIPPING_RATES_REGION";
})(UninstallActions || (UninstallActions = {}));
var OpenConsentIn;
(function(OpenConsentIn2) {
  OpenConsentIn2["NONE_VALUE"] = "NONE_VALUE";
  OpenConsentIn2["NEW_WINDOW"] = "NEW_WINDOW";
  OpenConsentIn2["NEW_TAB"] = "NEW_TAB";
})(OpenConsentIn || (OpenConsentIn = {}));
var RenewalType;
(function(RenewalType2) {
  RenewalType2["UNKNOWN"] = "UNKNOWN";
  RenewalType2["AUTO_RENEW_ON"] = "AUTO_RENEW_ON";
  RenewalType2["AUTO_RENEW_OFF"] = "AUTO_RENEW_OFF";
  RenewalType2["MANUAL"] = "MANUAL";
})(RenewalType || (RenewalType = {}));
var PaymentCollectionType;
(function(PaymentCollectionType2) {
  PaymentCollectionType2["PMT_UNKNOWN"] = "PMT_UNKNOWN";
  PaymentCollectionType2["SUBSCRIPTION"] = "SUBSCRIPTION";
  PaymentCollectionType2["NON_PAYABLE"] = "NON_PAYABLE";
  PaymentCollectionType2["ONE_TIME"] = "ONE_TIME";
})(PaymentCollectionType || (PaymentCollectionType = {}));
var IncompleteSetupReason;
(function(IncompleteSetupReason2) {
  IncompleteSetupReason2["NO_REASON"] = "NO_REASON";
  IncompleteSetupReason2["PENDING_CONSENT"] = "PENDING_CONSENT";
  IncompleteSetupReason2["PENDING_CONFIGURATION"] = "PENDING_CONFIGURATION";
  IncompleteSetupReason2["PENDING_REQUIRED_INSTALLATIONS"] = "PENDING_REQUIRED_INSTALLATIONS";
  IncompleteSetupReason2["PENDING_EDITOR"] = "PENDING_EDITOR";
})(IncompleteSetupReason || (IncompleteSetupReason = {}));
var AppGroup;
(function(AppGroup2) {
  AppGroup2["EMPTY"] = "EMPTY";
  AppGroup2["INSTALLED"] = "INSTALLED";
  AppGroup2["PENDING"] = "PENDING";
  AppGroup2["BUSINESS_SOLUTIONS"] = "BUSINESS_SOLUTIONS";
})(AppGroup || (AppGroup = {}));
var ComponentType2;
(function(ComponentType3) {
  ComponentType3["NONE"] = "NONE";
  ComponentType3["STUDIO"] = "STUDIO";
  ComponentType3["PLATFORM"] = "PLATFORM";
  ComponentType3["WORKER"] = "WORKER";
  ComponentType3["DASHBOARD"] = "DASHBOARD";
  ComponentType3["WIDGET"] = "WIDGET";
  ComponentType3["PAGE"] = "PAGE";
  ComponentType3["DASHBOARD_PLATFORM"] = "DASHBOARD_PLATFORM";
  ComponentType3["STUDIO_WIDGET"] = "STUDIO_WIDGET";
  ComponentType3["EMBEDDED_SCRIPT"] = "EMBEDDED_SCRIPT";
  ComponentType3["EXTENSION"] = "EXTENSION";
  ComponentType3["SNIPPET_SOLUTION"] = "SNIPPET_SOLUTION";
  ComponentType3["DATA_COMPONENT"] = "DATA_COMPONENT";
  ComponentType3["WEB"] = "WEB";
  ComponentType3["DC_CONFIG"] = "DC_CONFIG";
  ComponentType3["WIDGET_OUT_OF_IFRAME"] = "WIDGET_OUT_OF_IFRAME";
  ComponentType3["PAGE_OUT_OF_IFRAME"] = "PAGE_OUT_OF_IFRAME";
  ComponentType3["STATIC_FILE"] = "STATIC_FILE";
  ComponentType3["APP_CONFIG"] = "APP_CONFIG";
  ComponentType3["MULTIPLE_DASHBOARDS"] = "MULTIPLE_DASHBOARDS";
  ComponentType3["PAYMENTS_GATEWAY"] = "PAYMENTS_GATEWAY";
  ComponentType3["CODE_PACKAGE"] = "CODE_PACKAGE";
  ComponentType3["AUTOMATION_TRIGGER"] = "AUTOMATION_TRIGGER";
  ComponentType3["INVOICES_ACTIONS"] = "INVOICES_ACTIONS";
  ComponentType3["DASHBOARD_APPLICATION"] = "DASHBOARD_APPLICATION";
  ComponentType3["CONTACT_LABELS"] = "CONTACT_LABELS";
  ComponentType3["WIDGET_PLUGIN"] = "WIDGET_PLUGIN";
  ComponentType3["CROSS_SELL"] = "CROSS_SELL";
  ComponentType3["LOCAL_DELIVERY"] = "LOCAL_DELIVERY";
  ComponentType3["PAYMENT_PROVIDER"] = "PAYMENT_PROVIDER";
  ComponentType3["ECOM_MEMBERSHIPS"] = "ECOM_MEMBERSHIPS";
  ComponentType3["ECOM_LINE_ITEMS_ENRICHER"] = "ECOM_LINE_ITEMS_ENRICHER";
  ComponentType3["ECOM_SHIPPING_RATES"] = "ECOM_SHIPPING_RATES";
  ComponentType3["SHIPPING_LABEL_CARRIER"] = "SHIPPING_LABEL_CARRIER";
  ComponentType3["RESTAURANTS_POS"] = "RESTAURANTS_POS";
  ComponentType3["FICTIONAL_SHIPPING_PROVIDER"] = "FICTIONAL_SHIPPING_PROVIDER";
  ComponentType3["ALERT_ENRICHER"] = "ALERT_ENRICHER";
  ComponentType3["DATA_EXTENSIONS"] = "DATA_EXTENSIONS";
  ComponentType3["GENERIC_HOOKS"] = "GENERIC_HOOKS";
  ComponentType3["AUTOMATIONS_ACTION_PROVIDER"] = "AUTOMATIONS_ACTION_PROVIDER";
  ComponentType3["ECOM_CATALOG"] = "ECOM_CATALOG";
  ComponentType3["BACK_OFFICE_EXTENSION_CONTAINER"] = "BACK_OFFICE_EXTENSION_CONTAINER";
  ComponentType3["BACK_OFFICE_EXTENSION"] = "BACK_OFFICE_EXTENSION";
  ComponentType3["AUTOMATIONS_TRIGGER_PROVIDER"] = "AUTOMATIONS_TRIGGER_PROVIDER";
  ComponentType3["IDENTITY_PRE_REGISTRATION"] = "IDENTITY_PRE_REGISTRATION";
  ComponentType3["PREMIUM_PRODUCTS_PATHS"] = "PREMIUM_PRODUCTS_PATHS";
  ComponentType3["ECOM_CUSTOM_SCOPE"] = "ECOM_CUSTOM_SCOPE";
  ComponentType3["GIFT_CARDS_PROVIDER"] = "GIFT_CARDS_PROVIDER";
  ComponentType3["DEALER_EXTERNAL_FILTER_PROVIDER"] = "DEALER_EXTERNAL_FILTER_PROVIDER";
  ComponentType3["ECOM_DROPSHIPPING_PROVIDER"] = "ECOM_DROPSHIPPING_PROVIDER";
  ComponentType3["INVOICES_PROVIDER"] = "INVOICES_PROVIDER";
  ComponentType3["SEO_KEYWORDS_SUGGESTIONS"] = "SEO_KEYWORDS_SUGGESTIONS";
  ComponentType3["ECOM_DISCOUNTS_TRIGGER"] = "ECOM_DISCOUNTS_TRIGGER";
  ComponentType3["MULTILINGUAL_CONTENT_PROVIDER"] = "MULTILINGUAL_CONTENT_PROVIDER";
  ComponentType3["APPLICATION_AUTOMATION"] = "APPLICATION_AUTOMATION";
  ComponentType3["BACK_OFFICE_SIDEBAR_CATEGORY"] = "BACK_OFFICE_SIDEBAR_CATEGORY";
  ComponentType3["BACK_OFFICE_PAGE"] = "BACK_OFFICE_PAGE";
  ComponentType3["ECOM_ADDITIONAL_FEES"] = "ECOM_ADDITIONAL_FEES";
  ComponentType3["PING_USER_NOTIFICATION"] = "PING_USER_NOTIFICATION";
  ComponentType3["ECOM_RECOMMENDATIONS_PROVIDER"] = "ECOM_RECOMMENDATIONS_PROVIDER";
  ComponentType3["BOOKINGS_PRICING_PROVIDER"] = "BOOKINGS_PRICING_PROVIDER";
  ComponentType3["IDENTITY_AUTHENTICATOR"] = "IDENTITY_AUTHENTICATOR";
  ComponentType3["IDENTITY_IDP_CONNECTOR"] = "IDENTITY_IDP_CONNECTOR";
  ComponentType3["ITEMS_SELECTION_PROVIDER"] = "ITEMS_SELECTION_PROVIDER";
  ComponentType3["PORTFOLIO_SYNCED_PROJECTS_PROVIDER"] = "PORTFOLIO_SYNCED_PROJECTS_PROVIDER";
  ComponentType3["COMMUNICATION_CHANNEL"] = "COMMUNICATION_CHANNEL";
  ComponentType3["IDENTITY_POST_LOGIN"] = "IDENTITY_POST_LOGIN";
  ComponentType3["BACK_OFFICE_WIDGET"] = "BACK_OFFICE_WIDGET";
  ComponentType3["SOCIAL_MARKETING_DESIGN"] = "SOCIAL_MARKETING_DESIGN";
  ComponentType3["FORMS_SUBMISSIONS_PROVIDER"] = "FORMS_SUBMISSIONS_PROVIDER";
  ComponentType3["WIX_OFFERING"] = "WIX_OFFERING";
  ComponentType3["DEV_CENTER_TESTING_COMPONENT"] = "DEV_CENTER_TESTING_COMPONENT";
  ComponentType3["COMPONENTS_VALIDATOR_PROVIDER"] = "COMPONENTS_VALIDATOR_PROVIDER";
  ComponentType3["COMPONENTS_TRANSLATIONS_ADDITIONAL_FIELDS_PROVIDER"] = "COMPONENTS_TRANSLATIONS_ADDITIONAL_FIELDS_PROVIDER";
  ComponentType3["FORMS_SCHEMA_PROVIDER"] = "FORMS_SCHEMA_PROVIDER";
  ComponentType3["BOOKINGS_EXTERNAL_CALENDAR_PROVIDER"] = "BOOKINGS_EXTERNAL_CALENDAR_PROVIDER";
  ComponentType3["ECOM_DEFAULT_TAXATION_CATEGORY"] = "ECOM_DEFAULT_TAXATION_CATEGORY";
  ComponentType3["VIEWER_DYNAMIC_SITE_STRUCTURE_PROVIDER"] = "VIEWER_DYNAMIC_SITE_STRUCTURE_PROVIDER";
  ComponentType3["PING_UOU_NOTIFICATION"] = "PING_UOU_NOTIFICATION";
  ComponentType3["HEADLESS_OAUTH"] = "HEADLESS_OAUTH";
  ComponentType3["ECOM_TAX_CALCULATOR_SPI"] = "ECOM_TAX_CALCULATOR_SPI";
  ComponentType3["COMMENTS_MODERATION_PROVIDER"] = "COMMENTS_MODERATION_PROVIDER";
  ComponentType3["GRID_APP_FILES_TRANSFORMER"] = "GRID_APP_FILES_TRANSFORMER";
  ComponentType3["BENEFIT_PROGRAMS_POLICY_PROVIDER"] = "BENEFIT_PROGRAMS_POLICY_PROVIDER";
  ComponentType3["PREMIUM_CUSTOM_CHARGES"] = "PREMIUM_CUSTOM_CHARGES";
  ComponentType3["ECOM_VALIDATIONS"] = "ECOM_VALIDATIONS";
  ComponentType3["COMPONENT_REFERENCE_DATA_PROVIDER"] = "COMPONENT_REFERENCE_DATA_PROVIDER";
  ComponentType3["WIX_REVIEWS_PRODUCT_CATALOG"] = "WIX_REVIEWS_PRODUCT_CATALOG";
  ComponentType3["SOCIAL_MARKETING_DESIGNS_PROVIDER"] = "SOCIAL_MARKETING_DESIGNS_PROVIDER";
  ComponentType3["GOOGLE_BUSINESS_PROFILE_FEATURE_PROVIDER"] = "GOOGLE_BUSINESS_PROFILE_FEATURE_PROVIDER";
  ComponentType3["COMMENTS_FILTER_PROVIDER"] = "COMMENTS_FILTER_PROVIDER";
  ComponentType3["BILLING_TAX_ID_VALIDATOR"] = "BILLING_TAX_ID_VALIDATOR";
  ComponentType3["PING_SETTINGS_GROUP"] = "PING_SETTINGS_GROUP";
  ComponentType3["FORMS_SPAM_SUBMISSIONS_PROVIDER"] = "FORMS_SPAM_SUBMISSIONS_PROVIDER";
  ComponentType3["EDITOR_ADDON"] = "EDITOR_ADDON";
  ComponentType3["EXTERNAL_DATABASE_PROVIDER"] = "EXTERNAL_DATABASE_PROVIDER";
  ComponentType3["ECOM_PAYMENT_SETTINGS"] = "ECOM_PAYMENT_SETTINGS";
  ComponentType3["NOTIFICATION_TOPIC"] = "NOTIFICATION_TOPIC";
  ComponentType3["NOTIFICATION_PREFERENCES_FILTER_PROVIDER"] = "NOTIFICATION_PREFERENCES_FILTER_PROVIDER";
  ComponentType3["BOOKINGS_RESOURCE_TYPES_PROVIDER"] = "BOOKINGS_RESOURCE_TYPES_PROVIDER";
  ComponentType3["PRICING_PLANS_FORM_CONFIGURATION"] = "PRICING_PLANS_FORM_CONFIGURATION";
  ComponentType3["USER_NOTIFICATION"] = "USER_NOTIFICATION";
  ComponentType3["CONTACT_NOTIFICATION"] = "CONTACT_NOTIFICATION";
  ComponentType3["UNIFIED_PAGE"] = "UNIFIED_PAGE";
  ComponentType3["AVAILABILITY_TIME_SLOTS_CONFIGURATION_PROVIDER"] = "AVAILABILITY_TIME_SLOTS_CONFIGURATION_PROVIDER";
  ComponentType3["PROPOSAL_EDITOR_PROVIDER"] = "PROPOSAL_EDITOR_PROVIDER";
  ComponentType3["CUSTOM_TABLE_RESERVATIONS_PROVIDER"] = "CUSTOM_TABLE_RESERVATIONS_PROVIDER";
  ComponentType3["COMMENTS_CONTEXT_PROVIDER"] = "COMMENTS_CONTEXT_PROVIDER";
  ComponentType3["FORMS_SPAM_SUBMISSION_REPORTS_PROVIDER"] = "FORMS_SPAM_SUBMISSION_REPORTS_PROVIDER";
  ComponentType3["AUTOMATIONS_VELO_ACTION_PROVIDER"] = "AUTOMATIONS_VELO_ACTION_PROVIDER";
  ComponentType3["CALENDAR_EVENT_TYPE_PROVIDER"] = "CALENDAR_EVENT_TYPE_PROVIDER";
  ComponentType3["RESERVED"] = "RESERVED";
  ComponentType3["SMS_ACTION_MESSAGE"] = "SMS_ACTION_MESSAGE";
  ComponentType3["BOOKING_POLICY_PROVIDER"] = "BOOKING_POLICY_PROVIDER";
  ComponentType3["MULTI_SERVICE_BOOKING_POLICY_PROVIDER"] = "MULTI_SERVICE_BOOKING_POLICY_PROVIDER";
  ComponentType3["AI_ASSISTANT"] = "AI_ASSISTANT";
  ComponentType3["FORMS_SUBMISSIONS_EXTENSION_PROVIDER"] = "FORMS_SUBMISSIONS_EXTENSION_PROVIDER";
  ComponentType3["MULTILINGUAL_TRANSLATION_SCHEMA"] = "MULTILINGUAL_TRANSLATION_SCHEMA";
  ComponentType3["TAX_CALCULATION_PROVIDER"] = "TAX_CALCULATION_PROVIDER";
  ComponentType3["TAX_GROUPS_PROVIDER"] = "TAX_GROUPS_PROVIDER";
  ComponentType3["BACK_OFFICE_MODAL"] = "BACK_OFFICE_MODAL";
  ComponentType3["DEPLOYMENT_PIPELINE_PROVIDER"] = "DEPLOYMENT_PIPELINE_PROVIDER";
  ComponentType3["CUSTOM_ELEMENT_WIDGET"] = "CUSTOM_ELEMENT_WIDGET";
  ComponentType3["BACK_OFFICE_EXTENSION_WIDGET"] = "BACK_OFFICE_EXTENSION_WIDGET";
  ComponentType3["BACK_OFFICE_EXTENSION_MENU_ITEM"] = "BACK_OFFICE_EXTENSION_MENU_ITEM";
  ComponentType3["FORM_TEMPLATE"] = "FORM_TEMPLATE";
  ComponentType3["NOTIFICATION_CONTENT"] = "NOTIFICATION_CONTENT";
  ComponentType3["BROADCAST_LIST"] = "BROADCAST_LIST";
  ComponentType3["PARTNERS_PAYOUTS"] = "PARTNERS_PAYOUTS";
  ComponentType3["WIX_REVIEWS_ENTITY_CATALOG"] = "WIX_REVIEWS_ENTITY_CATALOG";
  ComponentType3["VELO_PUBLISH_PIPELINE_TASK_PROVIDER"] = "VELO_PUBLISH_PIPELINE_TASK_PROVIDER";
  ComponentType3["FUNCTIONS_SHOP_PRICE_PROVIDER"] = "FUNCTIONS_SHOP_PRICE_PROVIDER";
  ComponentType3["FUNCTION"] = "FUNCTION";
  ComponentType3["ECOM_CHECKOUT_CONTENT"] = "ECOM_CHECKOUT_CONTENT";
  ComponentType3["COMMUNICATION_CHANNEL_PROVIDER"] = "COMMUNICATION_CHANNEL_PROVIDER";
  ComponentType3["WEBHOOK"] = "WEBHOOK";
  ComponentType3["TOPOLOGY"] = "TOPOLOGY";
  ComponentType3["LOYALTY_CUSTOM_REWARDS"] = "LOYALTY_CUSTOM_REWARDS";
  ComponentType3["FUNCTION_RECIPE"] = "FUNCTION_RECIPE";
  ComponentType3["BACK_OFFICE_EXTERNAL_URL"] = "BACK_OFFICE_EXTERNAL_URL";
  ComponentType3["IDENTITY_FACTOR"] = "IDENTITY_FACTOR";
  ComponentType3["ECOM_DISCOUNTS"] = "ECOM_DISCOUNTS";
  ComponentType3["VELO_CUSTOM_CSS"] = "VELO_CUSTOM_CSS";
  ComponentType3["DEALER_ADAPTIVE_COMPONENT_PROVIDER"] = "DEALER_ADAPTIVE_COMPONENT_PROVIDER";
  ComponentType3["AI_ASSISTANT_ACTION"] = "AI_ASSISTANT_ACTION";
  ComponentType3["ECOM_INVENTORY"] = "ECOM_INVENTORY";
  ComponentType3["MONITORING"] = "MONITORING";
  ComponentType3["PAPI_PROVIDER"] = "PAPI_PROVIDER";
  ComponentType3["UNIFIED_LIGHTBOX"] = "UNIFIED_LIGHTBOX";
  ComponentType3["FORMS_EXTENSION_PROVIDER"] = "FORMS_EXTENSION_PROVIDER";
  ComponentType3["MULTILINGUAL_TRANSLATION_SCHEMA_GROUP"] = "MULTILINGUAL_TRANSLATION_SCHEMA_GROUP";
  ComponentType3["PANEL"] = "PANEL";
  ComponentType3["PREMIUM_PURCHASE_VALIDATIONS"] = "PREMIUM_PURCHASE_VALIDATIONS";
  ComponentType3["BROWSER_STORAGE"] = "BROWSER_STORAGE";
  ComponentType3["SDK_DEFINITION"] = "SDK_DEFINITION";
  ComponentType3["SITE_WIDGET_SLOT"] = "SITE_WIDGET_SLOT";
  ComponentType3["CALENDAR_EVENT_VALIDATION_PROVIDER"] = "CALENDAR_EVENT_VALIDATION_PROVIDER";
  ComponentType3["CLIENT_SIDE_SERVICE"] = "CLIENT_SIDE_SERVICE";
  ComponentType3["BILLING_SETTINGS"] = "BILLING_SETTINGS";
  ComponentType3["PATTERNS_WIZARD"] = "PATTERNS_WIZARD";
  ComponentType3["APPLICATION_PROFILE"] = "APPLICATION_PROFILE";
  ComponentType3["TEXT_TO_SPEECH_ACTION_MESSAGE"] = "TEXT_TO_SPEECH_ACTION_MESSAGE";
  ComponentType3["AUDIENCE_PROVIDER"] = "AUDIENCE_PROVIDER";
  ComponentType3["PRICING_PLANS_PRICE"] = "PRICING_PLANS_PRICE";
  ComponentType3["PRICING_PLAN_START_DATE_LIMITS"] = "PRICING_PLAN_START_DATE_LIMITS";
  ComponentType3["PRICING_PLAN_START_DATE_RULES"] = "PRICING_PLAN_START_DATE_RULES";
  ComponentType3["EVENTS_TICKET_RESERVATIONS"] = "EVENTS_TICKET_RESERVATIONS";
  ComponentType3["PAYMENTS_DISPUTE_SERVICE_PLUGIN"] = "PAYMENTS_DISPUTE_SERVICE_PLUGIN";
  ComponentType3["PRICING_PLANS_FEES"] = "PRICING_PLANS_FEES";
  ComponentType3["EDITOR_REACT_COMPONENT"] = "EDITOR_REACT_COMPONENT";
  ComponentType3["SUPPLIERS_HUB_MARKETPLACE"] = "SUPPLIERS_HUB_MARKETPLACE";
  ComponentType3["FORM_SCHEMA_DYNAMIC_VALUES"] = "FORM_SCHEMA_DYNAMIC_VALUES";
  ComponentType3["BLOG_PAYWALL_PROVIDER"] = "BLOG_PAYWALL_PROVIDER";
  ComponentType3["LOYALTY_CUSTOM_REWARDS_V2"] = "LOYALTY_CUSTOM_REWARDS_V2";
  ComponentType3["STORES_PRODUCT_RESTRICTIONS"] = "STORES_PRODUCT_RESTRICTIONS";
  ComponentType3["FORM_SUBMISSION_MODERATION"] = "FORM_SUBMISSION_MODERATION";
  ComponentType3["EVENTS_EVENT_BADGES"] = "EVENTS_EVENT_BADGES";
  ComponentType3["BILLING_OPERATION"] = "BILLING_OPERATION";
  ComponentType3["BACK_OFFICE_CUSTOMIZATION"] = "BACK_OFFICE_CUSTOMIZATION";
  ComponentType3["COMPONENT_ENRICHER_PROVIDER"] = "COMPONENT_ENRICHER_PROVIDER";
  ComponentType3["BACK_OFFICE_RESTRICTED_CUSTOMIZATION"] = "BACK_OFFICE_RESTRICTED_CUSTOMIZATION";
  ComponentType3["EDITOR_APP_PREVIEWS_POC"] = "EDITOR_APP_PREVIEWS_POC";
  ComponentType3["LEGENDS_PERSONA_CONFIGURATION"] = "LEGENDS_PERSONA_CONFIGURATION";
  ComponentType3["WIX_HOSTING_APP_DEPLOYMENT_PROVIDER"] = "WIX_HOSTING_APP_DEPLOYMENT_PROVIDER";
  ComponentType3["BACKEND_WORKER"] = "BACKEND_WORKER";
  ComponentType3["EVENT_TIME_SLOTS_CONFIGURATION_PROVIDER"] = "EVENT_TIME_SLOTS_CONFIGURATION_PROVIDER";
  ComponentType3["WIX_HOSTING_APP_ENVIRONMENT_PROVIDER"] = "WIX_HOSTING_APP_ENVIRONMENT_PROVIDER";
  ComponentType3["SDK_EXPORTS"] = "SDK_EXPORTS";
  ComponentType3["VIEWER_SERVICE"] = "VIEWER_SERVICE";
  ComponentType3["WIX_PAYMENTS_PROVIDER_ACCOUNT_SERVICE_PLUGIN"] = "WIX_PAYMENTS_PROVIDER_ACCOUNT_SERVICE_PLUGIN";
  ComponentType3["CONTACTS_SEGMENTS_V2_FILTER_PROVIDER"] = "CONTACTS_SEGMENTS_V2_FILTER_PROVIDER";
})(ComponentType2 || (ComponentType2 = {}));
var WidgetVertical;
(function(WidgetVertical2) {
  WidgetVertical2["NONE_VERTICAL"] = "NONE_VERTICAL";
  WidgetVertical2["TOP"] = "TOP";
  WidgetVertical2["CENTER_VERTICAL"] = "CENTER_VERTICAL";
  WidgetVertical2["BOTTOM"] = "BOTTOM";
})(WidgetVertical || (WidgetVertical = {}));
var WidgetHorizontal;
(function(WidgetHorizontal2) {
  WidgetHorizontal2["NONE_HORIZONTAL"] = "NONE_HORIZONTAL";
  WidgetHorizontal2["LEFT"] = "LEFT";
  WidgetHorizontal2["CENTER_HORIZONTAL"] = "CENTER_HORIZONTAL";
  WidgetHorizontal2["RIGHT"] = "RIGHT";
})(WidgetHorizontal || (WidgetHorizontal = {}));
var WidgetWidthType;
(function(WidgetWidthType2) {
  WidgetWidthType2["NONE_TYPE"] = "NONE_TYPE";
  WidgetWidthType2["CUSTOM"] = "CUSTOM";
  WidgetWidthType2["FULL"] = "FULL";
})(WidgetWidthType || (WidgetWidthType = {}));
var Region;
(function(Region2) {
  Region2["no_region"] = "no_region";
  Region2["header"] = "header";
  Region2["pageContainer"] = "pageContainer";
  Region2["footer"] = "footer";
})(Region || (Region = {}));
var UnitType;
(function(UnitType2) {
  UnitType2["NO_UNIT"] = "NO_UNIT";
  UnitType2["AUTO"] = "AUTO";
  UnitType2["PX"] = "PX";
  UnitType2["VH"] = "VH";
  UnitType2["VW"] = "VW";
  UnitType2["PERCENTAGE"] = "PERCENTAGE";
})(UnitType || (UnitType = {}));
var HorizontalDocking;
(function(HorizontalDocking2) {
  HorizontalDocking2["NO_HDOCKING"] = "NO_HDOCKING";
  HorizontalDocking2["LEFT_DOCKING"] = "LEFT_DOCKING";
  HorizontalDocking2["HCENTER"] = "HCENTER";
  HorizontalDocking2["RIGHT_DOCKING"] = "RIGHT_DOCKING";
})(HorizontalDocking || (HorizontalDocking = {}));
var VerticalDocking;
(function(VerticalDocking2) {
  VerticalDocking2["NO_VDOCKING"] = "NO_VDOCKING";
  VerticalDocking2["TOP_DOCKING"] = "TOP_DOCKING";
  VerticalDocking2["VCENTER"] = "VCENTER";
  VerticalDocking2["BOTTOM_DOCKING"] = "BOTTOM_DOCKING";
})(VerticalDocking || (VerticalDocking = {}));
var ReplacementType;
(function(ReplacementType2) {
  ReplacementType2["UNKNOWN_REPLACEMENT"] = "UNKNOWN_REPLACEMENT";
  ReplacementType2["REPLACING"] = "REPLACING";
  ReplacementType2["REPLACEABLE"] = "REPLACEABLE";
})(ReplacementType || (ReplacementType = {}));
var PluginInterface;
(function(PluginInterface2) {
  PluginInterface2["NONE_INTERFACE"] = "NONE_INTERFACE";
  PluginInterface2["REVIEWS"] = "REVIEWS";
  PluginInterface2["RATINGS_SUMMARY"] = "RATINGS_SUMMARY";
  PluginInterface2["RATINGS_SUMMARY_OOI_LIST"] = "RATINGS_SUMMARY_OOI_LIST";
  PluginInterface2["BOOKINGS_SERVICE"] = "BOOKINGS_SERVICE";
  PluginInterface2["BOOKINGS_FORM"] = "BOOKINGS_FORM";
  PluginInterface2["BASE"] = "BASE";
  PluginInterface2["EVENT"] = "EVENT";
  PluginInterface2["PRODUCT"] = "PRODUCT";
  PluginInterface2["CHECKOUT"] = "CHECKOUT";
  PluginInterface2["CATEGORY"] = "CATEGORY";
  PluginInterface2["BOOKINGS_CALENDAR"] = "BOOKINGS_CALENDAR";
  PluginInterface2["CART"] = "CART";
})(PluginInterface || (PluginInterface = {}));
var ExtensionExposureExposure;
(function(ExtensionExposureExposure2) {
  ExtensionExposureExposure2["PRIVATE"] = "PRIVATE";
  ExtensionExposureExposure2["INTERNAL"] = "INTERNAL";
  ExtensionExposureExposure2["PUBLIC"] = "PUBLIC";
})(ExtensionExposureExposure || (ExtensionExposureExposure = {}));
var Maturity;
(function(Maturity2) {
  Maturity2["ALPHA"] = "ALPHA";
  Maturity2["BETA"] = "BETA";
  Maturity2["GA"] = "GA";
  Maturity2["NOT_IMPLEMENTED"] = "NOT_IMPLEMENTED";
})(Maturity || (Maturity = {}));
var HTTPMethod;
(function(HTTPMethod2) {
  HTTPMethod2["UNKNOWN_METHOD"] = "UNKNOWN_METHOD";
  HTTPMethod2["GET"] = "GET";
  HTTPMethod2["POST"] = "POST";
})(HTTPMethod || (HTTPMethod = {}));
var InstallPage;
(function(InstallPage2) {
  InstallPage2["NO_PAGE"] = "NO_PAGE";
  InstallPage2["CURRENT"] = "CURRENT";
  InstallPage2["PAGE"] = "PAGE";
  InstallPage2["LIGHTBOX"] = "LIGHTBOX";
})(InstallPage || (InstallPage = {}));
var RegionType;
(function(RegionType2) {
  RegionType2["HEADER"] = "HEADER";
  RegionType2["BODY"] = "BODY";
  RegionType2["FOOTER"] = "FOOTER";
})(RegionType || (RegionType = {}));
var StaticContainer;
(function(StaticContainer2) {
  StaticContainer2["UNKNOWN_CONTAINER"] = "UNKNOWN_CONTAINER";
  StaticContainer2["HOMEPAGE"] = "HOMEPAGE";
})(StaticContainer || (StaticContainer = {}));
var EmbeddedScriptPages;
(function(EmbeddedScriptPages2) {
  EmbeddedScriptPages2["NONE_PAGES"] = "NONE_PAGES";
  EmbeddedScriptPages2["ONCE"] = "ONCE";
  EmbeddedScriptPages2["SPECIFIC_PAGES"] = "SPECIFIC_PAGES";
})(EmbeddedScriptPages || (EmbeddedScriptPages = {}));
var EmbeddedScriptPlacement;
(function(EmbeddedScriptPlacement2) {
  EmbeddedScriptPlacement2["NONE_PLACEMENT"] = "NONE_PLACEMENT";
  EmbeddedScriptPlacement2["HEAD"] = "HEAD";
  EmbeddedScriptPlacement2["BODY_START"] = "BODY_START";
  EmbeddedScriptPlacement2["BODY_END"] = "BODY_END";
})(EmbeddedScriptPlacement || (EmbeddedScriptPlacement = {}));
var EmbedCategory;
(function(EmbedCategory2) {
  EmbedCategory2["UNKNOWN"] = "UNKNOWN";
  EmbedCategory2["ESSENTIAL"] = "ESSENTIAL";
  EmbedCategory2["FUNCTIONAL"] = "FUNCTIONAL";
  EmbedCategory2["ANALYTICS"] = "ANALYTICS";
  EmbedCategory2["ADVERTISING"] = "ADVERTISING";
})(EmbedCategory || (EmbedCategory = {}));
var TemplateDefaultColor;
(function(TemplateDefaultColor2) {
  TemplateDefaultColor2["BACKGROUND"] = "BACKGROUND";
  TemplateDefaultColor2["SECONDARY_TEXTS"] = "SECONDARY_TEXTS";
  TemplateDefaultColor2["MAIN_TEXT_AND_ICONS"] = "MAIN_TEXT_AND_ICONS";
  TemplateDefaultColor2["BORDERS_AND_DIVIDERS"] = "BORDERS_AND_DIVIDERS";
  TemplateDefaultColor2["BUTTONS_AND_LINKS"] = "BUTTONS_AND_LINKS";
})(TemplateDefaultColor || (TemplateDefaultColor = {}));
var ThumbnailsSize;
(function(ThumbnailsSize2) {
  ThumbnailsSize2["SMALL"] = "SMALL";
  ThumbnailsSize2["MEDIUM"] = "MEDIUM";
  ThumbnailsSize2["LARGE"] = "LARGE";
  ThumbnailsSize2["XLARGE"] = "XLARGE";
})(ThumbnailsSize || (ThumbnailsSize = {}));
var BarAlignmentSelected;
(function(BarAlignmentSelected2) {
  BarAlignmentSelected2["ALIGN_LEFT"] = "ALIGN_LEFT";
  BarAlignmentSelected2["ALIGN_CENTER"] = "ALIGN_CENTER";
  BarAlignmentSelected2["ALIGN_RIGHT"] = "ALIGN_RIGHT";
})(BarAlignmentSelected || (BarAlignmentSelected = {}));
var DefaultTextStyle;
(function(DefaultTextStyle2) {
  DefaultTextStyle2["TITLE"] = "TITLE";
  DefaultTextStyle2["PARAGRAPH"] = "PARAGRAPH";
  DefaultTextStyle2["LOWER_HIERARCHY_TEXTS"] = "LOWER_HIERARCHY_TEXTS";
})(DefaultTextStyle || (DefaultTextStyle = {}));
var ElementType;
(function(ElementType2) {
  ElementType2["WIDGET"] = "WIDGET";
  ElementType2["PAGE"] = "PAGE";
})(ElementType || (ElementType = {}));
var ScriptType;
(function(ScriptType2) {
  ScriptType2["NO_SCRIPT_TYPE"] = "NO_SCRIPT_TYPE";
  ScriptType2["MODULE"] = "MODULE";
})(ScriptType || (ScriptType = {}));
var ExtensionType;
(function(ExtensionType2) {
  ExtensionType2["NONE_EXTENSION"] = "NONE_EXTENSION";
  ExtensionType2["PAYMENTS_GATEWAY_EXTENSION"] = "PAYMENTS_GATEWAY_EXTENSION";
  ExtensionType2["COUPONS_EXTENSION"] = "COUPONS_EXTENSION";
  ExtensionType2["DROPSHIPPING_EXTENSION"] = "DROPSHIPPING_EXTENSION";
  ExtensionType2["FULFILMENT_EXTENSION"] = "FULFILMENT_EXTENSION";
  ExtensionType2["DROPSHIPPING_SUPPLIER_EXTENSION"] = "DROPSHIPPING_SUPPLIER_EXTENSION";
  ExtensionType2["FULFILLMENT_CENTER_EXTENSION"] = "FULFILLMENT_CENTER_EXTENSION";
  ExtensionType2["RESTAURANTS_POS_EXTENSION"] = "RESTAURANTS_POS_EXTENSION";
  ExtensionType2["ART_STORE_EXTENSION"] = "ART_STORE_EXTENSION";
  ExtensionType2["ASCEND_AUTOMATION_EXTENSION"] = "ASCEND_AUTOMATION_EXTENSION";
  ExtensionType2["CONTACT_LABELS_EXTENSION"] = "CONTACT_LABELS_EXTENSION";
})(ExtensionType || (ExtensionType = {}));
var PrimitiveType;
(function(PrimitiveType2) {
  PrimitiveType2["UNKNOWN_PRIMITIVE_TYPE"] = "UNKNOWN_PRIMITIVE_TYPE";
  PrimitiveType2["TEXT"] = "TEXT";
  PrimitiveType2["BOOLEAN"] = "BOOLEAN";
  PrimitiveType2["NUMBER"] = "NUMBER";
})(PrimitiveType || (PrimitiveType = {}));
var SimpleType;
(function(SimpleType2) {
  SimpleType2["UNKNOWN_SIMPLE_TYPE"] = "UNKNOWN_SIMPLE_TYPE";
  SimpleType2["MONEY"] = "MONEY";
  SimpleType2["LINK"] = "LINK";
  SimpleType2["BACKOFFICE_LINK"] = "BACKOFFICE_LINK";
  SimpleType2["LIVESITE_LINK"] = "LIVESITE_LINK";
  SimpleType2["MULTILINGUAL"] = "MULTILINGUAL";
  SimpleType2["IMAGE_LINK"] = "IMAGE_LINK";
  SimpleType2["GUID"] = "GUID";
  SimpleType2["EMAIL"] = "EMAIL";
  SimpleType2["PHONE"] = "PHONE";
  SimpleType2["CONTACT_ID"] = "CONTACT_ID";
})(SimpleType || (SimpleType = {}));
var Exposure;
(function(Exposure2) {
  Exposure2["UNKNOWN_EXPOSURE"] = "UNKNOWN_EXPOSURE";
  Exposure2["SETUP"] = "SETUP";
  Exposure2["HIDDEN"] = "HIDDEN";
  Exposure2["EXPOSED"] = "EXPOSED";
})(Exposure || (Exposure = {}));
var PartialPaymentRestriction;
(function(PartialPaymentRestriction2) {
  PartialPaymentRestriction2["UNDEFINED"] = "UNDEFINED";
  PartialPaymentRestriction2["ALLOW"] = "ALLOW";
  PartialPaymentRestriction2["DISALLOW"] = "DISALLOW";
})(PartialPaymentRestriction || (PartialPaymentRestriction = {}));
var HostContainerId;
(function(HostContainerId2) {
  HostContainerId2["BUSINESS_MANAGER"] = "BUSINESS_MANAGER";
  HostContainerId2["BUSINESS_DASHBOARD_HOST"] = "BUSINESS_DASHBOARD_HOST";
  HostContainerId2["SIDEBAR_FOOTER"] = "SIDEBAR_FOOTER";
})(HostContainerId || (HostContainerId = {}));
var MandatoryField;
(function(MandatoryField2) {
  MandatoryField2["ZIPCODE"] = "ZIPCODE";
  MandatoryField2["CITY"] = "CITY";
  MandatoryField2["STATE"] = "STATE";
  MandatoryField2["ADDRESS"] = "ADDRESS";
  MandatoryField2["COUNTRY_CODE"] = "COUNTRY_CODE";
  MandatoryField2["EMAIL"] = "EMAIL";
  MandatoryField2["PHONE"] = "PHONE";
  MandatoryField2["FIRST_NAME"] = "FIRST_NAME";
  MandatoryField2["LAST_NAME"] = "LAST_NAME";
  MandatoryField2["STREET"] = "STREET";
  MandatoryField2["HOUSE_NUMBER"] = "HOUSE_NUMBER";
  MandatoryField2["TAX_IDENTIFIER"] = "TAX_IDENTIFIER";
})(MandatoryField || (MandatoryField = {}));
var MeasurementSystem;
(function(MeasurementSystem2) {
  MeasurementSystem2["Metric"] = "Metric";
  MeasurementSystem2["Imperial"] = "Imperial";
})(MeasurementSystem || (MeasurementSystem = {}));
var Default;
(function(Default2) {
  Default2["WIX"] = "WIX";
  Default2["POS"] = "POS";
})(Default || (Default = {}));
var HookType;
(function(HookType2) {
  HookType2["UNDEFINED"] = "UNDEFINED";
  HookType2["BEFORE_BLOCKING"] = "BEFORE_BLOCKING";
  HookType2["BEFORE"] = "BEFORE";
  HookType2["AFTER"] = "AFTER";
})(HookType || (HookType = {}));
var Type;
(function(Type2) {
  Type2["UNKNOWN_TYPE"] = "UNKNOWN_TYPE";
  Type2["WIDGET_COMPONENT"] = "WIDGET_COMPONENT";
  Type2["GENERIC"] = "GENERIC";
})(Type || (Type = {}));
var SourceType;
(function(SourceType2) {
  SourceType2["UNKNOWN_SOURCE_TYPE"] = "UNKNOWN_SOURCE_TYPE";
  SourceType2["DEV_CENTER"] = "DEV_CENTER";
  SourceType2["WIX_API"] = "WIX_API";
})(SourceType || (SourceType = {}));
var ExecutionType;
(function(ExecutionType2) {
  ExecutionType2["UNKNOWN_EXECUTION_TYPE"] = "UNKNOWN_EXECUTION_TYPE";
  ExecutionType2["SYNC"] = "SYNC";
  ExecutionType2["ASYNC"] = "ASYNC";
})(ExecutionType || (ExecutionType = {}));
var ExtendingComponentType;
(function(ExtendingComponentType2) {
  ExtendingComponentType2["INVALID"] = "INVALID";
  ExtendingComponentType2["BACK_OFFICE_MENU_ITEM"] = "BACK_OFFICE_MENU_ITEM";
  ExtendingComponentType2["BACK_OFFICE_EXTENSION_WIDGET"] = "BACK_OFFICE_EXTENSION_WIDGET";
})(ExtendingComponentType || (ExtendingComponentType = {}));
var BackOfficeHostingPlatforms;
(function(BackOfficeHostingPlatforms2) {
  BackOfficeHostingPlatforms2["NO_HOSTING_PLATFORM"] = "NO_HOSTING_PLATFORM";
  BackOfficeHostingPlatforms2["BUSINESS_MANAGER"] = "BUSINESS_MANAGER";
  BackOfficeHostingPlatforms2["ACCOUNT_MANAGER"] = "ACCOUNT_MANAGER";
  BackOfficeHostingPlatforms2["DEV_CENTER"] = "DEV_CENTER";
  BackOfficeHostingPlatforms2["ENTERPRISE"] = "ENTERPRISE";
  BackOfficeHostingPlatforms2["PARTNERS_DASHBOARD"] = "PARTNERS_DASHBOARD";
  BackOfficeHostingPlatforms2["FINANCIALS_INTERNAL_BO"] = "FINANCIALS_INTERNAL_BO";
  BackOfficeHostingPlatforms2["FED_GUILD_POC"] = "FED_GUILD_POC";
  BackOfficeHostingPlatforms2["STUDIO_DASHBOARD"] = "STUDIO_DASHBOARD";
  BackOfficeHostingPlatforms2["CHANNELS"] = "CHANNELS";
  BackOfficeHostingPlatforms2["DATA_TOOLS"] = "DATA_TOOLS";
  BackOfficeHostingPlatforms2["PSP_BACKOFFICE"] = "PSP_BACKOFFICE";
  BackOfficeHostingPlatforms2["RISE_PLATFORM_ACCOUNT_DASHBOARD"] = "RISE_PLATFORM_ACCOUNT_DASHBOARD";
  BackOfficeHostingPlatforms2["DEMO_DASHBOARD_ENTERPRISE"] = "DEMO_DASHBOARD_ENTERPRISE";
  BackOfficeHostingPlatforms2["AI_SCHEDULING_ASSISTANT_DASHBOARD"] = "AI_SCHEDULING_ASSISTANT_DASHBOARD";
  BackOfficeHostingPlatforms2["GETTING_PAID"] = "GETTING_PAID";
  BackOfficeHostingPlatforms2["DATA"] = "DATA";
  BackOfficeHostingPlatforms2["LITE_DASHBOARD"] = "LITE_DASHBOARD";
  BackOfficeHostingPlatforms2["PICASSO_EDITOR"] = "PICASSO_EDITOR";
})(BackOfficeHostingPlatforms || (BackOfficeHostingPlatforms = {}));
var DtsDefinitionType;
(function(DtsDefinitionType2) {
  DtsDefinitionType2["UNKNOWN"] = "UNKNOWN";
  DtsDefinitionType2["DTS_HTTP_LINK"] = "DTS_HTTP_LINK";
  DtsDefinitionType2["DTS_CONTENT"] = "DTS_CONTENT";
})(DtsDefinitionType || (DtsDefinitionType = {}));
var SlotDataType;
(function(SlotDataType2) {
  SlotDataType2["UNKNOWN"] = "UNKNOWN";
  SlotDataType2["WIDGET"] = "WIDGET";
  SlotDataType2["MENU"] = "MENU";
})(SlotDataType || (SlotDataType = {}));
var BackOfficeScriptAssetType;
(function(BackOfficeScriptAssetType2) {
  BackOfficeScriptAssetType2["UNKNOWN_TYPE"] = "UNKNOWN_TYPE";
  BackOfficeScriptAssetType2["MODULE"] = "MODULE";
})(BackOfficeScriptAssetType || (BackOfficeScriptAssetType = {}));
var ViewMode;
(function(ViewMode2) {
  ViewMode2["PAGE"] = "PAGE";
  ViewMode2["MODAL"] = "MODAL";
})(ViewMode || (ViewMode = {}));
var AlgorithmType;
(function(AlgorithmType2) {
  AlgorithmType2["UNSPECIFIED"] = "UNSPECIFIED";
  AlgorithmType2["RELATED_ITEMS"] = "RELATED_ITEMS";
  AlgorithmType2["GLOBAL"] = "GLOBAL";
})(AlgorithmType || (AlgorithmType = {}));
var LockableOperation;
(function(LockableOperation2) {
  LockableOperation2["UNSPECIFIED_LOCKABLE"] = "UNSPECIFIED_LOCKABLE";
  LockableOperation2["LOCKED"] = "LOCKED";
  LockableOperation2["UNLOCKED"] = "UNLOCKED";
})(LockableOperation || (LockableOperation = {}));
var RestrictedOperation;
(function(RestrictedOperation2) {
  RestrictedOperation2["UNSPECIFIED_RESTRICTED"] = "UNSPECIFIED_RESTRICTED";
  RestrictedOperation2["ALLOW"] = "ALLOW";
  RestrictedOperation2["WARN"] = "WARN";
})(RestrictedOperation || (RestrictedOperation = {}));
var BlockType;
(function(BlockType2) {
  BlockType2["UNKNOWN"] = "UNKNOWN";
  BlockType2["OR"] = "OR";
  BlockType2["AND"] = "AND";
})(BlockType || (BlockType = {}));
var Domain;
(function(Domain2) {
  Domain2["USER"] = "USER";
  Domain2["WIX"] = "WIX";
})(Domain || (Domain = {}));
var Status2;
(function(Status4) {
  Status4["ACTIVE"] = "ACTIVE";
  Status4["INACTIVE"] = "INACTIVE";
})(Status2 || (Status2 = {}));
var TextInputDisplayType;
(function(TextInputDisplayType2) {
  TextInputDisplayType2["TEXT"] = "TEXT";
  TextInputDisplayType2["PASSWORD"] = "PASSWORD";
  TextInputDisplayType2["CODE"] = "CODE";
})(TextInputDisplayType || (TextInputDisplayType = {}));
var ThumbnailType;
(function(ThumbnailType2) {
  ThumbnailType2["UNKNOWN_THUMBNAIL_TYPE"] = "UNKNOWN_THUMBNAIL_TYPE";
  ThumbnailType2["IMAGE"] = "IMAGE";
  ThumbnailType2["AVATAR"] = "AVATAR";
})(ThumbnailType || (ThumbnailType = {}));
var ImageShape;
(function(ImageShape2) {
  ImageShape2["UNKNOWN_SHAPE"] = "UNKNOWN_SHAPE";
  ImageShape2["CIRCLE"] = "CIRCLE";
  ImageShape2["SQUARE"] = "SQUARE";
  ImageShape2["RECTANGLE"] = "RECTANGLE";
})(ImageShape || (ImageShape = {}));
var AvatarShape;
(function(AvatarShape2) {
  AvatarShape2["UNKNOWN_SHAPE"] = "UNKNOWN_SHAPE";
  AvatarShape2["CIRCLE"] = "CIRCLE";
  AvatarShape2["SQUARE"] = "SQUARE";
})(AvatarShape || (AvatarShape = {}));
var Tag;
(function(Tag2) {
  Tag2["UNKNOWN_TAG"] = "UNKNOWN_TAG";
  Tag2["EMBEDDABLE"] = "EMBEDDABLE";
  Tag2["ECOM_EDIT_ORDER"] = "ECOM_EDIT_ORDER";
  Tag2["ECOM_CREATE_ORDER"] = "ECOM_CREATE_ORDER";
  Tag2["INBOX"] = "INBOX";
  Tag2["ECOM_CREATE_PAY_LINK"] = "ECOM_CREATE_PAY_LINK";
  Tag2["ECOM_DELIVERY_PROFILES"] = "ECOM_DELIVERY_PROFILES";
  Tag2["ADD_TO_CATEGORY"] = "ADD_TO_CATEGORY";
  Tag2["MY_BUSINESS"] = "MY_BUSINESS";
})(Tag || (Tag = {}));
var FilterSelectionType;
(function(FilterSelectionType2) {
  FilterSelectionType2["UNKNOWN_SELECTION"] = "UNKNOWN_SELECTION";
  FilterSelectionType2["SINGLE_SELECTION"] = "SINGLE_SELECTION";
  FilterSelectionType2["MULTIPLE_SELECTION"] = "MULTIPLE_SELECTION";
})(FilterSelectionType || (FilterSelectionType = {}));
var FilterOptionsType;
(function(FilterOptionsType2) {
  FilterOptionsType2["UNKNOWN_FILTER_OPTIONS"] = "UNKNOWN_FILTER_OPTIONS";
  FilterOptionsType2["STATIC"] = "STATIC";
  FilterOptionsType2["PROVIDER"] = "PROVIDER";
})(FilterOptionsType || (FilterOptionsType = {}));
var PaginationMode;
(function(PaginationMode2) {
  PaginationMode2["UNKNOWN_PAGINATION_MODE"] = "UNKNOWN_PAGINATION_MODE";
  PaginationMode2["CURSOR"] = "CURSOR";
  PaginationMode2["OFFSET"] = "OFFSET";
})(PaginationMode || (PaginationMode = {}));
var ChannelType;
(function(ChannelType2) {
  ChannelType2["UNKNOWN_CHANNEL_TYPE"] = "UNKNOWN_CHANNEL_TYPE";
  ChannelType2["DIRECT_MESSAGING"] = "DIRECT_MESSAGING";
  ChannelType2["EMAIL"] = "EMAIL";
  ChannelType2["SMS"] = "SMS";
})(ChannelType || (ChannelType = {}));
var AcceptedDirectMessageType;
(function(AcceptedDirectMessageType2) {
  AcceptedDirectMessageType2["TEXT"] = "TEXT";
  AcceptedDirectMessageType2["MEDIA"] = "MEDIA";
  AcceptedDirectMessageType2["CARD"] = "CARD";
  AcceptedDirectMessageType2["ANNOUNCEMENT"] = "ANNOUNCEMENT";
  AcceptedDirectMessageType2["FORM"] = "FORM";
})(AcceptedDirectMessageType || (AcceptedDirectMessageType = {}));
var MediaMimeType;
(function(MediaMimeType2) {
  MediaMimeType2["UNKNOWN_MIME_TYPE"] = "UNKNOWN_MIME_TYPE";
  MediaMimeType2["IMAGE_JPEG"] = "IMAGE_JPEG";
  MediaMimeType2["IMAGE_PNG"] = "IMAGE_PNG";
  MediaMimeType2["IMAGE_WEBP"] = "IMAGE_WEBP";
  MediaMimeType2["VIDEO_MPEG"] = "VIDEO_MPEG";
  MediaMimeType2["VIDEO_MP4"] = "VIDEO_MP4";
  MediaMimeType2["VIDEO_3GP"] = "VIDEO_3GP";
  MediaMimeType2["AUDIO_AAC"] = "AUDIO_AAC";
  MediaMimeType2["AUDIO_MP4"] = "AUDIO_MP4";
  MediaMimeType2["AUDIO_MPEG"] = "AUDIO_MPEG";
  MediaMimeType2["AUDIO_OGG"] = "AUDIO_OGG";
  MediaMimeType2["AUDIO_OPUS"] = "AUDIO_OPUS";
  MediaMimeType2["APPLICATION_PDF"] = "APPLICATION_PDF";
  MediaMimeType2["APPLICATION_OCTET_STREAM"] = "APPLICATION_OCTET_STREAM";
})(MediaMimeType || (MediaMimeType = {}));
var AcceptedSmsMessageType;
(function(AcceptedSmsMessageType2) {
  AcceptedSmsMessageType2["SMS"] = "SMS";
  AcceptedSmsMessageType2["MMS"] = "MMS";
})(AcceptedSmsMessageType || (AcceptedSmsMessageType = {}));
var InitDirection;
(function(InitDirection2) {
  InitDirection2["UNKNOWN_INIT_DIRECTION"] = "UNKNOWN_INIT_DIRECTION";
  InitDirection2["BUSINESS"] = "BUSINESS";
  InitDirection2["CUSTOMER"] = "CUSTOMER";
  InitDirection2["BOTH"] = "BOTH";
})(InitDirection || (InitDirection = {}));
var NodeType;
(function(NodeType2) {
  NodeType2["PARAGRAPH"] = "PARAGRAPH";
  NodeType2["TEXT"] = "TEXT";
  NodeType2["HEADING"] = "HEADING";
  NodeType2["BULLETED_LIST"] = "BULLETED_LIST";
  NodeType2["ORDERED_LIST"] = "ORDERED_LIST";
  NodeType2["LIST_ITEM"] = "LIST_ITEM";
  NodeType2["BLOCKQUOTE"] = "BLOCKQUOTE";
  NodeType2["CODE_BLOCK"] = "CODE_BLOCK";
  NodeType2["VIDEO"] = "VIDEO";
  NodeType2["DIVIDER"] = "DIVIDER";
  NodeType2["FILE"] = "FILE";
  NodeType2["GALLERY"] = "GALLERY";
  NodeType2["GIF"] = "GIF";
  NodeType2["HTML"] = "HTML";
  NodeType2["IMAGE"] = "IMAGE";
  NodeType2["LINK_PREVIEW"] = "LINK_PREVIEW";
  NodeType2["MAP"] = "MAP";
  NodeType2["POLL"] = "POLL";
  NodeType2["APP_EMBED"] = "APP_EMBED";
  NodeType2["BUTTON"] = "BUTTON";
  NodeType2["COLLAPSIBLE_LIST"] = "COLLAPSIBLE_LIST";
  NodeType2["TABLE"] = "TABLE";
  NodeType2["EMBED"] = "EMBED";
  NodeType2["COLLAPSIBLE_ITEM"] = "COLLAPSIBLE_ITEM";
  NodeType2["COLLAPSIBLE_ITEM_TITLE"] = "COLLAPSIBLE_ITEM_TITLE";
  NodeType2["COLLAPSIBLE_ITEM_BODY"] = "COLLAPSIBLE_ITEM_BODY";
  NodeType2["TABLE_CELL"] = "TABLE_CELL";
  NodeType2["TABLE_ROW"] = "TABLE_ROW";
  NodeType2["EXTERNAL"] = "EXTERNAL";
  NodeType2["AUDIO"] = "AUDIO";
  NodeType2["CAPTION"] = "CAPTION";
  NodeType2["LAYOUT"] = "LAYOUT";
  NodeType2["LAYOUT_CELL"] = "LAYOUT_CELL";
})(NodeType || (NodeType = {}));
var WidthType;
(function(WidthType2) {
  WidthType2["CONTENT"] = "CONTENT";
  WidthType2["SMALL"] = "SMALL";
  WidthType2["ORIGINAL"] = "ORIGINAL";
  WidthType2["FULL_WIDTH"] = "FULL_WIDTH";
})(WidthType || (WidthType = {}));
var PluginContainerDataAlignment;
(function(PluginContainerDataAlignment2) {
  PluginContainerDataAlignment2["CENTER"] = "CENTER";
  PluginContainerDataAlignment2["LEFT"] = "LEFT";
  PluginContainerDataAlignment2["RIGHT"] = "RIGHT";
})(PluginContainerDataAlignment || (PluginContainerDataAlignment = {}));
var ButtonDataType;
(function(ButtonDataType2) {
  ButtonDataType2["LINK"] = "LINK";
  ButtonDataType2["ACTION"] = "ACTION";
})(ButtonDataType || (ButtonDataType = {}));
var LinkTarget;
(function(LinkTarget2) {
  LinkTarget2["SELF"] = "SELF";
  LinkTarget2["BLANK"] = "BLANK";
  LinkTarget2["PARENT"] = "PARENT";
  LinkTarget2["TOP"] = "TOP";
})(LinkTarget || (LinkTarget = {}));
var TextAlignment;
(function(TextAlignment2) {
  TextAlignment2["AUTO"] = "AUTO";
  TextAlignment2["LEFT"] = "LEFT";
  TextAlignment2["RIGHT"] = "RIGHT";
  TextAlignment2["CENTER"] = "CENTER";
  TextAlignment2["JUSTIFY"] = "JUSTIFY";
})(TextAlignment || (TextAlignment = {}));
var LineStyle;
(function(LineStyle2) {
  LineStyle2["SINGLE"] = "SINGLE";
  LineStyle2["DOUBLE"] = "DOUBLE";
  LineStyle2["DASHED"] = "DASHED";
  LineStyle2["DOTTED"] = "DOTTED";
})(LineStyle || (LineStyle = {}));
var DividerDataWidth;
(function(DividerDataWidth2) {
  DividerDataWidth2["LARGE"] = "LARGE";
  DividerDataWidth2["MEDIUM"] = "MEDIUM";
  DividerDataWidth2["SMALL"] = "SMALL";
})(DividerDataWidth || (DividerDataWidth = {}));
var Alignment;
(function(Alignment2) {
  Alignment2["CENTER"] = "CENTER";
  Alignment2["LEFT"] = "LEFT";
  Alignment2["RIGHT"] = "RIGHT";
})(Alignment || (Alignment = {}));
var PDFSettingsViewMode;
(function(PDFSettingsViewMode2) {
  PDFSettingsViewMode2["NONE"] = "NONE";
  PDFSettingsViewMode2["FULL"] = "FULL";
  PDFSettingsViewMode2["MINI"] = "MINI";
})(PDFSettingsViewMode || (PDFSettingsViewMode = {}));
var LayoutType;
(function(LayoutType2) {
  LayoutType2["COLLAGE"] = "COLLAGE";
  LayoutType2["MASONRY"] = "MASONRY";
  LayoutType2["GRID"] = "GRID";
  LayoutType2["THUMBNAIL"] = "THUMBNAIL";
  LayoutType2["SLIDER"] = "SLIDER";
  LayoutType2["SLIDESHOW"] = "SLIDESHOW";
  LayoutType2["PANORAMA"] = "PANORAMA";
  LayoutType2["COLUMN"] = "COLUMN";
  LayoutType2["MAGIC"] = "MAGIC";
  LayoutType2["FULLSIZE"] = "FULLSIZE";
})(LayoutType || (LayoutType = {}));
var Orientation;
(function(Orientation2) {
  Orientation2["ROWS"] = "ROWS";
  Orientation2["COLUMNS"] = "COLUMNS";
})(Orientation || (Orientation = {}));
var Crop;
(function(Crop2) {
  Crop2["FILL"] = "FILL";
  Crop2["FIT"] = "FIT";
})(Crop || (Crop = {}));
var ThumbnailsAlignment;
(function(ThumbnailsAlignment2) {
  ThumbnailsAlignment2["TOP"] = "TOP";
  ThumbnailsAlignment2["RIGHT"] = "RIGHT";
  ThumbnailsAlignment2["BOTTOM"] = "BOTTOM";
  ThumbnailsAlignment2["LEFT"] = "LEFT";
  ThumbnailsAlignment2["NONE"] = "NONE";
})(ThumbnailsAlignment || (ThumbnailsAlignment = {}));
var GIFType;
(function(GIFType2) {
  GIFType2["NORMAL"] = "NORMAL";
  GIFType2["STICKER"] = "STICKER";
})(GIFType || (GIFType = {}));
var Source;
(function(Source2) {
  Source2["HTML"] = "HTML";
  Source2["ADSENSE"] = "ADSENSE";
})(Source || (Source = {}));
var StylesPosition;
(function(StylesPosition2) {
  StylesPosition2["START"] = "START";
  StylesPosition2["END"] = "END";
  StylesPosition2["TOP"] = "TOP";
  StylesPosition2["HIDDEN"] = "HIDDEN";
})(StylesPosition || (StylesPosition = {}));
var MapType;
(function(MapType2) {
  MapType2["ROADMAP"] = "ROADMAP";
  MapType2["SATELITE"] = "SATELITE";
  MapType2["HYBRID"] = "HYBRID";
  MapType2["TERRAIN"] = "TERRAIN";
})(MapType || (MapType = {}));
var ViewRole;
(function(ViewRole2) {
  ViewRole2["CREATOR"] = "CREATOR";
  ViewRole2["VOTERS"] = "VOTERS";
  ViewRole2["EVERYONE"] = "EVERYONE";
})(ViewRole || (ViewRole = {}));
var VoteRole;
(function(VoteRole2) {
  VoteRole2["SITE_MEMBERS"] = "SITE_MEMBERS";
  VoteRole2["ALL"] = "ALL";
})(VoteRole || (VoteRole = {}));
var PollLayoutType;
(function(PollLayoutType2) {
  PollLayoutType2["LIST"] = "LIST";
  PollLayoutType2["GRID"] = "GRID";
})(PollLayoutType || (PollLayoutType = {}));
var PollLayoutDirection;
(function(PollLayoutDirection2) {
  PollLayoutDirection2["LTR"] = "LTR";
  PollLayoutDirection2["RTL"] = "RTL";
})(PollLayoutDirection || (PollLayoutDirection = {}));
var BackgroundType;
(function(BackgroundType2) {
  BackgroundType2["COLOR"] = "COLOR";
  BackgroundType2["IMAGE"] = "IMAGE";
  BackgroundType2["GRADIENT"] = "GRADIENT";
})(BackgroundType || (BackgroundType = {}));
var DecorationType;
(function(DecorationType2) {
  DecorationType2["BOLD"] = "BOLD";
  DecorationType2["ITALIC"] = "ITALIC";
  DecorationType2["UNDERLINE"] = "UNDERLINE";
  DecorationType2["SPOILER"] = "SPOILER";
  DecorationType2["ANCHOR"] = "ANCHOR";
  DecorationType2["MENTION"] = "MENTION";
  DecorationType2["LINK"] = "LINK";
  DecorationType2["COLOR"] = "COLOR";
  DecorationType2["FONT_SIZE"] = "FONT_SIZE";
  DecorationType2["EXTERNAL"] = "EXTERNAL";
  DecorationType2["STRIKETHROUGH"] = "STRIKETHROUGH";
  DecorationType2["SUPERSCRIPT"] = "SUPERSCRIPT";
  DecorationType2["SUBSCRIPT"] = "SUBSCRIPT";
})(DecorationType || (DecorationType = {}));
var FontType;
(function(FontType2) {
  FontType2["PX"] = "PX";
  FontType2["EM"] = "EM";
})(FontType || (FontType = {}));
var AppType;
(function(AppType3) {
  AppType3["PRODUCT"] = "PRODUCT";
  AppType3["EVENT"] = "EVENT";
  AppType3["BOOKING"] = "BOOKING";
})(AppType || (AppType = {}));
var InitialExpandedItems;
(function(InitialExpandedItems2) {
  InitialExpandedItems2["FIRST"] = "FIRST";
  InitialExpandedItems2["ALL"] = "ALL";
  InitialExpandedItems2["NONE"] = "NONE";
})(InitialExpandedItems || (InitialExpandedItems = {}));
var Direction;
(function(Direction2) {
  Direction2["LTR"] = "LTR";
  Direction2["RTL"] = "RTL";
})(Direction || (Direction = {}));
var VerticalAlignment;
(function(VerticalAlignment2) {
  VerticalAlignment2["TOP"] = "TOP";
  VerticalAlignment2["MIDDLE"] = "MIDDLE";
  VerticalAlignment2["BOTTOM"] = "BOTTOM";
})(VerticalAlignment || (VerticalAlignment = {}));
var NullValue;
(function(NullValue2) {
  NullValue2["NULL_VALUE"] = "NULL_VALUE";
})(NullValue || (NullValue = {}));
var CalendarType;
(function(CalendarType2) {
  CalendarType2["UNDEFINED"] = "UNDEFINED";
  CalendarType2["GOOGLE"] = "GOOGLE";
  CalendarType2["I_CAL"] = "I_CAL";
  CalendarType2["MICROSOFT"] = "MICROSOFT";
  CalendarType2["OTHER"] = "OTHER";
})(CalendarType || (CalendarType = {}));
var ConnectMethod;
(function(ConnectMethod2) {
  ConnectMethod2["OAUTH"] = "OAUTH";
  ConnectMethod2["CREDENTIALS"] = "CREDENTIALS";
})(ConnectMethod || (ConnectMethod = {}));
var ListEventFromCalendars;
(function(ListEventFromCalendars2) {
  ListEventFromCalendars2["LIST_NOT_SUPPORTED"] = "LIST_NOT_SUPPORTED";
  ListEventFromCalendars2["LIST_FROM_PRIMARY_CALENDAR_ONLY"] = "LIST_FROM_PRIMARY_CALENDAR_ONLY";
  ListEventFromCalendars2["LIST_FROM_SPECIFIC_CALENDARS"] = "LIST_FROM_SPECIFIC_CALENDARS";
})(ListEventFromCalendars || (ListEventFromCalendars = {}));
var SyncToCalendar;
(function(SyncToCalendar2) {
  SyncToCalendar2["SYNC_NOT_SUPPORTED"] = "SYNC_NOT_SUPPORTED";
  SyncToCalendar2["SYNC_TO_PRIMARY_CALENDAR_ONLY"] = "SYNC_TO_PRIMARY_CALENDAR_ONLY";
  SyncToCalendar2["SYNC_TO_DEDICATED_CALENDAR"] = "SYNC_TO_DEDICATED_CALENDAR";
})(SyncToCalendar || (SyncToCalendar = {}));
var OAuthAppType;
(function(OAuthAppType2) {
  OAuthAppType2["OAUTH_APP_TYPE_UNSPECIFIED"] = "OAUTH_APP_TYPE_UNSPECIFIED";
  OAuthAppType2["WEB_APP"] = "WEB_APP";
  OAuthAppType2["MOBILE"] = "MOBILE";
  OAuthAppType2["OTHER"] = "OTHER";
})(OAuthAppType || (OAuthAppType = {}));
var OAuthTechnologies;
(function(OAuthTechnologies2) {
  OAuthTechnologies2["OAUTH_TECHNOLOGY_UNSPECIFIED"] = "OAUTH_TECHNOLOGY_UNSPECIFIED";
  OAuthTechnologies2["JAVASCRIPT"] = "JAVASCRIPT";
  OAuthTechnologies2["ANGULAR"] = "ANGULAR";
  OAuthTechnologies2["VUE"] = "VUE";
  OAuthTechnologies2["REACT"] = "REACT";
  OAuthTechnologies2["REACT_NATIVE"] = "REACT_NATIVE";
  OAuthTechnologies2["IOS"] = "IOS";
  OAuthTechnologies2["ANDROID"] = "ANDROID";
  OAuthTechnologies2["OTHER_TECHNOLOGY"] = "OTHER_TECHNOLOGY";
})(OAuthTechnologies || (OAuthTechnologies = {}));
var NavigationType;
(function(NavigationType2) {
  NavigationType2["NAVIGATE"] = "NAVIGATE";
  NavigationType2["MODAL"] = "MODAL";
})(NavigationType || (NavigationType = {}));
var State2;
(function(State3) {
  State3["UNKNOWN_STATE"] = "UNKNOWN_STATE";
  State3["DEFAULT_ON"] = "DEFAULT_ON";
  State3["DEFAULT_OFF"] = "DEFAULT_OFF";
  State3["ALWAYS_ON"] = "ALWAYS_ON";
})(State2 || (State2 = {}));
var NotificationTopicState;
(function(NotificationTopicState2) {
  NotificationTopicState2["UNKNOWN_STATE"] = "UNKNOWN_STATE";
  NotificationTopicState2["DEFAULT_ON"] = "DEFAULT_ON";
  NotificationTopicState2["DEFAULT_OFF"] = "DEFAULT_OFF";
  NotificationTopicState2["ALWAYS_ON"] = "ALWAYS_ON";
})(NotificationTopicState || (NotificationTopicState = {}));
var NotificationTopicType;
(function(NotificationTopicType2) {
  NotificationTopicType2["UNKNOWN_TYPE"] = "UNKNOWN_TYPE";
  NotificationTopicType2["PROMOTIONAL"] = "PROMOTIONAL";
  NotificationTopicType2["TRANSACTIONAL"] = "TRANSACTIONAL";
})(NotificationTopicType || (NotificationTopicType = {}));
var RecipientType;
(function(RecipientType2) {
  RecipientType2["UNKNOWN_RECIPIENT_TYPE"] = "UNKNOWN_RECIPIENT_TYPE";
  RecipientType2["WIX_USER"] = "WIX_USER";
  RecipientType2["CONTACT"] = "CONTACT";
})(RecipientType || (RecipientType = {}));
var PlanFormPricingOption;
(function(PlanFormPricingOption2) {
  PlanFormPricingOption2["UNKNOWN_OPTION"] = "UNKNOWN_OPTION";
  PlanFormPricingOption2["FREE"] = "FREE";
  PlanFormPricingOption2["ONE_TIME"] = "ONE_TIME";
  PlanFormPricingOption2["RECURRING"] = "RECURRING";
})(PlanFormPricingOption || (PlanFormPricingOption = {}));
var PlanFormDefaultSection;
(function(PlanFormDefaultSection2) {
  PlanFormDefaultSection2["UNKNWON_PLAN_FORM_SECTION"] = "UNKNWON_PLAN_FORM_SECTION";
  PlanFormDefaultSection2["INFO"] = "INFO";
  PlanFormDefaultSection2["BENEFITS"] = "BENEFITS";
  PlanFormDefaultSection2["PAGE_PERMISSIONS"] = "PAGE_PERMISSIONS";
  PlanFormDefaultSection2["PRICING_AND_DURATION"] = "PRICING_AND_DURATION";
  PlanFormDefaultSection2["ADVANCED_SETTINGS"] = "ADVANCED_SETTINGS";
})(PlanFormDefaultSection || (PlanFormDefaultSection = {}));
var PlanPeriodUnit;
(function(PlanPeriodUnit2) {
  PlanPeriodUnit2["UNDEFINED"] = "UNDEFINED";
  PlanPeriodUnit2["DAY"] = "DAY";
  PlanPeriodUnit2["WEEK"] = "WEEK";
  PlanPeriodUnit2["MONTH"] = "MONTH";
  PlanPeriodUnit2["YEAR"] = "YEAR";
})(PlanPeriodUnit || (PlanPeriodUnit = {}));
var TypedDynamicParamType;
(function(TypedDynamicParamType2) {
  TypedDynamicParamType2["UNKNOWN_TYPE"] = "UNKNOWN_TYPE";
  TypedDynamicParamType2["STRING"] = "STRING";
  TypedDynamicParamType2["ARRAY"] = "ARRAY";
})(TypedDynamicParamType || (TypedDynamicParamType = {}));
var UserNotificationDataRecipientFilterType;
(function(UserNotificationDataRecipientFilterType2) {
  UserNotificationDataRecipientFilterType2["UNKNOWN_TYPE"] = "UNKNOWN_TYPE";
  UserNotificationDataRecipientFilterType2["WIX_USERS"] = "WIX_USERS";
  UserNotificationDataRecipientFilterType2["SITE_CONTRIBUTORS"] = "SITE_CONTRIBUTORS";
})(UserNotificationDataRecipientFilterType || (UserNotificationDataRecipientFilterType = {}));
var UserNotificationDataInitiatorType;
(function(UserNotificationDataInitiatorType2) {
  UserNotificationDataInitiatorType2["UNKNOWN_TYPE"] = "UNKNOWN_TYPE";
  UserNotificationDataInitiatorType2["WIX_USER"] = "WIX_USER";
  UserNotificationDataInitiatorType2["WIX_APP"] = "WIX_APP";
})(UserNotificationDataInitiatorType || (UserNotificationDataInitiatorType = {}));
var Level;
(function(Level2) {
  Level2["UNKNOWN_LEVEL"] = "UNKNOWN_LEVEL";
  Level2["ACCOUNT"] = "ACCOUNT";
  Level2["USER"] = "USER";
  Level2["SITE"] = "SITE";
})(Level || (Level = {}));
var IconType;
(function(IconType2) {
  IconType2["UNKNOWN_TYPE"] = "UNKNOWN_TYPE";
  IconType2["URL"] = "URL";
  IconType2["SPOUT"] = "SPOUT";
  IconType2["INITIATOR"] = "INITIATOR";
})(IconType || (IconType = {}));
var MobileApplication;
(function(MobileApplication2) {
  MobileApplication2["UNKNOWN_MOBILE_APPLICATION"] = "UNKNOWN_MOBILE_APPLICATION";
  MobileApplication2["OWNER"] = "OWNER";
  MobileApplication2["SPACES"] = "SPACES";
  MobileApplication2["DINE"] = "DINE";
  MobileApplication2["FITNESS"] = "FITNESS";
  MobileApplication2["WIX_LIFE"] = "WIX_LIFE";
  MobileApplication2["WIX_PARTNERS"] = "WIX_PARTNERS";
  MobileApplication2["BRANDED"] = "BRANDED";
  MobileApplication2["WIX_STUDIO"] = "WIX_STUDIO";
  MobileApplication2["NO_SPECIFIC_TARGET_APP"] = "NO_SPECIFIC_TARGET_APP";
})(MobileApplication || (MobileApplication = {}));
var AndroidStyle;
(function(AndroidStyle2) {
  AndroidStyle2["UNKNOWN_ANDROID_STYLE"] = "UNKNOWN_ANDROID_STYLE";
  AndroidStyle2["BIG_TEXT"] = "BIG_TEXT";
  AndroidStyle2["MESSAGING"] = "MESSAGING";
  AndroidStyle2["INBOX"] = "INBOX";
})(AndroidStyle || (AndroidStyle = {}));
var ExposureRuleType;
(function(ExposureRuleType2) {
  ExposureRuleType2["UNKNOWN_TYPE"] = "UNKNOWN_TYPE";
  ExposureRuleType2["OPEN_TO_ALL"] = "OPEN_TO_ALL";
  ExposureRuleType2["OPEN_TO_NONE"] = "OPEN_TO_NONE";
  ExposureRuleType2["BOOLEAN_EXPERIMENT"] = "BOOLEAN_EXPERIMENT";
  ExposureRuleType2["CUSTOM_EXPERIMENT"] = "CUSTOM_EXPERIMENT";
})(ExposureRuleType || (ExposureRuleType = {}));
var Intent;
(function(Intent2) {
  Intent2["UNKNOWN_INTENT"] = "UNKNOWN_INTENT";
  Intent2["MARKETING"] = "MARKETING";
  Intent2["TRANSACTIONAL"] = "TRANSACTIONAL";
})(Intent || (Intent = {}));
var UserNotificationDataType;
(function(UserNotificationDataType2) {
  UserNotificationDataType2["UNKNOWN_TYPE"] = "UNKNOWN_TYPE";
  UserNotificationDataType2["REGULAR"] = "REGULAR";
  UserNotificationDataType2["ALERT_WARNING"] = "ALERT_WARNING";
  UserNotificationDataType2["ALERT_INFO"] = "ALERT_INFO";
})(UserNotificationDataType || (UserNotificationDataType = {}));
var RecipientFilterType;
(function(RecipientFilterType2) {
  RecipientFilterType2["UNKNOWN_TYPE"] = "UNKNOWN_TYPE";
  RecipientFilterType2["CONTACTS"] = "CONTACTS";
})(RecipientFilterType || (RecipientFilterType = {}));
var InitiatorType;
(function(InitiatorType2) {
  InitiatorType2["UNKNOWN_TYPE"] = "UNKNOWN_TYPE";
  InitiatorType2["CONTACT"] = "CONTACT";
  InitiatorType2["SITE_MEMBER"] = "SITE_MEMBER";
})(InitiatorType || (InitiatorType = {}));
var Environment;
(function(Environment2) {
  Environment2["LIVE_SITE"] = "LIVE_SITE";
  Environment2["BUSINESS_MANAGER"] = "BUSINESS_MANAGER";
})(Environment || (Environment = {}));
var SchemaScope;
(function(SchemaScope2) {
  SchemaScope2["UNKNOWN_SCOPE"] = "UNKNOWN_SCOPE";
  SchemaScope2["GLOBAL"] = "GLOBAL";
  SchemaScope2["SITE"] = "SITE";
})(SchemaScope || (SchemaScope = {}));
var FieldType;
(function(FieldType2) {
  FieldType2["UNDEFINED_TYPE"] = "UNDEFINED_TYPE";
  FieldType2["SHORT_TEXT"] = "SHORT_TEXT";
  FieldType2["LONG_TEXT"] = "LONG_TEXT";
  FieldType2["HTML"] = "HTML";
  FieldType2["RICH_CONTENT"] = "RICH_CONTENT";
  FieldType2["IMAGE"] = "IMAGE";
  FieldType2["IMAGE_LINK"] = "IMAGE_LINK";
  FieldType2["JSON"] = "JSON";
  FieldType2["VIDEO"] = "VIDEO";
  FieldType2["DOCUMENT"] = "DOCUMENT";
})(FieldType || (FieldType = {}));
var CustomElementScriptType;
(function(CustomElementScriptType2) {
  CustomElementScriptType2["NONE"] = "NONE";
  CustomElementScriptType2["COMMON_JS"] = "COMMON_JS";
  CustomElementScriptType2["ES_MODULE"] = "ES_MODULE";
})(CustomElementScriptType || (CustomElementScriptType = {}));
var HeightMode;
(function(HeightMode2) {
  HeightMode2["UNKNOWN"] = "UNKNOWN";
  HeightMode2["AUTO"] = "AUTO";
  HeightMode2["FIXED"] = "FIXED";
})(HeightMode || (HeightMode = {}));
var FieldFieldType;
(function(FieldFieldType2) {
  FieldFieldType2["UNKNOWN_FIELD_TYPE"] = "UNKNOWN_FIELD_TYPE";
  FieldFieldType2["INPUT"] = "INPUT";
  FieldFieldType2["DISPLAY"] = "DISPLAY";
})(FieldFieldType || (FieldFieldType = {}));
var Format;
(function(Format2) {
  Format2["UNKNOWN_FORMAT"] = "UNKNOWN_FORMAT";
  Format2["DATE"] = "DATE";
  Format2["TIME"] = "TIME";
  Format2["DATE_TIME"] = "DATE_TIME";
  Format2["EMAIL"] = "EMAIL";
  Format2["URL"] = "URL";
  Format2["UUID"] = "UUID";
  Format2["PHONE"] = "PHONE";
  Format2["URI"] = "URI";
  Format2["HOSTNAME"] = "HOSTNAME";
  Format2["COLOR_HEX"] = "COLOR_HEX";
  Format2["CURRENCY"] = "CURRENCY";
  Format2["LANGUAGE"] = "LANGUAGE";
  Format2["DATE_OPTIONAL_TIME"] = "DATE_OPTIONAL_TIME";
})(Format || (Format = {}));
var StringComponentType;
(function(StringComponentType2) {
  StringComponentType2["UNKNOWN_COMPONENT_TYPE"] = "UNKNOWN_COMPONENT_TYPE";
  StringComponentType2["TEXT_INPUT"] = "TEXT_INPUT";
  StringComponentType2["RADIO_GROUP"] = "RADIO_GROUP";
  StringComponentType2["DROPDOWN"] = "DROPDOWN";
  StringComponentType2["DATE_TIME"] = "DATE_TIME";
  StringComponentType2["PHONE_INPUT"] = "PHONE_INPUT";
  StringComponentType2["DATE_INPUT"] = "DATE_INPUT";
  StringComponentType2["TIME_INPUT"] = "TIME_INPUT";
  StringComponentType2["DATE_PICKER"] = "DATE_PICKER";
})(StringComponentType || (StringComponentType = {}));
var NumberOfColumns;
(function(NumberOfColumns2) {
  NumberOfColumns2["UNKNOWN"] = "UNKNOWN";
  NumberOfColumns2["ZERO"] = "ZERO";
  NumberOfColumns2["ONE"] = "ONE";
  NumberOfColumns2["TWO"] = "TWO";
  NumberOfColumns2["THREE"] = "THREE";
})(NumberOfColumns || (NumberOfColumns = {}));
var FirstDayOfWeek;
(function(FirstDayOfWeek2) {
  FirstDayOfWeek2["MONDAY"] = "MONDAY";
  FirstDayOfWeek2["SUNDAY"] = "SUNDAY";
})(FirstDayOfWeek || (FirstDayOfWeek = {}));
var NumberComponentType;
(function(NumberComponentType2) {
  NumberComponentType2["UNKNOWN"] = "UNKNOWN";
  NumberComponentType2["NUMBER_INPUT"] = "NUMBER_INPUT";
  NumberComponentType2["RATING_INPUT"] = "RATING_INPUT";
})(NumberComponentType || (NumberComponentType = {}));
var BooleanComponentType;
(function(BooleanComponentType2) {
  BooleanComponentType2["UNKNOWN_COMPONENT_TYPE"] = "UNKNOWN_COMPONENT_TYPE";
  BooleanComponentType2["CHECKBOX"] = "CHECKBOX";
})(BooleanComponentType || (BooleanComponentType = {}));
var ItemType;
(function(ItemType2) {
  ItemType2["UNKNOWN_ITEM_TYPE"] = "UNKNOWN_ITEM_TYPE";
  ItemType2["STRING"] = "STRING";
  ItemType2["NUMBER"] = "NUMBER";
  ItemType2["BOOLEAN"] = "BOOLEAN";
  ItemType2["INTEGER"] = "INTEGER";
  ItemType2["OBJECT"] = "OBJECT";
})(ItemType || (ItemType = {}));
var PropertiesTypeEnum;
(function(PropertiesTypeEnum2) {
  PropertiesTypeEnum2["UNKNOWN_PROPERTY_TYPE"] = "UNKNOWN_PROPERTY_TYPE";
  PropertiesTypeEnum2["STRING"] = "STRING";
  PropertiesTypeEnum2["NUMBER"] = "NUMBER";
  PropertiesTypeEnum2["BOOLEAN"] = "BOOLEAN";
  PropertiesTypeEnum2["INTEGER"] = "INTEGER";
  PropertiesTypeEnum2["ARRAY"] = "ARRAY";
})(PropertiesTypeEnum || (PropertiesTypeEnum = {}));
var ArrayComponentType;
(function(ArrayComponentType2) {
  ArrayComponentType2["UNKNOWN_COMPONENT_TYPE"] = "UNKNOWN_COMPONENT_TYPE";
  ArrayComponentType2["CHECKBOX_GROUP"] = "CHECKBOX_GROUP";
})(ArrayComponentType || (ArrayComponentType = {}));
var WixFileComponentType;
(function(WixFileComponentType2) {
  WixFileComponentType2["UNKNOWN_COMPONENT_TYPE"] = "UNKNOWN_COMPONENT_TYPE";
  WixFileComponentType2["FILE_UPLOAD"] = "FILE_UPLOAD";
  WixFileComponentType2["SIGNATURE"] = "SIGNATURE";
})(WixFileComponentType || (WixFileComponentType = {}));
var UploadFileFormat;
(function(UploadFileFormat2) {
  UploadFileFormat2["UNDEFINED"] = "UNDEFINED";
  UploadFileFormat2["VIDEO"] = "VIDEO";
  UploadFileFormat2["IMAGE"] = "IMAGE";
  UploadFileFormat2["AUDIO"] = "AUDIO";
  UploadFileFormat2["DOCUMENT"] = "DOCUMENT";
  UploadFileFormat2["ARCHIVE"] = "ARCHIVE";
})(UploadFileFormat || (UploadFileFormat = {}));
var PaymentComponentType;
(function(PaymentComponentType2) {
  PaymentComponentType2["UNKNOWN_COMPONENT_TYPE"] = "UNKNOWN_COMPONENT_TYPE";
  PaymentComponentType2["CHECKBOX_GROUP"] = "CHECKBOX_GROUP";
  PaymentComponentType2["DONATION_INPUT"] = "DONATION_INPUT";
})(PaymentComponentType || (PaymentComponentType = {}));
var ProductType;
(function(ProductType2) {
  ProductType2["UNKNOWN"] = "UNKNOWN";
  ProductType2["SHIPPABLE"] = "SHIPPABLE";
  ProductType2["DIGITAL"] = "DIGITAL";
})(ProductType || (ProductType = {}));
var PriceType;
(function(PriceType2) {
  PriceType2["UNKNOWN"] = "UNKNOWN";
  PriceType2["FIXED_PRICE"] = "FIXED_PRICE";
  PriceType2["DYNAMIC_PRICE"] = "DYNAMIC_PRICE";
})(PriceType || (PriceType = {}));
var ImageFit;
(function(ImageFit2) {
  ImageFit2["UNKNOWN_IMAGE_FIT"] = "UNKNOWN_IMAGE_FIT";
  ImageFit2["COVER"] = "COVER";
  ImageFit2["CONTAIN"] = "CONTAIN";
})(ImageFit || (ImageFit = {}));
var MultilineAddressComponentType;
(function(MultilineAddressComponentType2) {
  MultilineAddressComponentType2["UNKNOWN_COMPONENT_TYPE"] = "UNKNOWN_COMPONENT_TYPE";
  MultilineAddressComponentType2["MULTILINE_ADDRESS"] = "MULTILINE_ADDRESS";
})(MultilineAddressComponentType || (MultilineAddressComponentType = {}));
var DefaultCountryConfigType;
(function(DefaultCountryConfigType2) {
  DefaultCountryConfigType2["UNKNOWN_DEFAULT_COUNTRY"] = "UNKNOWN_DEFAULT_COUNTRY";
  DefaultCountryConfigType2["BY_IP"] = "BY_IP";
  DefaultCountryConfigType2["COUNTRY"] = "COUNTRY";
})(DefaultCountryConfigType || (DefaultCountryConfigType = {}));
var InputType;
(function(InputType2) {
  InputType2["UNKNOWN_INPUT_TYPE"] = "UNKNOWN_INPUT_TYPE";
  InputType2["STRING"] = "STRING";
  InputType2["NUMBER"] = "NUMBER";
  InputType2["BOOLEAN"] = "BOOLEAN";
  InputType2["ARRAY"] = "ARRAY";
  InputType2["OBJECT"] = "OBJECT";
  InputType2["WIX_FILE"] = "WIX_FILE";
  InputType2["PAYMENT"] = "PAYMENT";
  InputType2["MULTILINE_ADDRESS"] = "MULTILINE_ADDRESS";
})(InputType || (InputType = {}));
var EmailInfoTag;
(function(EmailInfoTag2) {
  EmailInfoTag2["UNTAGGED"] = "UNTAGGED";
  EmailInfoTag2["MAIN"] = "MAIN";
})(EmailInfoTag || (EmailInfoTag = {}));
var PhoneInfoTag;
(function(PhoneInfoTag2) {
  PhoneInfoTag2["UNTAGGED"] = "UNTAGGED";
  PhoneInfoTag2["MAIN"] = "MAIN";
})(PhoneInfoTag || (PhoneInfoTag = {}));
var AddressInfoTag;
(function(AddressInfoTag2) {
  AddressInfoTag2["UNTAGGED"] = "UNTAGGED";
  AddressInfoTag2["HOME"] = "HOME";
})(AddressInfoTag || (AddressInfoTag = {}));
var ConfirmationLevel;
(function(ConfirmationLevel2) {
  ConfirmationLevel2["UNKNOWN_CONFIRMATION_LEVEL"] = "UNKNOWN_CONFIRMATION_LEVEL";
  ConfirmationLevel2["SINGLE_CONFIRMATION"] = "SINGLE_CONFIRMATION";
  ConfirmationLevel2["DOUBLE_CONFIRMATION"] = "DOUBLE_CONFIRMATION";
})(ConfirmationLevel || (ConfirmationLevel = {}));
var ContactField;
(function(ContactField2) {
  ContactField2["UNKNOWN_CONTACT_FIELD"] = "UNKNOWN_CONTACT_FIELD";
  ContactField2["FIRST_NAME"] = "FIRST_NAME";
  ContactField2["LAST_NAME"] = "LAST_NAME";
  ContactField2["COMPANY"] = "COMPANY";
  ContactField2["POSITION"] = "POSITION";
  ContactField2["EMAIL"] = "EMAIL";
  ContactField2["PHONE"] = "PHONE";
  ContactField2["ADDRESS"] = "ADDRESS";
  ContactField2["BIRTHDATE"] = "BIRTHDATE";
  ContactField2["CUSTOM_FIELD"] = "CUSTOM_FIELD";
  ContactField2["SUBSCRIPTION"] = "SUBSCRIPTION";
  ContactField2["VAT_ID"] = "VAT_ID";
})(ContactField || (ContactField = {}));
var DisplayFieldType;
(function(DisplayFieldType2) {
  DisplayFieldType2["UNKNOWN_FIELD_TYPE"] = "UNKNOWN_FIELD_TYPE";
  DisplayFieldType2["RICH_CONTENT"] = "RICH_CONTENT";
  DisplayFieldType2["PAGE_NAVIGATION"] = "PAGE_NAVIGATION";
  DisplayFieldType2["LOGIN_BAR"] = "LOGIN_BAR";
})(DisplayFieldType || (DisplayFieldType = {}));
var ChangeableProperty;
(function(ChangeableProperty2) {
  ChangeableProperty2["UNKNOWN"] = "UNKNOWN";
  ChangeableProperty2["REQUIRED"] = "REQUIRED";
  ChangeableProperty2["HIDDEN"] = "HIDDEN";
  ChangeableProperty2["ALLOWED_VALUES"] = "ALLOWED_VALUES";
})(ChangeableProperty || (ChangeableProperty = {}));
var OverrideEntityType;
(function(OverrideEntityType2) {
  OverrideEntityType2["UNKNOWN"] = "UNKNOWN";
  OverrideEntityType2["FIELD"] = "FIELD";
})(OverrideEntityType || (OverrideEntityType = {}));
var Operator;
(function(Operator2) {
  Operator2["UNKNOWN"] = "UNKNOWN";
  Operator2["EQUAL"] = "EQUAL";
  Operator2["NOT_EQUAL"] = "NOT_EQUAL";
  Operator2["EMPTY"] = "EMPTY";
  Operator2["NOT_EMPTY"] = "NOT_EMPTY";
  Operator2["CONTAINS"] = "CONTAINS";
  Operator2["NOT_CONTAINS"] = "NOT_CONTAINS";
  Operator2["LESS_THAN"] = "LESS_THAN";
  Operator2["LESS_THAN_OR_EQUALS"] = "LESS_THAN_OR_EQUALS";
  Operator2["GREATER_THAN"] = "GREATER_THAN";
  Operator2["GREATER_THAN_OR_EQUALS"] = "GREATER_THAN_OR_EQUALS";
  Operator2["BEFORE"] = "BEFORE";
  Operator2["BEFORE_OR_EQUAL"] = "BEFORE_OR_EQUAL";
  Operator2["AFTER"] = "AFTER";
  Operator2["AFTER_OR_EQUAL"] = "AFTER_OR_EQUAL";
  Operator2["BETWEEN"] = "BETWEEN";
  Operator2["ANY"] = "ANY";
  Operator2["ARRAY_EQUAL"] = "ARRAY_EQUAL";
  Operator2["ARRAY_NOT_EQUAL"] = "ARRAY_NOT_EQUAL";
  Operator2["CHECKED"] = "CHECKED";
  Operator2["NOT_CHECKED"] = "NOT_CHECKED";
  Operator2["IN"] = "IN";
  Operator2["NOT_IN"] = "NOT_IN";
  Operator2["IS_DATE_OLDER_THAN"] = "IS_DATE_OLDER_THAN";
  Operator2["IS_DATE_OLDER_THAN_OR_EQUAL"] = "IS_DATE_OLDER_THAN_OR_EQUAL";
  Operator2["IS_DATE_NEWER_THAN"] = "IS_DATE_NEWER_THAN";
  Operator2["IS_DATE_NEWER_THAN_OR_EQUAL"] = "IS_DATE_NEWER_THAN_OR_EQUAL";
})(Operator || (Operator = {}));
var Target;
(function(Target2) {
  Target2["UNKNOWN_TARGET"] = "UNKNOWN_TARGET";
  Target2["SELF"] = "SELF";
  Target2["BLANK"] = "BLANK";
})(Target || (Target = {}));
var SubmitSuccessAction;
(function(SubmitSuccessAction2) {
  SubmitSuccessAction2["UNKNOWN_SUBMIT_SUCCESS_ACTION"] = "UNKNOWN_SUBMIT_SUCCESS_ACTION";
  SubmitSuccessAction2["NO_ACTION"] = "NO_ACTION";
  SubmitSuccessAction2["THANK_YOU_MESSAGE"] = "THANK_YOU_MESSAGE";
  SubmitSuccessAction2["REDIRECT"] = "REDIRECT";
})(SubmitSuccessAction || (SubmitSuccessAction = {}));
var TemplateType;
(function(TemplateType2) {
  TemplateType2["UNKNOWN"] = "UNKNOWN";
  TemplateType2["EXTENDABLE"] = "EXTENDABLE";
  TemplateType2["DEMO"] = "DEMO";
  TemplateType2["FORM_INPUT"] = "FORM_INPUT";
})(TemplateType || (TemplateType = {}));
var CustomFieldsType;
(function(CustomFieldsType2) {
  CustomFieldsType2["ANY"] = "ANY";
})(CustomFieldsType || (CustomFieldsType = {}));
var EditableProperties;
(function(EditableProperties2) {
  EditableProperties2["UNKNOWN"] = "UNKNOWN";
  EditableProperties2["REQUIRED"] = "REQUIRED";
  EditableProperties2["HIDDEN"] = "HIDDEN";
  EditableProperties2["LABEL"] = "LABEL";
  EditableProperties2["PLACEHOLDER"] = "PLACEHOLDER";
  EditableProperties2["DESCRIPTION"] = "DESCRIPTION";
  EditableProperties2["INPUT_VALUE_LIMITS"] = "INPUT_VALUE_LIMITS";
  EditableProperties2["DEFAULT_VALUE"] = "DEFAULT_VALUE";
})(EditableProperties || (EditableProperties = {}));
var RequiredIndicator;
(function(RequiredIndicator2) {
  RequiredIndicator2["UNKNOWN_INDICATOR"] = "UNKNOWN_INDICATOR";
  RequiredIndicator2["ASTERISK"] = "ASTERISK";
  RequiredIndicator2["TEXT"] = "TEXT";
  RequiredIndicator2["NONE"] = "NONE";
})(RequiredIndicator || (RequiredIndicator = {}));
var RequiredIndicatorPlacement;
(function(RequiredIndicatorPlacement2) {
  RequiredIndicatorPlacement2["UNKNOWN_PLACEMENT"] = "UNKNOWN_PLACEMENT";
  RequiredIndicatorPlacement2["AFTER_FIELD_TITLE"] = "AFTER_FIELD_TITLE";
  RequiredIndicatorPlacement2["BEFORE_FIELD_TITLE"] = "BEFORE_FIELD_TITLE";
})(RequiredIndicatorPlacement || (RequiredIndicatorPlacement = {}));
var WixCodePublishTaskName;
(function(WixCodePublishTaskName2) {
  WixCodePublishTaskName2["UNKNOWN"] = "UNKNOWN";
  WixCodePublishTaskName2["USER_CODE_BUNDLE"] = "USER_CODE_BUNDLE";
  WixCodePublishTaskName2["EXPORTED_FUNCTIONS_ANALYSIS"] = "EXPORTED_FUNCTIONS_ANALYSIS";
  WixCodePublishTaskName2["IMPORTED_NAMESPACES_ANALYSIS"] = "IMPORTED_NAMESPACES_ANALYSIS";
  WixCodePublishTaskName2["PAGE_DETAILS_ANALYSIS"] = "PAGE_DETAILS_ANALYSIS";
  WixCodePublishTaskName2["USER_CODE_VALIDATION"] = "USER_CODE_VALIDATION";
  WixCodePublishTaskName2["CODE_PACKAGES_REGISTRATION"] = "CODE_PACKAGES_REGISTRATION";
})(WixCodePublishTaskName || (WixCodePublishTaskName = {}));
var ChannelConfigurationChannelType;
(function(ChannelConfigurationChannelType2) {
  ChannelConfigurationChannelType2["UNKNOWN_CHANNEL_TYPE"] = "UNKNOWN_CHANNEL_TYPE";
  ChannelConfigurationChannelType2["DIRECT_MESSAGING"] = "DIRECT_MESSAGING";
  ChannelConfigurationChannelType2["EMAIL"] = "EMAIL";
  ChannelConfigurationChannelType2["SMS"] = "SMS";
})(ChannelConfigurationChannelType || (ChannelConfigurationChannelType = {}));
var AcceptedMessageTypesAcceptedDirectMessageType;
(function(AcceptedMessageTypesAcceptedDirectMessageType2) {
  AcceptedMessageTypesAcceptedDirectMessageType2["UNKNOWN_DIRECT_MESSAGE_TYPE"] = "UNKNOWN_DIRECT_MESSAGE_TYPE";
  AcceptedMessageTypesAcceptedDirectMessageType2["MEDIA"] = "MEDIA";
  AcceptedMessageTypesAcceptedDirectMessageType2["CARD"] = "CARD";
  AcceptedMessageTypesAcceptedDirectMessageType2["MINIMAL"] = "MINIMAL";
  AcceptedMessageTypesAcceptedDirectMessageType2["FORM"] = "FORM";
  AcceptedMessageTypesAcceptedDirectMessageType2["SYSTEM"] = "SYSTEM";
})(AcceptedMessageTypesAcceptedDirectMessageType || (AcceptedMessageTypesAcceptedDirectMessageType = {}));
var AcceptedMessageTypesAcceptedSmsMessageType;
(function(AcceptedMessageTypesAcceptedSmsMessageType2) {
  AcceptedMessageTypesAcceptedSmsMessageType2["UNKNOWN_SMS_MESSAGE_TYPE"] = "UNKNOWN_SMS_MESSAGE_TYPE";
  AcceptedMessageTypesAcceptedSmsMessageType2["SMS"] = "SMS";
  AcceptedMessageTypesAcceptedSmsMessageType2["MMS"] = "MMS";
})(AcceptedMessageTypesAcceptedSmsMessageType || (AcceptedMessageTypesAcceptedSmsMessageType = {}));
var FactorType;
(function(FactorType2) {
  FactorType2["UNKNOWN_FACTOR_TYPE"] = "UNKNOWN_FACTOR_TYPE";
  FactorType2["PASSWORD"] = "PASSWORD";
  FactorType2["SMS"] = "SMS";
  FactorType2["CALL"] = "CALL";
  FactorType2["EMAIL"] = "EMAIL";
  FactorType2["TOTP"] = "TOTP";
  FactorType2["PUSH"] = "PUSH";
  FactorType2["WEBAUTHN"] = "WEBAUTHN";
})(FactorType || (FactorType = {}));
var MonitoringType;
(function(MonitoringType2) {
  MonitoringType2["UNKNOWN_PROVIDER"] = "UNKNOWN_PROVIDER";
  MonitoringType2["SENTRY"] = "SENTRY";
  MonitoringType2["PANORAMA"] = "PANORAMA";
})(MonitoringType || (MonitoringType = {}));
var Escalation;
(function(Escalation2) {
  Escalation2["BREAK_BUILD"] = "BREAK_BUILD";
  Escalation2["BLOCK_ROLLOUT"] = "BLOCK_ROLLOUT";
  Escalation2["BLOCK_FUTURE_ROLLOUT"] = "BLOCK_FUTURE_ROLLOUT";
  Escalation2["OPEN_BUG_SLA"] = "OPEN_BUG_SLA";
  Escalation2["VISIBILITY"] = "VISIBILITY";
})(Escalation || (Escalation = {}));
var ImportanceLevel;
(function(ImportanceLevel2) {
  ImportanceLevel2["LOW"] = "LOW";
  ImportanceLevel2["MEDIUM"] = "MEDIUM";
  ImportanceLevel2["HIGH"] = "HIGH";
})(ImportanceLevel || (ImportanceLevel = {}));
var PanelSizeWidth;
(function(PanelSizeWidth2) {
  PanelSizeWidth2["UNKNOWN_WIDTH"] = "UNKNOWN_WIDTH";
  PanelSizeWidth2["SMALL"] = "SMALL";
  PanelSizeWidth2["MEDIUM"] = "MEDIUM";
  PanelSizeWidth2["LARGE"] = "LARGE";
})(PanelSizeWidth || (PanelSizeWidth = {}));
var PanelContentType;
(function(PanelContentType2) {
  PanelContentType2["UNKNOWN_PanelContentType"] = "UNKNOWN_PanelContentType";
  PanelContentType2["url"] = "url";
  PanelContentType2["code"] = "code";
})(PanelContentType || (PanelContentType = {}));
var StorageType;
(function(StorageType2) {
  StorageType2["UNKNOWN_STORAGE_TYPE"] = "UNKNOWN_STORAGE_TYPE";
  StorageType2["COOKIE"] = "COOKIE";
  StorageType2["LOCALSTORAGE"] = "LOCALSTORAGE";
  StorageType2["SESSIONSTORAGE"] = "SESSIONSTORAGE";
  StorageType2["INDEXDB"] = "INDEXDB";
})(StorageType || (StorageType = {}));
var ConsentCategory;
(function(ConsentCategory2) {
  ConsentCategory2["UNKNOWN_CATEGORY"] = "UNKNOWN_CATEGORY";
  ConsentCategory2["essential"] = "essential";
  ConsentCategory2["functional"] = "functional";
  ConsentCategory2["analytics"] = "analytics";
  ConsentCategory2["advertising"] = "advertising";
})(ConsentCategory || (ConsentCategory = {}));
var StorageDomain;
(function(StorageDomain2) {
  StorageDomain2["UNKNOWN_STORAGE_DOMAIN"] = "UNKNOWN_STORAGE_DOMAIN";
  StorageDomain2["SITE_DOMAIN"] = "SITE_DOMAIN";
  StorageDomain2["WIX_DOMAIN"] = "WIX_DOMAIN";
  StorageDomain2["APP_DOMAIN"] = "APP_DOMAIN";
})(StorageDomain || (StorageDomain = {}));
var Method;
(function(Method2) {
  Method2["CREATE"] = "CREATE";
  Method2["UPDATE"] = "UPDATE";
  Method2["CANCEL"] = "CANCEL";
})(Method || (Method = {}));
var ParticipantType;
(function(ParticipantType2) {
  ParticipantType2["UNKNOWN_TYPE"] = "UNKNOWN_TYPE";
  ParticipantType2["WIX_USER"] = "WIX_USER";
  ParticipantType2["CONTACT"] = "CONTACT";
  ParticipantType2["ANONYMOUS"] = "ANONYMOUS";
})(ParticipantType || (ParticipantType = {}));
var InterfaceConfigurationType;
(function(InterfaceConfigurationType2) {
  InterfaceConfigurationType2["UNKNOWN_TYPE"] = "UNKNOWN_TYPE";
  InterfaceConfigurationType2["WIDGET_COMPONENT"] = "WIDGET_COMPONENT";
  InterfaceConfigurationType2["ITEM_SELECTION"] = "ITEM_SELECTION";
})(InterfaceConfigurationType || (InterfaceConfigurationType = {}));
var Scope;
(function(Scope2) {
  Scope2["UNKNOWN_SCOPE"] = "UNKNOWN_SCOPE";
  Scope2["SPECIFIC_SITE"] = "SPECIFIC_SITE";
  Scope2["NON_SPECIFIC_SITE"] = "NON_SPECIFIC_SITE";
})(Scope || (Scope = {}));
var CssPropertyType;
(function(CssPropertyType2) {
  CssPropertyType2["UNKNOWN_CssPropertyType"] = "UNKNOWN_CssPropertyType";
  CssPropertyType2["background"] = "background";
  CssPropertyType2["backgroundSize"] = "backgroundSize";
  CssPropertyType2["backgroundColor"] = "backgroundColor";
  CssPropertyType2["backgroundImage"] = "backgroundImage";
  CssPropertyType2["backgroundClip"] = "backgroundClip";
  CssPropertyType2["backgroundOrigin"] = "backgroundOrigin";
  CssPropertyType2["backgroundPosition"] = "backgroundPosition";
  CssPropertyType2["backgroundRepeat"] = "backgroundRepeat";
  CssPropertyType2["backgroundAttachment"] = "backgroundAttachment";
  CssPropertyType2["fill"] = "fill";
  CssPropertyType2["fillOpacity"] = "fillOpacity";
  CssPropertyType2["margin"] = "margin";
  CssPropertyType2["marginTop"] = "marginTop";
  CssPropertyType2["marginRight"] = "marginRight";
  CssPropertyType2["marginBottom"] = "marginBottom";
  CssPropertyType2["marginLeft"] = "marginLeft";
  CssPropertyType2["marginInlineStart"] = "marginInlineStart";
  CssPropertyType2["marginInlineEnd"] = "marginInlineEnd";
  CssPropertyType2["padding"] = "padding";
  CssPropertyType2["paddingTop"] = "paddingTop";
  CssPropertyType2["paddingRight"] = "paddingRight";
  CssPropertyType2["paddingBottom"] = "paddingBottom";
  CssPropertyType2["paddingLeft"] = "paddingLeft";
  CssPropertyType2["paddingInlineStart"] = "paddingInlineStart";
  CssPropertyType2["paddingInlineEnd"] = "paddingInlineEnd";
  CssPropertyType2["border"] = "border";
  CssPropertyType2["borderWidth"] = "borderWidth";
  CssPropertyType2["borderStyle"] = "borderStyle";
  CssPropertyType2["borderColor"] = "borderColor";
  CssPropertyType2["borderTop"] = "borderTop";
  CssPropertyType2["borderTopColor"] = "borderTopColor";
  CssPropertyType2["borderTopWidth"] = "borderTopWidth";
  CssPropertyType2["borderTopStyle"] = "borderTopStyle";
  CssPropertyType2["borderRight"] = "borderRight";
  CssPropertyType2["borderRightColor"] = "borderRightColor";
  CssPropertyType2["borderRightWidth"] = "borderRightWidth";
  CssPropertyType2["borderRightStyle"] = "borderRightStyle";
  CssPropertyType2["borderBottom"] = "borderBottom";
  CssPropertyType2["borderBottomColor"] = "borderBottomColor";
  CssPropertyType2["borderBottomWidth"] = "borderBottomWidth";
  CssPropertyType2["borderBottomStyle"] = "borderBottomStyle";
  CssPropertyType2["borderLeft"] = "borderLeft";
  CssPropertyType2["borderLeftColor"] = "borderLeftColor";
  CssPropertyType2["borderLeftWidth"] = "borderLeftWidth";
  CssPropertyType2["borderLeftStyle"] = "borderLeftStyle";
  CssPropertyType2["borderInlineStart"] = "borderInlineStart";
  CssPropertyType2["borderInlineStartColor"] = "borderInlineStartColor";
  CssPropertyType2["borderInlineStartWidth"] = "borderInlineStartWidth";
  CssPropertyType2["borderInlineStartStyle"] = "borderInlineStartStyle";
  CssPropertyType2["borderInlineEnd"] = "borderInlineEnd";
  CssPropertyType2["borderInlineEndColor"] = "borderInlineEndColor";
  CssPropertyType2["borderInlineEndWidth"] = "borderInlineEndWidth";
  CssPropertyType2["borderInlineEndStyle"] = "borderInlineEndStyle";
  CssPropertyType2["borderRadius"] = "borderRadius";
  CssPropertyType2["borderTopLeftRadius"] = "borderTopLeftRadius";
  CssPropertyType2["borderTopRightRadius"] = "borderTopRightRadius";
  CssPropertyType2["borderBottomRightRadius"] = "borderBottomRightRadius";
  CssPropertyType2["borderBottomLeftRadius"] = "borderBottomLeftRadius";
  CssPropertyType2["borderStartStartRadius"] = "borderStartStartRadius";
  CssPropertyType2["borderStartEndRadius"] = "borderStartEndRadius";
  CssPropertyType2["borderEndStartRadius"] = "borderEndStartRadius";
  CssPropertyType2["borderEndEndRadius"] = "borderEndEndRadius";
  CssPropertyType2["font"] = "font";
  CssPropertyType2["fontFamily"] = "fontFamily";
  CssPropertyType2["fontSize"] = "fontSize";
  CssPropertyType2["fontStretch"] = "fontStretch";
  CssPropertyType2["fontStyle"] = "fontStyle";
  CssPropertyType2["fontVariant"] = "fontVariant";
  CssPropertyType2["fontWeight"] = "fontWeight";
  CssPropertyType2["lineHeight"] = "lineHeight";
  CssPropertyType2["color"] = "color";
  CssPropertyType2["letterSpacing"] = "letterSpacing";
  CssPropertyType2["writingMode"] = "writingMode";
  CssPropertyType2["textAlign"] = "textAlign";
  CssPropertyType2["textTransform"] = "textTransform";
  CssPropertyType2["textShadow"] = "textShadow";
  CssPropertyType2["textOverflow"] = "textOverflow";
  CssPropertyType2["textIndent"] = "textIndent";
  CssPropertyType2["textDecoration"] = "textDecoration";
  CssPropertyType2["textDecorationColor"] = "textDecorationColor";
  CssPropertyType2["textDecorationLine"] = "textDecorationLine";
  CssPropertyType2["textDecorationStyle"] = "textDecorationStyle";
  CssPropertyType2["textDecorationThickness"] = "textDecorationThickness";
  CssPropertyType2["boxShadow"] = "boxShadow";
  CssPropertyType2["opacity"] = "opacity";
  CssPropertyType2["overflow"] = "overflow";
  CssPropertyType2["display"] = "display";
  CssPropertyType2["alignSelf"] = "alignSelf";
  CssPropertyType2["justifyContent"] = "justifyContent";
  CssPropertyType2["alignItems"] = "alignItems";
  CssPropertyType2["flexDirection"] = "flexDirection";
  CssPropertyType2["gap"] = "gap";
  CssPropertyType2["height"] = "height";
  CssPropertyType2["width"] = "width";
  CssPropertyType2["columnGap"] = "columnGap";
  CssPropertyType2["rowGap"] = "rowGap";
  CssPropertyType2["filter"] = "filter";
  CssPropertyType2["backdropFilter"] = "backdropFilter";
  CssPropertyType2["objectFit"] = "objectFit";
  CssPropertyType2["objectPosition"] = "objectPosition";
  CssPropertyType2["mixBlendMode"] = "mixBlendMode";
  CssPropertyType2["isolation"] = "isolation";
  CssPropertyType2["stroke"] = "stroke";
  CssPropertyType2["strokeWidth"] = "strokeWidth";
  CssPropertyType2["strokeOpacity"] = "strokeOpacity";
})(CssPropertyType || (CssPropertyType = {}));
var CssDataType;
(function(CssDataType2) {
  CssDataType2["UNKNOWN_CssDataType"] = "UNKNOWN_CssDataType";
  CssDataType2["number"] = "number";
  CssDataType2["length"] = "length";
  CssDataType2["color"] = "color";
  CssDataType2["angle"] = "angle";
  CssDataType2["percentage"] = "percentage";
  CssDataType2["lengthPercentage"] = "lengthPercentage";
  CssDataType2["blendMode"] = "blendMode";
  CssDataType2["customEnum"] = "customEnum";
  CssDataType2["string"] = "string";
  CssDataType2["time"] = "time";
})(CssDataType || (CssDataType = {}));
var FilterFunction;
(function(FilterFunction2) {
  FilterFunction2["UNKNOWN_FilterFunctions"] = "UNKNOWN_FilterFunctions";
  FilterFunction2["blur"] = "blur";
  FilterFunction2["brightness"] = "brightness";
  FilterFunction2["contrast"] = "contrast";
  FilterFunction2["drop_shadow"] = "drop_shadow";
  FilterFunction2["grayscale"] = "grayscale";
  FilterFunction2["hue_rotate"] = "hue_rotate";
  FilterFunction2["invert"] = "invert";
  FilterFunction2["opacity"] = "opacity";
  FilterFunction2["sepia"] = "sepia";
  FilterFunction2["saturate"] = "saturate";
})(FilterFunction || (FilterFunction = {}));
var DisplayValueEnumDisplayValue;
(function(DisplayValueEnumDisplayValue2) {
  DisplayValueEnumDisplayValue2["UNKNOWN_DisplayValue"] = "UNKNOWN_DisplayValue";
  DisplayValueEnumDisplayValue2["none"] = "none";
  DisplayValueEnumDisplayValue2["block"] = "block";
  DisplayValueEnumDisplayValue2["inline"] = "inline";
  DisplayValueEnumDisplayValue2["flow"] = "flow";
  DisplayValueEnumDisplayValue2["flowRoot"] = "flowRoot";
  DisplayValueEnumDisplayValue2["table"] = "table";
  DisplayValueEnumDisplayValue2["flex"] = "flex";
  DisplayValueEnumDisplayValue2["grid"] = "grid";
  DisplayValueEnumDisplayValue2["list_item"] = "list_item";
  DisplayValueEnumDisplayValue2["contents"] = "contents";
  DisplayValueEnumDisplayValue2["inline_block"] = "inline_block";
  DisplayValueEnumDisplayValue2["inline_table"] = "inline_table";
  DisplayValueEnumDisplayValue2["inline_flex"] = "inline_flex";
  DisplayValueEnumDisplayValue2["inline_grid"] = "inline_grid";
})(DisplayValueEnumDisplayValue || (DisplayValueEnumDisplayValue = {}));
var WritingModeValue;
(function(WritingModeValue2) {
  WritingModeValue2["UNKNOWN_WritingModeValue"] = "UNKNOWN_WritingModeValue";
  WritingModeValue2["horizontalTb"] = "horizontalTb";
  WritingModeValue2["verticalRl"] = "verticalRl";
  WritingModeValue2["verticalLr"] = "verticalLr";
  WritingModeValue2["sidewaysRl"] = "sidewaysRl";
  WritingModeValue2["sidewaysLr"] = "sidewaysLr";
})(WritingModeValue || (WritingModeValue = {}));
var BackgroundModeEnum;
(function(BackgroundModeEnum2) {
  BackgroundModeEnum2["UNKNOWN_BackgroundModeEnum"] = "UNKNOWN_BackgroundModeEnum";
  BackgroundModeEnum2["shapeDividerSvg"] = "shapeDividerSvg";
})(BackgroundModeEnum || (BackgroundModeEnum = {}));
var DataType;
(function(DataType2) {
  DataType2["UNKNOWN_DataType"] = "UNKNOWN_DataType";
  DataType2["text"] = "text";
  DataType2["textEnum"] = "textEnum";
  DataType2["number"] = "number";
  DataType2["booleanValue"] = "booleanValue";
  DataType2["a11y"] = "a11y";
  DataType2["link"] = "link";
  DataType2["image"] = "image";
  DataType2["video"] = "video";
  DataType2["vectorArt"] = "vectorArt";
  DataType2["audio"] = "audio";
  DataType2["schema"] = "schema";
  DataType2["localDate"] = "localDate";
  DataType2["localTime"] = "localTime";
  DataType2["localDateTime"] = "localDateTime";
  DataType2["webUrl"] = "webUrl";
  DataType2["email"] = "email";
  DataType2["phone"] = "phone";
  DataType2["hostname"] = "hostname";
  DataType2["regex"] = "regex";
  DataType2["guid"] = "guid";
  DataType2["richText"] = "richText";
  DataType2["container"] = "container";
  DataType2["arrayItems"] = "arrayItems";
  DataType2["direction"] = "direction";
  DataType2["menuItems"] = "menuItems";
})(DataType || (DataType = {}));
var A11yAttributes;
(function(A11yAttributes2) {
  A11yAttributes2["Unknown_AriaAttributes"] = "Unknown_AriaAttributes";
  A11yAttributes2["tabIndex"] = "tabIndex";
  A11yAttributes2["ariaLevel"] = "ariaLevel";
  A11yAttributes2["ariaExpanded"] = "ariaExpanded";
  A11yAttributes2["ariaDisabled"] = "ariaDisabled";
  A11yAttributes2["ariaAtomic"] = "ariaAtomic";
  A11yAttributes2["ariaHidden"] = "ariaHidden";
  A11yAttributes2["ariaBusy"] = "ariaBusy";
  A11yAttributes2["multiline"] = "multiline";
  A11yAttributes2["ariaAutocomplete"] = "ariaAutocomplete";
  A11yAttributes2["ariaPressed"] = "ariaPressed";
  A11yAttributes2["ariaHaspopup"] = "ariaHaspopup";
  A11yAttributes2["ariaRelevant"] = "ariaRelevant";
  A11yAttributes2["role"] = "role";
  A11yAttributes2["ariaLive"] = "ariaLive";
  A11yAttributes2["ariaCurrent"] = "ariaCurrent";
  A11yAttributes2["ariaLabel"] = "ariaLabel";
  A11yAttributes2["ariaRoledescription"] = "ariaRoledescription";
  A11yAttributes2["ariaDescribedby"] = "ariaDescribedby";
  A11yAttributes2["ariaLabelledby"] = "ariaLabelledby";
  A11yAttributes2["ariaErrormessage"] = "ariaErrormessage";
  A11yAttributes2["ariaOwns"] = "ariaOwns";
  A11yAttributes2["ariaControls"] = "ariaControls";
  A11yAttributes2["tag"] = "tag";
  A11yAttributes2["ariaMultiline"] = "ariaMultiline";
  A11yAttributes2["ariaInvalid"] = "ariaInvalid";
})(A11yAttributes || (A11yAttributes = {}));
var LinkType;
(function(LinkType2) {
  LinkType2["UNKNOWN_LinkType"] = "UNKNOWN_LinkType";
  LinkType2["externalLink"] = "externalLink";
  LinkType2["anchorLink"] = "anchorLink";
  LinkType2["emailLink"] = "emailLink";
  LinkType2["phoneLink"] = "phoneLink";
  LinkType2["dynamicPageLink"] = "dynamicPageLink";
  LinkType2["pageLink"] = "pageLink";
  LinkType2["whatsAppLink"] = "whatsAppLink";
  LinkType2["documentLink"] = "documentLink";
  LinkType2["popupLink"] = "popupLink";
  LinkType2["addressLink"] = "addressLink";
  LinkType2["edgeAnchorLinks"] = "edgeAnchorLinks";
  LinkType2["loginToWixLink"] = "loginToWixLink";
})(LinkType || (LinkType = {}));
var ContainerType;
(function(ContainerType2) {
  ContainerType2["UNKNOWN_CONTAINER_TYPE"] = "UNKNOWN_CONTAINER_TYPE";
  ContainerType2["simple"] = "simple";
  ContainerType2["slot"] = "slot";
  ContainerType2["placeholder"] = "placeholder";
  ContainerType2["template"] = "template";
})(ContainerType || (ContainerType = {}));
var ResizeDirection;
(function(ResizeDirection2) {
  ResizeDirection2["UNKNOWN_ResizeDirection"] = "UNKNOWN_ResizeDirection";
  ResizeDirection2["horizontal"] = "horizontal";
  ResizeDirection2["vertical"] = "vertical";
  ResizeDirection2["horizontalAndVertical"] = "horizontalAndVertical";
  ResizeDirection2["aspectRatio"] = "aspectRatio";
  ResizeDirection2["none"] = "none";
})(ResizeDirection || (ResizeDirection = {}));
var ElementTypeEnumElementType;
(function(ElementTypeEnumElementType2) {
  ElementTypeEnumElementType2["UNKNOWN_ElementType"] = "UNKNOWN_ElementType";
  ElementTypeEnumElementType2["inlineElement"] = "inlineElement";
  ElementTypeEnumElementType2["refElement"] = "refElement";
})(ElementTypeEnumElementType || (ElementTypeEnumElementType = {}));
var ActionType;
(function(ActionType2) {
  ActionType2["UNKNOWN_ActionType"] = "UNKNOWN_ActionType";
  ActionType2["data"] = "data";
  ActionType2["event"] = "event";
  ActionType2["panel"] = "panel";
  ActionType2["forward"] = "forward";
  ActionType2["style"] = "style";
})(ActionType || (ActionType = {}));
var PanelType;
(function(PanelType2) {
  PanelType2["UNKNOWN_PanelType"] = "UNKNOWN_PanelType";
  PanelType2["panelId"] = "panelId";
  PanelType2["inline"] = "inline";
})(PanelType || (PanelType = {}));
var ActionName;
(function(ActionName2) {
  ActionName2["UNKNOWN_ActionName"] = "UNKNOWN_ActionName";
  ActionName2["settings"] = "settings";
  ActionName2["design"] = "design";
  ActionName2["media"] = "media";
  ActionName2["manageItems"] = "manageItems";
  ActionName2["dashboard"] = "dashboard";
  ActionName2["custom"] = "custom";
})(ActionName || (ActionName = {}));
var SizingType;
(function(SizingType2) {
  SizingType2["UNKNOWN_SizingType"] = "UNKNOWN_SizingType";
  SizingType2["content"] = "content";
  SizingType2["stretched"] = "stretched";
  SizingType2["pixels"] = "pixels";
})(SizingType || (SizingType = {}));
var Archetype;
(function(Archetype2) {
  Archetype2["UNKNOWN_Archetype"] = "UNKNOWN_Archetype";
  Archetype2["Button"] = "Button";
  Archetype2["LoginButton"] = "LoginButton";
  Archetype2["Image"] = "Image";
  Archetype2["Gallery"] = "Gallery";
  Archetype2["Video"] = "Video";
  Archetype2["Audio"] = "Audio";
  Archetype2["Text"] = "Text";
  Archetype2["TextInput"] = "TextInput";
  Archetype2["RichTextEditor"] = "RichTextEditor";
  Archetype2["SignatureInput"] = "SignatureInput";
  Archetype2["Checkbox"] = "Checkbox";
  Archetype2["RadioGroup"] = "RadioGroup";
  Archetype2["Switch"] = "Switch";
  Archetype2["Dropdown"] = "Dropdown";
  Archetype2["DatePicker"] = "DatePicker";
  Archetype2["TimePicker"] = "TimePicker";
  Archetype2["Ratings"] = "Ratings";
  Archetype2["RatingInput"] = "RatingInput";
  Archetype2["Menu"] = "Menu";
  Archetype2["Pagination"] = "Pagination";
  Archetype2["Slider"] = "Slider";
  Archetype2["Container"] = "Container";
  Archetype2["Carousel"] = "Carousel";
  Archetype2["Accordion"] = "Accordion";
  Archetype2["Tabs"] = "Tabs";
  Archetype2["ProgressBar"] = "ProgressBar";
  Archetype2["Upload"] = "Upload";
  Archetype2["Social"] = "Social";
  Archetype2["Breadcrumbs"] = "Breadcrumbs";
  Archetype2["SearchBox"] = "SearchBox";
  Archetype2["Map"] = "Map";
  Archetype2["Line"] = "Line";
  Archetype2["Logo"] = "Logo";
  Archetype2["Avatar"] = "Avatar";
  Archetype2["Captcha"] = "Captcha";
  Archetype2["VectorArt"] = "VectorArt";
  Archetype2["AnimatedGraphic"] = "AnimatedGraphic";
})(Archetype || (Archetype = {}));
var NativeStateType;
(function(NativeStateType2) {
  NativeStateType2["UNKNOWN_NativeStateType"] = "UNKNOWN_NativeStateType";
  NativeStateType2["hover"] = "hover";
  NativeStateType2["focus"] = "focus";
  NativeStateType2["disabled"] = "disabled";
  NativeStateType2["invalid"] = "invalid";
})(NativeStateType || (NativeStateType = {}));
var ContentResizeDirection;
(function(ContentResizeDirection2) {
  ContentResizeDirection2["UNKNOWN_ContentResizeDirection"] = "UNKNOWN_ContentResizeDirection";
  ContentResizeDirection2["horizontal"] = "horizontal";
  ContentResizeDirection2["vertical"] = "vertical";
  ContentResizeDirection2["horizontalAndVertical"] = "horizontalAndVertical";
  ContentResizeDirection2["none"] = "none";
})(ContentResizeDirection || (ContentResizeDirection = {}));
var InteractionsTrigger;
(function(InteractionsTrigger2) {
  InteractionsTrigger2["UNKNOWN_TRIGGER"] = "UNKNOWN_TRIGGER";
  InteractionsTrigger2["hover"] = "hover";
  InteractionsTrigger2["click"] = "click";
  InteractionsTrigger2["viewEnter"] = "viewEnter";
  InteractionsTrigger2["pageVisible"] = "pageVisible";
  InteractionsTrigger2["viewProgress"] = "viewProgress";
  InteractionsTrigger2["pointerMove"] = "pointerMove";
  InteractionsTrigger2["animationEnd"] = "animationEnd";
})(InteractionsTrigger || (InteractionsTrigger = {}));
var RestrictionLevel;
(function(RestrictionLevel2) {
  RestrictionLevel2["UNKNOWN_RESTRICTION_TYPE"] = "UNKNOWN_RESTRICTION_TYPE";
  RestrictionLevel2["WARNING"] = "WARNING";
  RestrictionLevel2["LOCKED"] = "LOCKED";
})(RestrictionLevel || (RestrictionLevel = {}));
var ElementDisplayOption;
(function(ElementDisplayOption2) {
  ElementDisplayOption2["UNKNOWN_OPTION"] = "UNKNOWN_OPTION";
  ElementDisplayOption2["REMOVE"] = "REMOVE";
  ElementDisplayOption2["CUSTOMIZED"] = "CUSTOMIZED";
})(ElementDisplayOption || (ElementDisplayOption = {}));
var SidebarDataType;
(function(SidebarDataType2) {
  SidebarDataType2["UNKNOWN_TYPE"] = "UNKNOWN_TYPE";
  SidebarDataType2["CATEGORY"] = "CATEGORY";
  SidebarDataType2["PAGE"] = "PAGE";
  SidebarDataType2["SEPARATOR"] = "SEPARATOR";
})(SidebarDataType || (SidebarDataType = {}));
var SidebarEntityType;
(function(SidebarEntityType2) {
  SidebarEntityType2["UNKNOWN_TYPE"] = "UNKNOWN_TYPE";
  SidebarEntityType2["CATEGORY"] = "CATEGORY";
  SidebarEntityType2["PAGE"] = "PAGE";
  SidebarEntityType2["APP"] = "APP";
})(SidebarEntityType || (SidebarEntityType = {}));
var OperatorEnumOperator;
(function(OperatorEnumOperator2) {
  OperatorEnumOperator2["UNKNOWN_OPERATOR"] = "UNKNOWN_OPERATOR";
  OperatorEnumOperator2["EQ"] = "EQ";
  OperatorEnumOperator2["NEQ"] = "NEQ";
  OperatorEnumOperator2["GT"] = "GT";
  OperatorEnumOperator2["GTE"] = "GTE";
  OperatorEnumOperator2["LT"] = "LT";
  OperatorEnumOperator2["LTE"] = "LTE";
  OperatorEnumOperator2["BETWEEN"] = "BETWEEN";
  OperatorEnumOperator2["IN"] = "IN";
  OperatorEnumOperator2["NIN"] = "NIN";
  OperatorEnumOperator2["EMPTY"] = "EMPTY";
  OperatorEnumOperator2["NOT_EMPTY"] = "NOT_EMPTY";
  OperatorEnumOperator2["STARTS_WITH"] = "STARTS_WITH";
  OperatorEnumOperator2["HAS_ALL"] = "HAS_ALL";
  OperatorEnumOperator2["HAS_SOME"] = "HAS_SOME";
  OperatorEnumOperator2["ANY"] = "ANY";
})(OperatorEnumOperator || (OperatorEnumOperator = {}));
var ParameterType;
(function(ParameterType2) {
  ParameterType2["UNKNOWN_PARAMETER_TYPE"] = "UNKNOWN_PARAMETER_TYPE";
  ParameterType2["STRING"] = "STRING";
  ParameterType2["INT"] = "INT";
  ParameterType2["DECIMAL"] = "DECIMAL";
  ParameterType2["BOOL"] = "BOOL";
  ParameterType2["CURRENCY"] = "CURRENCY";
  ParameterType2["COUNTRY_SUBDIVISION"] = "COUNTRY_SUBDIVISION";
  ParameterType2["LANGUAGE"] = "LANGUAGE";
})(ParameterType || (ParameterType = {}));
var FilterValueDefinitionMode;
(function(FilterValueDefinitionMode2) {
  FilterValueDefinitionMode2["UNKNOWN_FILTER_VALUE_DEFINITION_MODE"] = "UNKNOWN_FILTER_VALUE_DEFINITION_MODE";
  FilterValueDefinitionMode2["ANY_VALUE_OF_SPECIFIED_TYPE"] = "ANY_VALUE_OF_SPECIFIED_TYPE";
  FilterValueDefinitionMode2["ITEM_SELECTION"] = "ITEM_SELECTION";
  FilterValueDefinitionMode2["STRING_LIST"] = "STRING_LIST";
  FilterValueDefinitionMode2["INT_LIST"] = "INT_LIST";
  FilterValueDefinitionMode2["DECIMAL_LIST"] = "DECIMAL_LIST";
  FilterValueDefinitionMode2["BOOL_LIST"] = "BOOL_LIST";
  FilterValueDefinitionMode2["CURRENCY_LIST"] = "CURRENCY_LIST";
  FilterValueDefinitionMode2["COUNTRY_SUBDIVISION_LIST"] = "COUNTRY_SUBDIVISION_LIST";
  FilterValueDefinitionMode2["LANGUAGE_TAG_LIST"] = "LANGUAGE_TAG_LIST";
})(FilterValueDefinitionMode || (FilterValueDefinitionMode = {}));
var TimeConstraintType;
(function(TimeConstraintType2) {
  TimeConstraintType2["UNKNOWN_TIME_CONSTRAINT_TYPE"] = "UNKNOWN_TIME_CONSTRAINT_TYPE";
  TimeConstraintType2["LAST_DAYS"] = "LAST_DAYS";
  TimeConstraintType2["LAST_HOURS"] = "LAST_HOURS";
  TimeConstraintType2["NEXT_DAYS"] = "NEXT_DAYS";
  TimeConstraintType2["NEXT_HOURS"] = "NEXT_HOURS";
  TimeConstraintType2["BETWEEN"] = "BETWEEN";
  TimeConstraintType2["TODAY"] = "TODAY";
  TimeConstraintType2["THIS_MONTH"] = "THIS_MONTH";
})(TimeConstraintType || (TimeConstraintType = {}));
var ValueConstraintType;
(function(ValueConstraintType2) {
  ValueConstraintType2["UNKNOWN_VALUE_CONSTRAINT_TYPE"] = "UNKNOWN_VALUE_CONSTRAINT_TYPE";
  ValueConstraintType2["NONE"] = "NONE";
  ValueConstraintType2["PREDEFINED_VALUES"] = "PREDEFINED_VALUES";
  ValueConstraintType2["MIN_MAX_RANGE"] = "MIN_MAX_RANGE";
  ValueConstraintType2["RANGE"] = "RANGE";
})(ValueConstraintType || (ValueConstraintType = {}));
var SaleType;
(function(SaleType2) {
  SaleType2["UNKNOWN_SALE_TYPE"] = "UNKNOWN_SALE_TYPE";
  SaleType2["CYBER_MONDAY"] = "CYBER_MONDAY";
  SaleType2["NOVEMBER_SALE"] = "NOVEMBER_SALE";
  SaleType2["BLACK_FRIDAY"] = "BLACK_FRIDAY";
  SaleType2["DEVELOPER_SALE"] = "DEVELOPER_SALE";
})(SaleType || (SaleType = {}));
var DiscountType;
(function(DiscountType2) {
  DiscountType2["UNKNOWN_DISCOUNT_TYPE"] = "UNKNOWN_DISCOUNT_TYPE";
  DiscountType2["FIXED_AMOUNT"] = "FIXED_AMOUNT";
  DiscountType2["PERCENTAGE"] = "PERCENTAGE";
})(DiscountType || (DiscountType = {}));

// ../../node_modules/@wix/ambassador-devcenter-apps-v1-app-permission/build/es/http.impl.js
init_esm_shims();
var _appPermission = {
  createdDate: "google.protobuf.Timestamp",
  updatedDate: "google.protobuf.Timestamp"
};
var _createAppPermissionRequest = { appPermission: "_appPermission" };
var _createAppPermissionResponse = { appPermission: "_appPermission" };
function resolveWixDevcenterAppsPermissionsV1AppPermissionsServiceUrl(opts) {
  var domainToMappings = {
    "dev._base_domain_": [
      {
        srcPath: "/app-permissions-service",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/app-permissions-service",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/app-permissions-service",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/apps/v1/app-permissions",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/app-permissions-service",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/app-permissions-service",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
function createAppPermission(payload) {
  var _a = serializer(_createAppPermissionRequest, { _appPermission }), toReq = _a.toJSON, fromReq = _a.fromJSON;
  var fromRes = serializer(_createAppPermissionResponse, {
    _appPermission
  }).fromJSON;
  function __createAppPermission(_a2) {
    var host = _a2.host;
    var serializedData = toReq(payload);
    var metadata = {
      entityFqdn: "wix.devcenter.apps.v1.app_permission",
      method: "POST",
      methodFqn: "wix.devcenter.apps.permissions.v1.AppPermissionsService.CreateAppPermission",
      url: resolveWixDevcenterAppsPermissionsV1AppPermissionsServiceUrl({
        protoPath: "/v1/app-permissions",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: fromRes
    };
    return metadata;
  }
  __createAppPermission.fromReq = fromReq;
  __createAppPermission.__isAmbassador = true;
  return __createAppPermission;
}

// ../dev-center-client/src/schemas.ts
init_esm_shims();

// ../../node_modules/@wix/ambassador-devcenter-apps-v1-app-template/build/es/types.impl.js
init_esm_shims();
var DevelopmentTool;
(function(DevelopmentTool2) {
  DevelopmentTool2["UNKNOWN_TEMPLATE_TYPE"] = "UNKNOWN_TEMPLATE_TYPE";
  DevelopmentTool2["DEV_CENTER"] = "DEV_CENTER";
  DevelopmentTool2["BLOCKS"] = "BLOCKS";
  DevelopmentTool2["CLI"] = "CLI";
})(DevelopmentTool || (DevelopmentTool = {}));
var PackageManager;
(function(PackageManager2) {
  PackageManager2["UNKNOWN_PACKAGE_MANAGER"] = "UNKNOWN_PACKAGE_MANAGER";
  PackageManager2["NPM"] = "NPM";
  PackageManager2["YARN"] = "YARN";
})(PackageManager || (PackageManager = {}));
var DeploymentHostType;
(function(DeploymentHostType2) {
  DeploymentHostType2["UNKNOWN_DEPLOYMENT_HOST_TYPE"] = "UNKNOWN_DEPLOYMENT_HOST_TYPE";
  DeploymentHostType2["VERCEL"] = "VERCEL";
  DeploymentHostType2["NETLIFY"] = "NETLIFY";
})(DeploymentHostType || (DeploymentHostType = {}));
var NamespaceType;
(function(NamespaceType2) {
  NamespaceType2["UNKNOWN_TYPE"] = "UNKNOWN_TYPE";
  NamespaceType2["ACCOUNT"] = "ACCOUNT";
  NamespaceType2["COMPANY"] = "COMPANY";
})(NamespaceType || (NamespaceType = {}));
var Status3;
(function(Status4) {
  Status4["ACTIVE"] = "ACTIVE";
  Status4["DEPRECATED"] = "DEPRECATED";
  Status4["ARCHIVED"] = "ARCHIVED";
})(Status3 || (Status3 = {}));
var AppType2;
(function(AppType3) {
  AppType3["SITE_APP"] = "SITE_APP";
  AppType3["API_KEY"] = "API_KEY";
  AppType3["VELO_APP"] = "VELO_APP";
  AppType3["PRIVATE_APP"] = "PRIVATE_APP";
  AppType3["HEADLESS"] = "HEADLESS";
  AppType3["FUNCTION_ACTIVATION_APP"] = "FUNCTION_ACTIVATION_APP";
})(AppType2 || (AppType2 = {}));
var CreatedByType2;
(function(CreatedByType3) {
  CreatedByType3["USER"] = "USER";
  CreatedByType3["APP"] = "APP";
})(CreatedByType2 || (CreatedByType2 = {}));
var ShowConsentBehaviour;
(function(ShowConsentBehaviour2) {
  ShowConsentBehaviour2["DEFAULT_BEHAVIOUR"] = "DEFAULT_BEHAVIOUR";
  ShowConsentBehaviour2["SHOW_CONSENT"] = "SHOW_CONSENT";
  ShowConsentBehaviour2["SKIP_CONSENT"] = "SKIP_CONSENT";
})(ShowConsentBehaviour || (ShowConsentBehaviour = {}));
var RedirectTarget;
(function(RedirectTarget2) {
  RedirectTarget2["UNKNOWN_REDIRECT_TARGET"] = "UNKNOWN_REDIRECT_TARGET";
  RedirectTarget2["DASHBOARD"] = "DASHBOARD";
  RedirectTarget2["EDITOR"] = "EDITOR";
})(RedirectTarget || (RedirectTarget = {}));
var OpenButtonAction;
(function(OpenButtonAction2) {
  OpenButtonAction2["UNKNOWN_ACTION"] = "UNKNOWN_ACTION";
  OpenButtonAction2["OPEN_COMPONENT_ID"] = "OPEN_COMPONENT_ID";
})(OpenButtonAction || (OpenButtonAction = {}));

// ../dev-center-client/src/schemas.ts
var createAppSchema = z.object({
  app: z.object({
    id: z.string()
  })
});
var createOAuthAppSchema = z.object({
  oAuthApp: z.object({
    id: z.string()
  })
});
var topologySchema = z.object({
  compId: z.string(),
  compName: z.string(),
  compType: z.literal(ComponentType.TOPOLOGY),
  compData: z.object({
    topology: z.object({
      topology: z.object({
        backendApiUrl: z.string().min(1)
      })
    })
  })
}).passthrough();
var componentsSchema = z.array(
  z.discriminatedUnion("compType", [
    topologySchema,
    ...Object.values(ComponentType).filter((type) => type !== ComponentType.TOPOLOGY).map((type) => {
      return z.object({
        compId: z.string(),
        compName: z.string().optional(),
        compData: z.unknown(),
        compType: z.literal(type)
      }).passthrough();
    })
  ])
);
var appByVersionSchema = z.object({
  appId: z.string(),
  data: z.object({
    components: componentsSchema
  })
});
var appVersionSchema = z.object({
  appVersions: z.array(
    z.object({
      /** Application ID */
      id: z.string(),
      fullVersion: z.string().optional().default("0.0.0"),
      /** indicates if this is the draft instance of the version. */
      state: z.string().optional(),
      /** status of the app */
      status: z.string().optional()
    })
  )
});
var appSecretsSchema = z.object({
  appSecret: z.string(),
  webhookPublicKey: z.string()
});
var appSchema = z.object({
  app: z.object({
    appSecrets: appSecretsSchema
  })
});
var createDevelopmentSiteSchema = z.object({
  developmentSite: z.object({
    id: z.string(),
    siteName: z.string()
  })
});
var getDevelopmentSitesSchema = z.object({
  developmentSites: z.array(z.object({}))
});
var getDevelopmentSitesLimitSchema = z.object({
  limit: z.number()
});
var developerAppSchema = z.object({
  id: z.string(),
  appData: z.object({
    appName: z.string()
  })
});
var getDeveloperAppsSchema = z.object({
  myApps: z.array(developerAppSchema)
});
var requiredApp = z.object({ id: z.string(), name: z.string() });
var queryAppSchema = z.object({
  apps: z.array(requiredApp)
});
var marketListingSchema = z.object({
  id: z.string(),
  appId: z.string(),
  revision: z.string(),
  languageCode: z.string(),
  installationRequirement: z.object({
    requiredApps: z.array(
      z.object({
        id: z.string(),
        slug: z.string()
      })
    )
  }).optional()
});
var managedAppsSchema = z.object({
  managedApps: z.array(
    z.object({
      appId: z.string(),
      installationDetails: z.object({
        group: z.nativeEnum(AppGroup)
      })
    })
  )
});
var cliCLoneDataSchema = z.object({
  repositoryUrl: z.string(),
  pathInRepository: z.string().optional()
});
var appTemplateSchema = z.object({
  id: z.string(),
  title: z.string(),
  subtitle: z.string(),
  developmentTools: z.array(z.nativeEnum(DevelopmentTool)),
  cliCloneData: cliCLoneDataSchema.optional()
});
var queryAppTemplateSchema = z.object({
  appTemplates: z.array(appTemplateSchema)
});
var cliAppTemplateSchema = appTemplateSchema.extend({
  cliCloneData: cliCLoneDataSchema
});

// ../dev-center-client/src/dev-center-client.ts
var EMPTY_SITE_TEMPLATE = "24bafe8a-7c80-4b63-8a71-08752062f246";
var DevCenterClient = class {
  constructor(httpClient) {
    this.httpClient = httpClient;
    const { minRetryTimeout } = getTestOverrides();
    this.retryOptions = {
      retries: 3,
      minTimeout: minRetryTimeout,
      maxTimeout: 3 * 1e3
    };
  }
  retryOptions;
  addPermission = async ({
    appId,
    permissionId
  }) => {
    try {
      const { data } = await pRetry(
        () => this.httpClient.request(
          createAppPermission({
            appPermission: {
              appId,
              permission: {
                permissionId
              }
            }
          })
        ),
        this.retryOptions
      );
      return data;
    } catch (e) {
      throw new CliError({
        code: CliErrorCode.FailedToAddPermission(),
        cause: e
      });
    }
  };
  createApp = async (app) => {
    try {
      const { data } = await pRetry(
        () => this.httpClient.request(createApp({ app })),
        this.retryOptions
      );
      return createAppSchema.parse(data).app;
    } catch (e) {
      throw new CliError({
        code: CliErrorCode.FailedToCreateDevCenterApp(),
        cause: e
      });
    }
  };
  createOAuthApp = async (oAuthApp) => {
    try {
      const { data } = await this.httpClient.request(
        createOAuthApp({ oAuthApp })
      );
      return createOAuthAppSchema.parse(data).oAuthApp;
    } catch (e) {
      throw new CliError({
        code: CliErrorCode.FailedToCreateDevCenterOAuthApp(),
        cause: e
      });
    }
  };
  updateOAuthApp = async (oAuthApp) => {
    try {
      const { data } = await this.httpClient.request(
        updateOAuthApp({
          oAuthApp,
          mask: ["allowedDomains", "allowedRedirectUris"]
        })
      );
      return createOAuthAppSchema.parse(data).oAuthApp;
    } catch (e) {
      throw new CliError({
        code: CliErrorCode.FailedToUpdateDevCenterOAuthApp(),
        cause: e
      });
    }
  };
  getAppSecret = async (req) => {
    try {
      const { data } = await pRetry(
        () => this.httpClient.request(
          getApp({
            withSecrets: true,
            ...req
          })
        ),
        this.retryOptions
      );
      return appSchema.parse(data).app.appSecrets;
    } catch (e) {
      if (isHttpError(e) && e.response?.status === 403 && isHttpApplicationError(e) && e.response.data.details.applicationError.code === "PERMISSION_DENIED") {
        throw new CliError({
          code: CliErrorCode.DeniedPermissionToGetAppSecret(),
          cause: e
        });
      }
      throw new CliError({
        code: CliErrorCode.FailedToGetAppSecret(),
        cause: e
      });
    }
  };
  queryAppVersion = async ({ appId }) => {
    try {
      const { data } = await pRetry(
        () => this.httpClient.request(
          queryAppVersion({
            query: {
              paging: {
                limit: 1
              },
              filter: { appId },
              sort: [{ fieldName: "updatedDate", order: SortOrder.DESC }]
            }
          })
        ),
        this.retryOptions
      );
      const parsedData = appVersionSchema.parse(data);
      const lastVersion = parsedData.appVersions[0];
      if (!lastVersion) {
        throw new CliError({
          code: CliErrorCode.AppForVersionNotFound({
            version: "latest"
          }),
          cause: null
        });
      }
      return {
        latestVersion: lastVersion.fullVersion,
        isPublished: lastVersion.state === State.PRODUCTION,
        isLocked: lastVersion.state === State.LOCKED_DRAFT
      };
    } catch (cause) {
      if (isHttpError(cause) && cause.response?.status === 404) {
        throw new CliError({
          code: CliErrorCode.AppForVersionNotFound({
            version: "latest"
          }),
          cause
        });
      }
      throw new CliError({
        code: CliErrorCode.FailedToFetchAppForVersion(),
        cause
      });
    }
  };
  /**
   * This method is deprecated.
   * Use `queryAppVersion()` if you need to get information about app version.
   */
  getAppByVersion = async ({
    appId,
    version = "latest"
  }) => {
    try {
      const { data } = await pRetry(
        () => this.httpClient.request(getAppByVersion({ appId, version })),
        this.retryOptions
      );
      return appByVersionSchema.parse(data);
    } catch (cause) {
      if (isHttpError(cause) && cause.response?.status === 404) {
        throw new CliError({
          code: CliErrorCode.AppForVersionNotFound({
            version
          }),
          cause
        });
      }
      throw new CliError({
        code: CliErrorCode.FailedToFetchAppForVersion(),
        cause
      });
    }
  };
  createDevSite = async () => {
    try {
      const { data } = await pRetry(
        () => this.httpClient.request(
          createDevelopmentSite({
            templateId: EMPTY_SITE_TEMPLATE
          })
        ),
        this.retryOptions
      );
      const { id, siteName } = createDevelopmentSiteSchema.parse(data).developmentSite;
      const site = {
        id,
        displayName: siteName.replaceAll("-", " "),
        viewUrl: ""
      };
      return site;
    } catch (e) {
      throw new CliError({
        code: CliErrorCode.FailedToCreateDevelopmentSite(),
        cause: e
      });
    }
  };
  getDevSitesCount = async () => {
    try {
      const { data } = await pRetry(
        () => this.httpClient.request(getDevelopmentSites({})),
        this.retryOptions
      );
      return getDevelopmentSitesSchema.parse(data).developmentSites.length;
    } catch (e) {
      throw new CliError({
        code: CliErrorCode.FailedToGetDevelopmentSites(),
        cause: e
      });
    }
  };
  getDeveloperApps = async () => {
    try {
      const { data } = await pRetry(
        () => this.httpClient.request(
          queryMyApps({
            query: {
              paging: {
                limit: 500
              }
            }
          })
        ),
        this.retryOptions
      );
      return getDeveloperAppsSchema.parse(data).myApps;
    } catch (e) {
      throw new CliError({
        code: CliErrorCode.FailedToGetDeveloperApps(),
        cause: e
      });
    }
  };
  getDevSitesLimit = async () => {
    try {
      const { data } = await pRetry(
        () => this.httpClient.request(getDevelopmentSitesLimit({})),
        this.retryOptions
      );
      return getDevelopmentSitesLimitSchema.parse(data).limit;
    } catch (e) {
      throw new CliError({
        code: CliErrorCode.FailedToGetDevelopmentSitesLimit(),
        cause: e
      });
    }
  };
  getMarketListing = async (appId) => {
    try {
      const { data } = await pRetry(
        () => this.httpClient.request(
          queryMarketListing({
            query: {
              filter: {
                appId,
                languageCode: "en"
              }
            }
          })
        ),
        this.retryOptions
      );
      return data.marketListing?.[0] ? marketListingSchema.parse(data.marketListing[0]) : void 0;
    } catch (e) {
      throw new CliError({
        code: CliErrorCode.FailedToGetMarketListing(),
        cause: e
      });
    }
  };
  getRequiredAppIds = async (appId) => {
    try {
      const marketListing = await this.getMarketListing(appId);
      return marketListing?.installationRequirement?.requiredApps.map(
        ({ id }) => id
      ) ?? [];
    } catch (e) {
      throw new CliError({
        code: CliErrorCode.FailedToGetRequiredApps(),
        cause: e
      });
    }
  };
  getRequiredApps = async (appId) => {
    const requiredAppIds = await this.getRequiredAppIds(appId);
    if (requiredAppIds.length === 0) {
      return [];
    }
    try {
      const { data } = await pRetry(
        () => this.httpClient.request(
          queryApp({
            query: {
              filter: {
                id: {
                  $in: requiredAppIds
                }
              }
            }
          })
        ),
        this.retryOptions
      );
      return queryAppSchema.parse(data).apps;
    } catch (e) {
      throw new CliError({
        code: CliErrorCode.FailedToQueryApps(),
        cause: e
      });
    }
  };
  addRequiredApp = async ({
    appId,
    appIdToRequire
  }) => {
    try {
      const marketListing = await this.getMarketListing(appId);
      const appAlreadyRequired = marketListing?.installationRequirement?.requiredApps.some(
        (requiredApp2) => requiredApp2.id === appIdToRequire
      );
      if (marketListing && !appAlreadyRequired) {
        const newRequiredApps = [
          ...marketListing.installationRequirement?.requiredApps ?? [],
          {
            id: appIdToRequire,
            slug: appIdToRequire
          }
        ];
        await pRetry(
          () => this.httpClient.request(
            updateMarketListing({
              marketListing: {
                ...marketListing,
                appId,
                installationRequirement: {
                  requiredApps: newRequiredApps
                }
              },
              mask: ["installationRequirement.requiredApps"]
            })
          ),
          this.retryOptions
        );
      }
    } catch (e) {
      throw new CliError({
        code: CliErrorCode.FailedToAddRequiredApp(),
        cause: e
      });
    }
  };
  queryCliAppTemplates = async () => {
    try {
      const { data } = await pRetry(
        () => this.httpClient.request(listAppTemplates({})),
        this.retryOptions
      );
      const { appTemplates } = queryAppTemplateSchema.parse(data);
      return appTemplates.filter(
        (template) => template.developmentTools.includes(DevelopmentTool.CLI)
      ).map((template) => cliAppTemplateSchema.parse(template));
    } catch (e) {
      throw new CliError({
        code: CliErrorCode.FailedToQueryCliAppTemplates(),
        cause: e
      });
    }
  };
  getCliAppTemplate = async (id) => {
    const cliAppTemplates = await this.queryCliAppTemplates();
    return cliAppTemplates.find((template) => template.id === id) ?? null;
  };
  createAppFromTemplate = async ({ templateId, name }, template) => {
    try {
      const { data } = await pRetry(() => {
        const createFromTemplate = template.developmentTools.includes(
          DevelopmentTool.BLOCKS
        ) ? createBlocksAppFromTemplate : createAppFromTemplate;
        return this.httpClient.request(
          createFromTemplate({
            templateId,
            templateConfig: { appInfo: { name } }
          })
        );
      }, this.retryOptions);
      return createAppSchema.parse(data).app;
    } catch (e) {
      throw new CliError({
        code: CliErrorCode.FailedToCreateDevCenterAppFromTemplate(),
        cause: e,
        info: { templateId }
      });
    }
  };
  getAppInstallationDetails = async (appId) => {
    try {
      const { data } = await pRetry(
        () => this.httpClient.request(managedApps({})),
        this.retryOptions
      );
      return managedAppsSchema.parse(data).managedApps.find((app) => app.appId === appId)?.installationDetails;
    } catch (e) {
      throw new CliError({
        code: CliErrorCode.FailedToGetAppInstallationDetails({ appId }),
        cause: e
      });
    }
  };
  isAppInstalled = async (appId) => {
    const installationDetails = await this.getAppInstallationDetails(appId);
    return installationDetails?.group === AppGroup.PENDING || installationDetails?.group === AppGroup.INSTALLED;
  };
  isAppInstalledInEditor = async (appId) => {
    const installationDetails = await this.getAppInstallationDetails(appId);
    return installationDetails?.group === AppGroup.INSTALLED;
  };
};

// ../dev-center-client/src/useDevCenterClient.ts
function useDevCenterClient() {
  const httpClient = useHttpClient({ type: "backoffice" });
  return (0, import_react.useMemo)(() => new DevCenterClient(httpClient), [httpClient]);
}

// ../dev-center-client/src/index.ts
init_esm_shims();

export {
  useDevCenterClient
};
