// @wix/cli/build/chunk-WYHHEOWO.js
import { createRequire as _createRequire } from 'node:module';
const require = _createRequire(import.meta.url);
import {
  z
} from "./chunk-ZXYGJZOO.js";
import {
  init_esm_shims
} from "./chunk-4EFJZ3GQ.js";

// ../cli-test-overrides/src/index.ts
init_esm_shims();
import process from "node:process";
var testOverridesSchema = z.object({
  biLoggerBaseUrl: z.string().url().optional(),
  devEditorBaseUrl: z.string().url().optional(),
  inkDebug: z.boolean().optional(),
  minRetryTimeout: z.number().optional(),
  pollInterval: z.number().optional(),
  pollTimeout: z.number().optional(),
  debounceQueueWait: z.number().optional(),
  sentryDsn: z.string().optional(),
  userBrowserTestEndpoint: z.string().url().optional(),
  dataDir: z.string().optional(),
  wixCliDependencyVersion: z.string().optional(),
  wixCliAppDependencyVersion: z.string().optional(),
  disableTTY: z.boolean().optional(),
  waitForLoginInterval: z.number().optional(),
  registryUrl: z.string().optional(),
  fsCacheDir: z.string().optional(),
  localTunnel: z.boolean().optional(),
  customElementsCreationDelay: z.number().optional(),
  sdkAuthServerBaseUrl: z.string().optional(),
  miniflareRequestsRewrites: z.array(z.object({ from: z.string(), to: z.string() })).optional()
});
var TEST_OVERRIDES_ENV_VAR = "WIX_CLI_TEST_OVERRIDES";
function getTestOverrides() {
  const rawOverrides = process.env[TEST_OVERRIDES_ENV_VAR];
  if (!rawOverrides) {
    return {};
  }
  return testOverridesSchema.parse(JSON.parse(rawOverrides));
}
function isTestEnv() {
  return !!process.env[TEST_OVERRIDES_ENV_VAR];
}
function setupTestEnv() {
  if (isTestEnv()) {
    process.stdout.columns = 300;
    process.stderr.columns = 300;
    process.stdin.isTTY = !getTestOverrides().disableTTY;
    process.stdin.setRawMode = () => process.stdin;
  }
}

export {
  getTestOverrides,
  setupTestEnv
};
