// @wix/cli/build/chunk-6RDHKDZ4.js
import { createRequire as _createRequire } from 'node:module';
const require = _createRequire(import.meta.url);
import {
  init_esm_shims
} from "./chunk-4EFJZ3GQ.js";

// ../../node_modules/@wix/ambassador-velo-ide-v1-gridapp-filestructure/build/es/types.impl.js
init_esm_shims();
var FileType;
(function(FileType2) {
  FileType2["FILE"] = "FILE";
  FileType2["DIRECTORY"] = "DIRECTORY";
})(FileType || (FileType = {}));
var GridAppLayout;
(function(GridAppLayout2) {
  GridAppLayout2["UNKNOWN_GRID_APP_LAYOUT"] = "UNKNOWN_GRID_APP_LAYOUT";
  GridAppLayout2["ONLINE"] = "ONLINE";
  GridAppLayout2["GITHUB"] = "GITHUB";
  GridAppLayout2["GITHUB_BLOCKS"] = "GITHUB_BLOCKS";
  GridAppLayout2["WIX_IDE"] = "WIX_IDE";
})(GridAppLayout || (GridAppLayout = {}));
var Action;
(function(Action2) {
  Action2["UNKNOWN"] = "UNKNOWN";
  Action2["UPDATE"] = "UPDATE";
  Action2["CREATE"] = "CREATE";
  Action2["MOVE"] = "MOVE";
  Action2["OVERWRITE"] = "OVERWRITE";
  Action2["COPY"] = "COPY";
})(Action || (Action = {}));
var WebhookIdentityType;
(function(WebhookIdentityType2) {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
})(WebhookIdentityType || (WebhookIdentityType = {}));

export {
  GridAppLayout
};
