// @wix/cli/build/InstallCommand-FTCIFD2A.js
import { createRequire as _createRequire } from 'node:module';
const require = _createRequire(import.meta.url);
import {
  createDependenciesService
} from "./chunk-D2J3WUXG.js";
import "./chunk-RUPEFN2Q.js";
import {
  useVeloClient
} from "./chunk-Z2OLEKZU.js";
import "./chunk-UGNUBZYN.js";
import "./chunk-A6POD5GK.js";
import "./chunk-6RDHKDZ4.js";
import "./chunk-ZS5GINIZ.js";
import {
  useProjectModel
} from "./chunk-FLKYXGBD.js";
import "./chunk-O3SZ563W.js";
import "./chunk-RWTQFIPN.js";
import "./chunk-W54PG6O5.js";
import "./chunk-4R5RNIW3.js";
import "./chunk-67NUBEMP.js";
import "./chunk-WWUS5636.js";
import "./chunk-SJKVYZPL.js";
import {
  Trans
} from "./chunk-UQLHXG4X.js";
import "./chunk-KJ37XZQA.js";
import {
  Alert,
  Badge,
  Spinner,
  Text,
  useAsync
} from "./chunk-RXATQXHV.js";
import {
  require_react
} from "./chunk-NRAQAV6T.js";
import "./chunk-SQ3KPTIH.js";
import "./chunk-Z4MHKCET.js";
import "./chunk-C4SUTB4O.js";
import "./chunk-TKDDSH55.js";
import "./chunk-62BYZXT7.js";
import "./chunk-WYHHEOWO.js";
import "./chunk-Z3Y5NFYG.js";
import "./chunk-ZXYGJZOO.js";
import {
  require_lib
} from "./chunk-IVMK6TVY.js";
import {
  __toESM,
  init_esm_shims
} from "./chunk-4EFJZ3GQ.js";

// ../cli-site/src/components/InstallCommand/InstallCommand.tsx
init_esm_shims();
var import_react = __toESM(require_react(), 1);
var import_variant = __toESM(require_lib(), 1);
var InstallCommand = ({ packageName }) => {
  const { model } = useProjectModel();
  const veloClient = useVeloClient();
  const { status } = useAsync(async () => {
    const { projectFolder } = model;
    const service = await createDependenciesService(projectFolder, veloClient);
    await service.install(packageName);
  }, []);
  return (0, import_variant.match)(status, {
    Loading: () => /* @__PURE__ */ import_react.default.createElement(
      Spinner,
      {
        text: packageName ? /* @__PURE__ */ import_react.default.createElement(
          Trans,
          {
            i18nKey: "install_command.package.loading",
            values: { packageName },
            components: [/* @__PURE__ */ import_react.default.createElement(Text, { bold: true })]
          }
        ) : /* @__PURE__ */ import_react.default.createElement(Trans, { i18nKey: "install_command.all_packages.loading" })
      }
    ),
    Error: () => null,
    Success: () => /* @__PURE__ */ import_react.default.createElement(Alert, { type: "success" }, /* @__PURE__ */ import_react.default.createElement(Badge, { skin: "success" }, /* @__PURE__ */ import_react.default.createElement(Trans, { i18nKey: "general.success" })), " ", packageName ? /* @__PURE__ */ import_react.default.createElement(
      Trans,
      {
        i18nKey: "install_command.package.success",
        values: { packageName },
        components: [/* @__PURE__ */ import_react.default.createElement(Text, { bold: true })]
      }
    ) : /* @__PURE__ */ import_react.default.createElement(Trans, { i18nKey: "install_command.all_packages.success" }))
  });
};
export {
  InstallCommand
};
