// @wix/cli/build/PreviewCommand-XHDA2QXP.js
import { createRequire as _createRequire } from 'node:module';
const require = _createRequire(import.meta.url);
import {
  DeploymentSource,
  getDefaultBranch,
  getDeploymentSourceData
} from "./chunk-ZVYFPO7Z.js";
import "./chunk-NBOB6NEY.js";
import "./chunk-BTL5GHWG.js";
import "./chunk-FTL4OWDC.js";
import {
  useProjectModel
} from "./chunk-WUREUJ5V.js";
import "./chunk-TMLDHPS6.js";
import "./chunk-V23RKBE2.js";
import {
  DeploymentStatus,
  useDocumentManagementClient
} from "./chunk-YCMVJTQP.js";
import "./chunk-ZMOO2TWX.js";
import "./chunk-CSMKGG4V.js";
import {
  useVeloClient
} from "./chunk-Z2OLEKZU.js";
import "./chunk-UGNUBZYN.js";
import "./chunk-A6POD5GK.js";
import "./chunk-6RDHKDZ4.js";
import "./chunk-ZS5GINIZ.js";
import "./chunk-RWTQFIPN.js";
import "./chunk-W54PG6O5.js";
import "./chunk-4R5RNIW3.js";
import "./chunk-67NUBEMP.js";
import {
  writeCrashReport
} from "./chunk-WWUS5636.js";
import "./chunk-SJKVYZPL.js";
import {
  Trans,
  useTranslation
} from "./chunk-UQLHXG4X.js";
import "./chunk-KJ37XZQA.js";
import {
  Alert,
  Link,
  SelectInput,
  Spinner,
  useAsync,
  useAsyncCallback
} from "./chunk-RXATQXHV.js";
import {
  require_react
} from "./chunk-NRAQAV6T.js";
import "./chunk-SQ3KPTIH.js";
import "./chunk-Z4MHKCET.js";
import "./chunk-C4SUTB4O.js";
import {
  getDebugLogFilePath
} from "./chunk-TKDDSH55.js";
import "./chunk-62BYZXT7.js";
import "./chunk-WYHHEOWO.js";
import "./chunk-Z3Y5NFYG.js";
import "./chunk-ZXYGJZOO.js";
import {
  require_lib
} from "./chunk-IVMK6TVY.js";
import {
  __toESM,
  init_esm_shims
} from "./chunk-4EFJZ3GQ.js";

// ../cli-site-old/src/preview/PreviewCommand.tsx
init_esm_shims();
var import_react3 = __toESM(require_react(), 1);
var import_variant2 = __toESM(require_lib(), 1);

// ../cli-site-old/src/preview/SelectPreviewSource.tsx
init_esm_shims();
var import_react = __toESM(require_react(), 1);
var import_variant = __toESM(require_lib(), 1);
var SelectPreviewSource = ({
  projectFolder,
  onSubmit
}) => {
  const { t } = useTranslation();
  const { status } = useAsync(
    () => getDefaultBranch(
      (e) => writeCrashReport(getDebugLogFilePath(projectFolder), e)
    ),
    []
  );
  return (0, import_variant.match)(status, {
    Error: () => null,
    Loading: () => /* @__PURE__ */ import_react.default.createElement(Spinner, null),
    Success: ({ result: defaultBranchName }) => /* @__PURE__ */ import_react.default.createElement(
      SelectInput,
      {
        label: t("preview_command.prompt_what_to_preview"),
        options: [
          {
            key: "remote",
            value: DeploymentSource.remote(),
            title: defaultBranchName ? t("preview_command.prompt_what_to_preview_choice_latest", {
              branchName: defaultBranchName
            }) : t(
              "preview_command.prompt_what_to_preview_choice_latest_fallback"
            )
          },
          {
            key: "local",
            value: DeploymentSource.local(),
            title: t("preview_command.prompt_what_to_preview_choice_local")
          }
        ],
        onSubmit
      }
    )
  });
};

// ../cli-site-old/src/preview/useCreatePreview.ts
init_esm_shims();
var import_react2 = __toESM(require_react(), 1);
function useCreatePreview() {
  const { model } = useProjectModel();
  const veloClient = useVeloClient();
  const dmClient = useDocumentManagementClient();
  return (0, import_react2.useCallback)(
    async (previewSource) => {
      const deploymentSourceData = await getDeploymentSourceData(
        model,
        previewSource === "local" ? DeploymentSource.local() : DeploymentSource.remote()
      );
      const preview = await veloClient.deploySitePreview(deploymentSourceData);
      return {
        deploymentId: preview.deploymentId,
        previewUrl: preview.deploymentShortUrl ?? preview.deploymentUrl,
        deploymentStatus: preview.isPublishPipelineDeployment ? await dmClient.getDeploymentDetails(preview.deploymentId) : DeploymentStatus.Success()
      };
    },
    [dmClient, model, veloClient]
  );
}

// ../cli-site-old/src/preview/PreviewCommand.tsx
var PreviewCommand = (options) => {
  const { t } = useTranslation();
  const { model } = useProjectModel();
  const createPreview = useCreatePreview();
  const { status, execute: handleCreatingPreview } = useAsyncCallback(
    async (_, previewSource) => {
      const { previewUrl, deploymentStatus } = await createPreview(previewSource);
      if ((0, import_variant2.isType)(deploymentStatus, DeploymentStatus.Error)) {
        throw deploymentStatus.deploymentError;
      }
      return previewUrl;
    }
  );
  useAsync(async () => {
    if (options.source) {
      void handleCreatingPreview(options.source);
    }
  }, []);
  return /* @__PURE__ */ import_react3.default.createElement(import_react3.default.Fragment, null, !options.source && /* @__PURE__ */ import_react3.default.createElement(
    SelectPreviewSource,
    {
      projectFolder: model.projectFolder,
      onSubmit: ({ value }) => handleCreatingPreview(value.type)
    }
  ), (0, import_variant2.match)(status, {
    NotRequested: () => null,
    Error: () => null,
    Loading: () => /* @__PURE__ */ import_react3.default.createElement(Spinner, { text: t("preview_command.creating_preview") }),
    Success: ({ result: previewUrl }) => /* @__PURE__ */ import_react3.default.createElement(Alert, { type: "success" }, /* @__PURE__ */ import_react3.default.createElement(
      Trans,
      {
        i18nKey: "preview_command.preview_created",
        components: [/* @__PURE__ */ import_react3.default.createElement(Link, { url: previewUrl })]
      }
    ))
  }));
};
export {
  PreviewCommand
};
