// @wix/cli/build/PreviewCommand-IK553J6G.js
import { createRequire as _createRequire } from 'node:module';
const require = _createRequire(import.meta.url);
import {
  useCreatePreview
} from "./chunk-TO26ON5U.js";
import {
  DeploymentStatus
} from "./chunk-YCMVJTQP.js";
import "./chunk-ZMOO2TWX.js";
import "./chunk-CSMKGG4V.js";
import "./chunk-5LRGJ3DG.js";
import "./chunk-RUPEFN2Q.js";
import "./chunk-Z2OLEKZU.js";
import "./chunk-UGNUBZYN.js";
import "./chunk-A6POD5GK.js";
import "./chunk-6RDHKDZ4.js";
import "./chunk-ZS5GINIZ.js";
import "./chunk-FLKYXGBD.js";
import "./chunk-4R5RNIW3.js";
import "./chunk-67NUBEMP.js";
import "./chunk-WWUS5636.js";
import "./chunk-SJKVYZPL.js";
import {
  Trans
} from "./chunk-UQLHXG4X.js";
import "./chunk-KJ37XZQA.js";
import {
  Alert,
  Badge,
  Link,
  Spinner,
  useAsync
} from "./chunk-RXATQXHV.js";
import {
  require_react
} from "./chunk-NRAQAV6T.js";
import "./chunk-SQ3KPTIH.js";
import "./chunk-Z4MHKCET.js";
import "./chunk-C4SUTB4O.js";
import "./chunk-TKDDSH55.js";
import "./chunk-62BYZXT7.js";
import "./chunk-WYHHEOWO.js";
import "./chunk-Z3Y5NFYG.js";
import "./chunk-ZXYGJZOO.js";
import {
  require_lib
} from "./chunk-IVMK6TVY.js";
import {
  __toESM,
  init_esm_shims
} from "./chunk-4EFJZ3GQ.js";

// ../cli-site/src/components/PreviewCommand/PreviewCommand.tsx
init_esm_shims();
var import_react = __toESM(require_react(), 1);
var import_variant = __toESM(require_lib(), 1);
var PreviewCommand = () => {
  const createPreview = useCreatePreview();
  const { status } = useAsync(async () => {
    const { previewUrl, deploymentStatus } = await createPreview();
    if ((0, import_variant.isType)(deploymentStatus, DeploymentStatus.Error)) {
      throw deploymentStatus.deploymentError;
    }
    return previewUrl;
  }, []);
  return (0, import_variant.match)(status, {
    Error: () => null,
    Loading: () => /* @__PURE__ */ import_react.default.createElement(Spinner, { text: /* @__PURE__ */ import_react.default.createElement(Trans, { i18nKey: "preview_command.loading" }) }),
    Success: ({ result: previewUrl }) => /* @__PURE__ */ import_react.default.createElement(Alert, { type: "success" }, /* @__PURE__ */ import_react.default.createElement(Badge, { skin: "success" }, /* @__PURE__ */ import_react.default.createElement(Trans, { i18nKey: "general.success" })), " ", /* @__PURE__ */ import_react.default.createElement(
      Trans,
      {
        i18nKey: "preview_command.success",
        components: [/* @__PURE__ */ import_react.default.createElement(Link, { url: previewUrl })]
      }
    ))
  });
};
export {
  PreviewCommand
};
